<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Oomycete Data Curation – SWELTR</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./curate_amf.html" rel="prev">
<link href="./assets/favicon.ico" rel="icon">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-59898bd1c6b9d2bb783127feaa000c76.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-d329e753491efaeac79c98c4b193a686.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-59898bd1c6b9d2bb783127feaa000c76.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-1bbff6656cb516badf17a29f6ac1cfc7.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-1374017376f980318f71eaad7b0ee285.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="site_libs/bootstrap/bootstrap-1bbff6656cb516badf17a29f6ac1cfc7.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/quarto-contrib/downloadthis-1.9.1/downloadthis.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="site_libs/core-js-2.5.3/shim.min.js"></script><script src="site_libs/react-18.2.0/react.min.js"></script><script src="site_libs/react-18.2.0/react-dom.min.js"></script><script src="site_libs/reactwidget-2.0.0/react-tools.js"></script><link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="site_libs/reactable-0.4.4/reactable.css" rel="stylesheet">
<script src="site_libs/reactable-binding-0.4.4/reactable.js"></script><link rel="stylesheet" href="assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const queryPrefersDark = window.matchMedia('(prefers-color-scheme: dark)');
    const darkModeDefault = queryPrefersDark.matches;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    queryPrefersDark.addEventListener("change", e => {
      if(window.localStorage.getItem("quarto-color-scheme") !== null)
        return;
      const alternate = e.matches
      toggleColorMode(alternate);
      localAlternateSentinel = e.matches ? 'alternate' : 'default'; // this is used alongside local storage!
      toggleGiscusIfUsed(alternate, darkModeDefault);
    });
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="./index.html" class="navbar-brand navbar-brand-logo">
    <img src="./assets/icon.png" alt="" class="navbar-logo"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/sweltr/depth/"> <i class="bi bi-github" role="img" aria-label="SWELTR Depth">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./curate_ssu.html">B. Data Curation</a></li><li class="breadcrumb-item"><a href="./curate_oo.html">Oomycete Data Curation</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">A. Processing</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./primers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Primer Removal</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./lotus3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ASV Inference</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">B. Data Curation</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./curate_ssu.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">SSU Data Curation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./curate_its.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ITS Data Curation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./curate_amf.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">AMF Data Curation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./curate_oo.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Oomycete Data Curation</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="3"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#overview" id="toc-overview" class="nav-link active" data-scroll-target="#overview">Overview</a>
  <ul class="collapse">
<li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data">Data</a></li>
  <li><a href="#citable-resources" id="toc-citable-resources" class="nav-link" data-scroll-target="#citable-resources">Citable resources</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</li>
  <li>
<a href="#workflow" id="toc-workflow" class="nav-link" data-scroll-target="#workflow">Workflow</a>
  <ul class="collapse">
<li><a href="#required-packages" id="toc-required-packages" class="nav-link" data-scroll-target="#required-packages">Required Packages</a></li>
  <li><a href="#review-results" id="toc-review-results" class="nav-link" data-scroll-target="#review-results">Review Results</a></li>
  <li><a href="#sample-data" id="toc-sample-data" class="nav-link" data-scroll-target="#sample-data">Sample Data</a></li>
  <li>
<a href="#prep-data-for-microeco" id="toc-prep-data-for-microeco" class="nav-link" data-scroll-target="#prep-data-for-microeco">Prep Data for microeco</a>
  <ul class="collapse">
<li><a href="#a.-taxonomy-table" id="toc-a.-taxonomy-table" class="nav-link" data-scroll-target="#a.-taxonomy-table">A. Taxonomy Table</a></li>
  <li><a href="#b.-sequence-table" id="toc-b.-sequence-table" class="nav-link" data-scroll-target="#b.-sequence-table">B. Sequence Table</a></li>
  <li><a href="#c.-sample-table" id="toc-c.-sample-table" class="nav-link" data-scroll-target="#c.-sample-table">C. Sample Table</a></li>
  </ul>
</li>
  <li>
<a href="#create-a-microtable-object" id="toc-create-a-microtable-object" class="nav-link" data-scroll-target="#create-a-microtable-object">Create a Microtable Object</a>
  <ul class="collapse">
<li><a href="#d.-add-representative-sequence" id="toc-d.-add-representative-sequence" class="nav-link" data-scroll-target="#d.-add-representative-sequence">D. Add Representative Sequence</a></li>
  </ul>
</li>
  <li>
<a href="#curate-the-data-set" id="toc-curate-the-data-set" class="nav-link" data-scroll-target="#curate-the-data-set">Curate the Data Set</a>
  <ul class="collapse">
<li><a href="#remove-non-fungal-asvs" id="toc-remove-non-fungal-asvs" class="nav-link" data-scroll-target="#remove-non-fungal-asvs">Remove non-fungal ASVs</a></li>
  <li><a href="#remove-negative-controls-nc" id="toc-remove-negative-controls-nc" class="nav-link" data-scroll-target="#remove-negative-controls-nc">Remove Negative Controls (NC)</a></li>
  <li><a href="#remove-low-count-samples" id="toc-remove-low-count-samples" class="nav-link" data-scroll-target="#remove-low-count-samples">Remove Low-Count Samples</a></li>
  </ul>
</li>
  </ul>
</li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">Summary</a></li>
  <li>Links</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/sweltr/depth/edit/main/curate_oo.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sweltr/depth/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="https://doi.org/10.25573/data.16828063"><i class="bi bi-link-45deg"></i>Data</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./curate_ssu.html">B. Data Curation</a></li><li class="breadcrumb-item"><a href="./curate_oo.html">Oomycete Data Curation</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Oomycete Data Curation</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
<p class="subtitle lead">Workflow for curating the ASV dataset–generating microeco &amp; phyloseq objects, removing spurious ASVs and low-count/negative control samples.</p>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="overview" class="level1"><h1>Overview</h1>
<p>On this page we use the phyloseq output from LotuS3 to format the otu, taxonomy, and sample data tables so we can create a microeco object. Once that is complete we curate the dataset by removing possible contaminants, negative control samples, NA kingdoms, and low-count samples.</p>
<section id="data" class="level2"><h2 class="anchored" data-anchor-id="data">Data</h2>
<p>All you need to run this workflow is the phyloseq object generated by the LotuS3 pipeline.</p>
<div class="callout callout-style-default callout-note no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<i class="fa-solid fa-download" aria-label="download"></i> &nbsp; phyloseq_oo.Rdata
</div>
</div>
<div class="callout-body-container callout-body">
<p><a href="#dnldts33142" onclick="fetch('data:application/x-gzip;base64,H4sIAAAAAAAAA+1dCXhTxRae0L1QWgFBEAQqKqKU7AsKY2ihINAWKIKChDQNbWzalCQFq6iILMoDsYAKiLIvsu+L7DuCoOwCsigoIMITVHgqy5tJJzczaW6btDdtgQzfoTNnzpz5Z/vvTHJzb9eEnrLInpEAgCAQHBwEgkJQFFTXJnXroOspi5M3l2iUagCCqyCtCASDCPQ3NDcz32bsD0RBdd1yIiz2PJ1dn2Y2oswoAB7bDAKhsgbRPeY3EPhDlVKWqwaKHS8RZoNGPJkxxCbYTY/WPcDl2iOpSnQR5G9Nyi7Fa5RlClVWF58fcpBNBz3FY9gXSe2S6wum+lOkRzKz5DI4VNlHJfogGUv+ChHCATtHGlPxeqBwvJyBHqPoYnyGFJPXG3jVV/dFeL2E/GbAd050sw+d6mWxMT7W44cQvJ9K/I3WUxKaftfQ9HvGg7Ev/SLUdWV/ySZFQnHrAIcwH/3VLwWGQBA2BPYpgfAghnp+9h9YV8KGjh50uI+ru+kEHleR85qndMsg18+gTDd7bNdCmLpDmgrjx5sQfNgt3ZBKlGIuy6tQ5d5C8hny2Q0UHS+3IIovJhOdsURatIca7Tser0LDkk14g7O9fH3lj3NAAz/4rEQh+nrh38huFYujsgRRk8K/4S8jGetFgb7+REOFz6l4aT+DCYT7PwixJ7qX91W9yd8YSueM1ylqLvL1PB0IlSfcy/P0XgjF7SNVVDyl8E8o/jwbrycxj5+zaMhi0N/Z6O98dBk7gmQyWp7rWfOwI56rDE5yS/8XnQ3Gob+j3Aw97QPjPPv0R3gUt3crq2uMzxkCz9chrYvqlC+i/zqXzl9ENS+MnGe/b9H+6A02q8os1MRJAFT9DSXwvhLtU0Q8e5WQtR6UqqKqoB5Uwtl/eI7hMS7HMfU6+PG8IDrlpaH792T3aWg0V2CHpVmfkTz6/cTfs6WHI2h4oqIBPMChuM8svA1TAYiSUOkgEDgHBkLlDOV1v09Fz//78fyxzoOuNGfkGoD//PCQW9q9H+m0BJQthLulpcj9v0hSfXcV1NtNkeZdOdEVVPY0SWhQd0bx2Ll/5i5i/4rOA+5emMhe6Nxkp2yTkci9wyPousF98lwJNjWLyXO/30HkQUfnOUNMCXUGQsUGPm48V0xeScF93t6P/Ctk0FQ0gAc0lNe8dFsPMa+VY92eQmA9li4E+s0vIehhYfyEthPGj8cQGHufQoQ338nfL2ET2u7/4qYjZ7GI9uho04zSB+ZRIDxIAc/3ID/7D4RACIRAqEwhwEuuUNGf/9+v4U0Puro+lC/LuPDN711l8Okp9CqqqvKhwHXg4DyD4t8lP85jYy6d69il6L85XhpbkdC/+73NY4fw1j4HQP3l6OxRlnv178dQDr8VCU4pweA+5LyIa+i/lr6XS7hAJYS8Lpbmfg0P964JHh6Ea39FtPFe7dd7FXcgBIIQwZ/z/35eW6FsssoIAXw259EXd++HjyESlqFwcc+QKa8g8Jyq5cWeKUovbJ2BQIW+/q8iaIv/6yg29PCgK8Xn7qLTAIRdKqqPXuy7r0KHADx5FBR9ZoCHUD3HB7+Ir+ISAGiyHzVTW0psnsIm34uIhHomVyAIGyrL3sCb3/lUFqz+DkJ/F+jht3WBQEIZnzEnWHhQ5rbQ4V76fSD9WdyDNt7Oexzoz3zJOaZaDeDf+x+KC55+IxII93bwYW3VySzZplyCN78HeFA4g25naZ47GggVE8oyP4v73VwgFA1ZSAa66SrqGhoIZQv34ffg5RpKO+9reWHzbil9U0Hk4V4ZkftzgfBnk87PyB714KRGCWm+sJT8dX+ukHvwZR88CzB9HlTCMzpLDLHoDEA/2xvff6BDR7tB6O8j/MVEH5WhzsYlm3gVSvvMNdyu4DKU9zX4+/nFxQSRh9+5BFXWz6Twc6MiSrRiQpC3v1W/H4PMLb3Ey3J4f4fnf+DaFwieQmD/HwhlCYH58+AEvL8W4uzr6V07AgSRp/18aQLffT9qJPhZ31JKN4iKf0K5mI3+yyeJBZTNr6DoO8vaeq6uihgw548gvr00/VxM8rx70cesznHPxNvo77hClegf9N8t6mudYu5fDnI+sxM/u8iL+4dqk+ehAzuPQTFnDd7gTRn6t/vOc84XSDzdk1HaOUg9h0nkaT9akc+GLkcuDkouv7qcIWKzF0aennFIfg8f0hlNEZ7naDny0fq4p77nDIRACIRAeFBC4KwheGBekB2UbsoGhVscrN6M5Gk3kyi7/nW9Tm816qyWgTZQ+CxrsndkDEMMZr0NGVSJoZTUG7hZ67BcvSFLn4HUgLYPz83MN1vw27wBuONWJByBzdFnGzGImgRxjAM1eQ+4tttLEqWMTknUTnAopVbSWSoNnZJJ6ZSCtVTRKbmaTomZPImYTmmYlJyxVEsoYCoapYJOqBR0IamCytIwWWIGlZRJyZiWKphySsZSyfSCXM7kMSkF0xoZ2+lMSs74lLJ5dC+wThR0loZBqWKQqJn2qCRMHmupolyyA0DnaOiEku5yhYxO0GYS2oydWVIxlcV1Bk5I6ZGWK2lwtG857VvOtEFDQ6DrYWYK218yulYmwY23I8FOXqaT5cxIqZnmKplyKmbw2dnMLkcpC5JdgExKxowP3Wp27cuZ3qE7XsF0Fd1qBW0mY6Yg00oxg1bM4JPQ9bINEdNZbkTirCwYV0Y3iiEHdimz05vxp6HBs4ZSJVUVM/OZxaNgqYHpdSnlgZ5qErozVfSUltF9LqEXkpQZJ7oaliwYOyWd0LD9z9IWw5k0CBmNVcnwmZieBsx4qplmqKluoDuV7ns17UtJ58jpIaddMTTDdCLdOwxDSpkEswaZqa9mVj8zHelaVYwHGTMvJOyUZtY0w4NSGrqKnjJyKs50NV1CwlwcGCKRsBOVvXAwM4MeYzUDjiFSxoOMbTxzBaATShqtlBlmplpmTTA1sbOYaaKSbSIDiRlsGXv9YwzVTErM5jHVyZhFo2KWPts7TKtVzHxmyYnxL6Znl4Zd1+w1guVwZq6x1yQJs2SlTEqDP7CNAXg3W6iNThGLdQlIxEiSkpJcakmhWiLW9ZBrGTVSiaVF1UglVhRVIxXqYVYt5XzHi1k18e2uJr7d1cQ3o5ZRuNswag43q+Zws2oON6WWU7hZNYebVXO4WTWHm1IrKNzxjJrDzao53Kyaw02plRRuVs3hZtUcblbN4abUKgp3AqPmcLNqDjer5nBTajWFm1VzuFk1h5tVc7gptYbC3ZZRc7hZNYebVXO4XWqJmMLNqjncrJrDzao53Ph7EfdjXwQ6gJIjpCgIremQF5xU0i4vJ8PkKSGC7hF+2wc0UbSLuAggf+u3zcvSW/Mtdr0uKyNd1yHHYLTa9UadzZhuslVSmyKN4Fd4nhs+VSI4ap+Mi0YqBE9ls7k/wBeZt1EdDJn2zHyLLddiNen9qCxmfnGRat3sVn22MceSazLry1HnHRJmQVcUVO/gezMnhEZRabunonQ+LczKBLzyzGOfS1am5twvOn8zt1AXubJgAG5/KwOm+3W8hTtkRGjNA4wWs96u51Mw5SK1NoMlO99g4bKLaoo21ZtSD7LGp46KaqO3mdJNXii9mYXUCszNzC9xlVYKG2/aVTTyiGMeC1Sz/2AK3kkPjsNKMgcbMWc3fheV3c6naRuezHKPr2kfXPteealRlm55lrUvqnXOM1isZfNR2dM+rbGKBit0uhxn8P2eLt1FVvBLT0N6k15eZvdZcyqzWWSyIdNqQdmujXkRjTfzzRs/RTUed/eCHz4iEkw5TLVFFEULV2+bZ7XYHciMdqOtWG3R3vCltLc+uUh0N4s1XW8t4ifZmmYyu2tjEo2WDLPFZnPTN3C1WGcw2zxPjrI3Q8COiE6w2DNN6Ua3fB51dW0G6iKDV1of+ZS3uyqbTZmOkz549R8WwXui8jssB88+RdwOVbzV0F/sGfXFKUuHIyLZbdWWrChjyx9NLpEgfbEVxsRPXVWje3aa0WzJRVdjvk4shddKq/Bp3VQW0H5RlGb2PBg945/erdjLX4VUyp5EeO0rsxnXlgauIwy/tVdG3gxFVHymNd+GPRmMeiP/JZo5RfFX6dsO2jc1B6YqOVtRiD2o+JzHZ+qNdpSVj/Ya5pLUHo5GPpX32q2rce3zcy0GtJVxWUcWHrsoTTXnkYvS0cctdHrjO4vn5GcYcxjnJWsas8cmfvdeG0YWGnrqkqovGXPseUx3VUsxGx0bPEoXkWjJzs2kYSca9Ta71bteLpaTeFGXp02ZTlY+ePUflvLsiXKoy38dxXcQ4q2mEXPmKbtd6dBW62TMtVuyTXZ6pXqpE+zcVHKbvLAVxsQb4NW76XOtFrMxI4fmN5+0rmOUp/4sy6BUlK56itFqyXEneF+0Pq3NiJR8tDGgLyUlKspStnwUHg5ZFdwqQb3XTy55TyFEPb5MOuFOXuW3URC8UvZ0w2tfmc08nrx4rb0y8upiz5y8fPHldhjjLelp21wWnesqlmK1WC0GtE9nV4knrSdHNdsbrbkWu9VkyDQZzWZ9sUdP18krFRewZBvR3shTiVi3s5aunz7b5+GJjjfrB5gMplx0haLPl+QIRqka0mcw/rroYxm/kevIVR5G7se0shtW19pyjdYME89gPkad6XwZlqrkTOfdBCmW7nhrLU+bMh3qfPDqPyz3UMTtNMXbEW6npDLblQ5tQ3pjzO9cWDPBjmEl1+CFrTAm3gCPpo5W9EeGvqmpk5gngqqQEX1wzBqzm/LyNPSJXiMLzxjUDPFGU8bi5akpurwqSQMFr0Jwh9HUjCtZLdyBs/w2MYJXyh7q+EmkEpt5PHAWs6P3wqisWyX2RMlbTXQbfXZans1ksDD7fB517c56g9WSbso1o/+KXCcb0gdDL1oWTR0vmbXiUV183U96OIDq0LHJ587FPzA0mk32N0z6nFLdfFQNLXVThlmfk57n+tVigt5gzzcjYCbOKXva5IVKnzb5jagzog+Ndj8M8hb12rBqijEHnbVRO/Oy3Wtjjow+oAwvPDIWdVhm0uQF4Y1NJTkTBSJ+jbiOJ36fCFXbWI0DTIWHEl6jSO6IUnRbqs21IE7MNOfry6TxBmvRUu5bbR+6y6fVVi0FXTctuXjfxTGCtzpPZF16b+Wh83DQTTRb0kyObs7JcBnyqLliYSmM38hOetTVaINQOlYtWtwbTSnb4Jua9GKmPRNlFKMTatWW56WC3Y2WpjpfPdBbVb83z23PyltftU7GPIMebzQsrh2WB12dFJsxL91icHyJYMtFfxBDGbOd2WivYkUzAu9pPLAMtwelZlZR3dOeN6Vl66uH6S2oTt8PbWpLuRf1JhJF7R1tuWVyVZ3e4RXjq6pzR1fWCgORBzxSldueCT6TalPbMV22KSfPjo50Pl0+olybJBe8hygl8mri6Ig3w6MboVpZzXX58sZ3NL1H8apETabEAOPr+hxfuLCY4mX0zEUiU9ybU9O1ddGlGw2m9LzSbZOiKD+cc++VPtX1CL3H0WWiY7q1cDPO9hbJN1gMlsp/ko1mditezc8SSkRR+xnB11IMu3/h/D9M7010uZlWvdF1PS82M5ZnB6PL1lvtphz8eVIxS5XarrjA8Kjr8mxkkIHjMcdBdwH7/p2wQn3IXbc36eD38KBzYHZ+quttOqG4MG3k4fU5Ydgm8LqcwOtyAq/LCbwuJ/C6HFd/Bl6XE3hdTuB1OYHX5VCU8MC/LifMqU2wZOtNOc4UOuKYuZNaSLzjPY8kkWxNN1qddu302SZzvjMr0ZiT57QL65aLzntG/OrGIq91tOnRaRA/CVYUhPep1QH76p7FLXItFnMLdIyzZJsMthY2u9WkM9ns2ZY0dCQwtsBFW3Tr0bZTatcWXbU9dAnaVG2L5OQWUrFE0yJFLG6egESMJCkpSddVEtdPb7P3j8t441lB/Uo5v8LAlhS6l4ib95BrBYTt5tcPsJFrsdQPsGm/foCNXIsVfoBN+/UDbOQaMZXwsGm/QsOWcnMwXiwkbDe/foBN5qDgsGm/foBN5qDgsGm/foBN5qDgsGm/QsOWUfzaRkDYbn79AJvjV4Fh0379AJvjV4Fh0379AJvjV4Fh036Fhi2n+FVI2G5+/QCb41eBYdN+/QCb41eBYdN+/QCb41eBYdN+hYatoPg1XkDYbn79AJvjV4Fh0379AJvjV4Fh0379AJvjV4Fh036Fhq2k+FVI2G5+/QCb41eBYdN+/QCb41eBYdN+/QCb41eBYdN+hYatovg1QUDYbn79AJvjV4Fh0379AJvjV4Fh0379AJvjV4Fh036Fhq2m+FVI2G5+/QCb41eBYdN+/QCb41eBYdN+/QCb41eBYdN+hYatofi1rYCw3fz6ATbHrwLDpv36ATbHrwLDpv36ATbHrwLDpv0KDFsipvhVQNjufv0Am+NXgWHTfv0Am+NXgWHTfv0Am+NXgWHTfinY7NeqwRaLThKI3+dxdsyDUribGlBU4kNU6kNU5kNU7kNU4UNU6UNU5UNU7UNU431UIi45yg5leJIxQ283DXD+Tj+4h96a7UM8LN6SY7dazKVJBuq6V+pi54woyRmRexERexEJ+OGJsB0fKhbrxK77yXDKde+7BKVctyBKUcp1+5pCrJMEyj2w5fiWr1agSBuBIvECRRIEirQta4Tt+OgOqd0kFktzt0uuU124F28u8ayWelbLPKvlntUKz2qlZ7XKs1rtWe3chcS4NUfMo5fw6KU8ehmPXs6jV/DolTx6FY9ezaPnaa+Up71SHvxSHvxSHvxSHvxSHvxSHvxSHvwyHvwynvGS8YyXjKe9Mp72ynjaK+Npr4ynvTKe9sp42ivnaa+cp71y5z3O3EW5m1YlVnalU5JyTknLOSUr55S8nFOK8ki5ncMM9gxDhl2fwaUz9Oif3V5R6Qy9QY/gVFhab8gw2PWGCkzbDUgyKiptx6OBQJV7ugi7KcTqdnRKwqSkTErGpORMSsGklExK1S5Q331Vnxu7JWrjtfHxianOtDYVJbTxiVw6PhWpUrl0qjYxPlHrskdZqIArnRqPPWg5e5yJzLl0YqLW4TBQf6B+pn52XtZAdlpsi9whOMhGG8gJ5ARyAjlcDssYtRILqQhxC+EglAzkBfICeYE8v+Wxv3QPydFnO34Cj4mpOlFGOO7PaGcyOz9Wjupm7J9nzDGYcjK65jl/gh+cYrY4j7gRqVaj3p5tzHEqglON2bnOOhKMufZMkghL0ZusyA9JxiTr2uHKdLg2m66D8/Pyqh1Uug456cbXXaoQE05z+Qr3/FBHvvMjxKh2FutAvTU9xWrK5p4GENXVOMBotRmJku2KCKtlYBzdHdxn7SlisS6h8PsyXVJSkkstKVRLxLoeci2jLvwmpYi68CuVIurC71ZYtZTzHS9m1cS3u5r4dlcT34xaRuFuw6g53Kyaw82qOdyUWk7hZtUcblbN4WbVHG5KraBwxzNqDjer5nCzag43pVZSuFk1h5tVc7hZNYebUqso3AmMmsPNqjncrJrDTanVFG5WzeFm1RxuVs3hptQaCndbRs3hZtUcblbN4XapJWIKN6vmcLNqDjer5nC3dVvP4XHdZIXPBWEfSBeJ71aL64efHQg8P9euqk2fnWs2Fj4BxKun2hV9ggjK19mtRlSF4wkiIaiKKCT/oPgWJFuJbCOyHckOJDuJ4PguIruJYJuviewhgn3tJfINEazbh2Q/km+JfEfkAJKDRHD8EBFsexjJESI4fpQI9nUMyfdIjhPB8RNITiL5gQiOn0JyGskZJGeJ/EjkJyLniGCb80RwmZ+J/EIE+7qA5CKSS0Rw/Fckl4ng+G9ErhC5iuS/SH5Hcg3JdSJ/EMG6P4lgm7+I3CByE8n/iNwkvv4mgsfvXyK3kNwmQuaAQ4dt7haKCBQKxiQSFQpukwjNFVEQkmAkIURwPBRJGJJwIhFEsC6SSFUi2LYakSgi2Gd1JNFIYog8RKQGEjQXRbWIPEykNpGaxKYOkkeI4HhdJPWI4PijSOojaUAExx9D0hBJIySNkcQieZwIjjch0pjYPEHkSSK47FNEsK+mSJ4m0pTU2YzIM0ieJdKcCNbFIWlBRExEgkRKBMdlRLCtnIiCCO4zJRIVETUSDZGWSJ5D8jySVkhaE4FIXiCiJdIG4PsHCqUNsWlFyiYAx3f5DkkgPtsRwXUkImlPBMc7EFETbC8i6YikE5HOAN+DUSjJRFKI4LwuRLoS6UYklUhH4rM7kpeI9CDSkwjOexnJK0RwHM8xPId7EemN5FUifYjokPQlguN6ImlEDEjSieC4kUg/IthnBpJMIjhuQvIakiwiZiTZRHKIWIjgvFwiuAziapGVCI7biNiJYN95RAYQGVgoDq5DHCh6HUk+ERx/g8ibSAYReYvI20RQnoPr8drHXI+vEzTfozWLrhqFXI94wMH1aG3jx646uB7xgIPrHRcvUMj1aP07uB6tfcejWmm+x0+r+g4Jms8c3yMeAIgDHFyP1rjj6atOvkdcABAPOLge8QCoA1i+RzwA6gKW7xEXALQmi/A94gIH1yNecHA9WssOrkfrG6B17+B6xAEAcYKD6xE/OLgecQN4ArB8j3gBID7g+B7xANqTF3I94gEH1z8DPPM94gYH1yNuAGitO7ge8QJAnODgenwbMuZ6/A28k+8RLwDEBwBxgYPr0Xrj+B7xAcf3aC06uB7xAkBrFqA17OD75wuHGKC1DjA34ARa++AFUITvAf4sCHMDxfcA8wUeY8QLAPMDHmvMC3isERcAzAl4vDEf4DFH6xZ0BAzfA8wJeJzx2qf4HmB+wGOM+QGPcQoSxAsAcwLhe4B4AWBOIHwP0LoHmA8ovgeIFwDmBB6+B4gbAOYIPJ6IFwBawwBzAh5bzAl4TBEXgL7AwfUA8QHAXED4HiAeAIgPAOYBPMaYBwjfA7z+8XgiLgB4rRK+B3ht43HFfED4HuD7TDEn4DFFnAAwH+CxzQUOrgd434Z5gPA9wDxA+B5gLvDA9wDzAuF7gLkBjyHmB8L3APEBQLwA8M8a8JrHXED4HiA+AIgLwDvAwfVgMJJ3Acf3YAiS94CD68FQJMOQDAcM34MRgON78D5wcD34ADi4HoxE8h/g4HowCji4HowGDq4HHwIH14MxSD5CUgA4vgdjgYPrwTjg4HowHsnHgON78AmST5FMQDIROLgeTAIc34PPgIPrwWTA8T34HMkXwMH1YApwcD2YCji+B9OQTAcOrgczgIPrwUwkswDH92A2kjmA4XswF3B8D74EDq4H84CD68F8JAuAg+vBQiSLAMf3YDGSJcDB9WApcHA9WIZkOXBwPVgBHFwPVoJCrl+FZDWSNYDje7AWyVfAwfVgHeD4HqwHDq4HG4CD68FGJJuA47HZ7o/Sdthh7sfcgdfsW3D6ozcShjZ/Dm43LrnQf3JPuLBe7Rm3t4XCjVcGzGv71AtwyZ6QeSE5N+FUycn5tvo14dGItlUumfrBoYldv7fOugInDn/5S6u0O1yy98nLozMtcJ8l58qf6VvgyhHJ69a12gTXWYe9ox1zGM4cPcd6o/ZuuDmjQ4b9ja/h6A2r0oLPZsNp15opck79CgfrR29tOmoSHP513LbddxLhpkTt+Ctx5+Hh+bMWhG0ZBLe1GPrb3Mc18ODswUsX350Nd/U4PuOQ/nc4dVSriY92rg0/7fx9z5lfZcLNx54HKY22wz1pg99/c8wjcA28cPV0wSW44HLVr4e2fBnOXr+n0YuxBrg2fHfY6JHpcLFU+63ilw5wy8qNL3fdsB9uLrgxRtHnLhwSIetyul8e/KTX6lmjQ+fCcd3y/thXew+cZ3rh0qllL8HNt2CVrNXvwu3nDs0c0v0knJZR9ZWha7PggcvzXh0Vq4Y/Xp392ZDH8uCmc82jd6XVhTvWj9NU6V4PHr5sXhS6ZSL8Nkf1xZK0DPjdjaymY9Lmw48nHbDb1i+HO5qlrjrS80347dBHktaP2gNPTTlgfuTYRXg0pv7aYT0Hw23Lt04eP6Eh/Or5Atu5k9Hw4P6Uh0asUMId8xfcPZn3ETw1cGNWq+0auN0U2tl4SAq3VtlfYAseCneundSmeU4uXJ3yz+7pusFwUfzXo4KUU+DcAfo9Pwf/BhfEGh6t//Q1uL3u9BiT6hU48uWk47qRp+E7bVNCah39BG5o+sul9wp2wJ9nPn/pwJsT4eo3m/x268Z4ePL7UW/Ws1+F+/dHPlzj+im4MuynVzeelsFVLVbtmyC/BfcqZb/ZW06AZyceCvu2hxaeTqn2zAD4HJzXde+AHlfscEXfXxfXHP43PBY3RHe7y4/wRNOsK9rxd+DaznOr5my2w+8+/Vd31dAbflJTVP3zgx/C7+4UnNO0UcNp9W7ExS5ZA2c9krzwodxPYNo0pSj1n1/h0g+PGAadmAXnTu2z+Ze9a+D6g/NPTA3pDmetGPLi+GtV4Zo34ut9ZY+Cm2+LmyWe2Ai3tg5pqdrxBly45MzwtJN14FLxnLekwWfh9kNDzQrUjoM/qm4rW16CXz33XtQvf7wGv9rQqeDLyBQ478/5Wz698i9cX2/ilqdjQ+BnTVp8puu6DA40Ne/Ss8FIuH7oe/Oes2+DC8+ceOlMuzC4e4LZ/PbncXDxqCn9ezWIh8tWXv32pVoRcN3gZo2+XDYMLogeeSB+wl64uFeT5CcutYD7Og3+44V1deDuZTMntz37OPxyequ2f/XPg/Z+d35v/dgceHh5ZMLoFTvg59NSTJPbt4Ere1zdu+s/Y+Ghw9M/ST+khAWHD/e/8e4LrRX5H/4kHtUSbv201esNjS259OzeNZpLtt+GH8cNyJw0ewNcEjEoK3LMLbhlzNt38+Ysh0fPjEsf8UY8/O7DW/YuC3+FR/vM6LL0WCg8kDT+x5dmSeCM91ol/g6PwC2rmu29fSEezgu2xny7VQIX6TaCBbGvwpHH+ze8/U0HOAh0f+cpbRf48ZlP94U3NcJtRt3oJ75qB9f2XnStT5YOTp4/KbbZdBfO6Ydu1BpouAJPnFJdnlbvK/hZQcpf6473hUdrhAQbhzaH31wcq9Q1OA4XT+nU5Ozrl+HOvi9GR8uuw9y7817XrlkHf7iVMFZ2sD1csmbRvyse+wBuXVTt+/lvJcJZPyf22vfXSTghX3zjhn0SXNZx8jJN+DW4Kf7ik+HmJnBNTLUOww/X4HAUbB7f4aFhc+GnwydbXsufCaf2GfRFYvo78NDYO7/VurkDTpcFHVT2/h/cvT2p7TvL9fCLm8Pie4/oAkcfun14w7Zn4Oc3L1zv9VsqXDL4l2Dw2gq443rC9xPfrQH3iLOCbvXqCue0nLRH9/cRxJOxDzeq9xjcMmV8I0v9d+C6pr9/H2u7BTc0+zd2639vwqVL3z8mlUfBJY9r+xxVZMBtb3SeZwo5Dw++3Xd8ysfz4YgDp209nh4AJ3c7NPD6L3/DFde6Rw6KyoNfmf/uWrAbwgULl2pOvDUAThrZ/GDoe2vh+8ufnXH6qSSunUdbXXwxv44RfvPBpJ27T/SEK08bP7rz2jvw4JRvOhztEAEPa1rvOp34G9w3959e+v/sgsteW9424UcNXND58Oh/686HO29nLVr4yhS4evD66h2O/AF79Y1MahFshcawxOadkqfCS880eL5918Zw9v92gOONB8INVTquWNG1OVx8dcHYC9ei4NLv361aJeNDuCrvSqRq4h24rnGrDcnH4jh8U+o1v5Lxy3L4yRePdu4qrQ3nnMq/22v2YFhg/3xTxzU5cGL/xz4OSa4O+3fqn9Lj/Ddw7DPR4Z9+1RMODe3xS37+Ljio2Ws/Tbshhx+pUw+OmrYUDh8mybx0Zhwc38Z83DRyPhy5La1ar387woUL96/JuDoULhjav0rL4XPhkocP/W/AK1U5HJ+9l/NZE0tvOD9/uObWqNZwYs8WUe/W0sFVM0RxL64ZAr/8OPb9c31fhktNi+ZALYDb7/zwdNXnH4Nb9bZ/fr/6EzxQd8K5mF51OH9m0QuzXmv5sysNopvNHnLdmW6t2fHQDy21Z7B+Re2f5K504zELr06zcOXevLq5Hsy/BrPO5O+b1ieC02fdbTL60MWpcNjK1ttbF5g4fV4zY62xv8fDSUunj1sYmeryM/bJTh+sbAmz5WHSzEa/wo9+nVtQa8DDMKtf+BPtaraCWR0LLo6pfhnOm23tc/J6e7j771d3HgwywlfzNXsajH0Fdv7iZL9l3c1w2c1VB5eeHcb5fevw/CGtY2vCn2v92fg/c5rApfGnPwhuMwhOUtrg/tNiOH9Y8Kzfj6+CW04PnbToignOy9/fp+eounDe9Rs7z9YYD9fmbnpox+SZcONfr6t7P94U2vakbd1rXw8LFlWZFn5xAByw8OOFQ43LYcGLV0K+2LUffrK3y/SRQU3gcqkiOW/9DLiu1dsDzk+JgqOOnZl8ZNgs+NngpbbTg+a7+n3c5k7XTz8MzRfvvF3n6ZGc/mBGQUfDvtvwk9unj3Vvth1+c3bOZa3oKS4/Z+awrr1broIjpoTV3jN5F7R0q7FsZ6cCbvycdpv+bdH/6uZ/4M4R2zblPLECLnt+2LioKaPglPfO67+LPgOPXQLH0vYkwp9mjRGvuDMLLov88uEn+38Bz/ddNfdyZgGcXWuVvP3edXBp+K6V1w8cgZfksTvbvPoT/C7LXn9ZZnt4eENE2/yQf+GqHySr102Tw+Mtngoat682XHpxqrhWZxtcHnLgiU61jsLl51v8kZW8Bp4ab92oHtkX7rna9+Tqm03hjpRWQ/66tgFuGtz27AdXgXOecfizG2cvSa3xBXyrXT116N8XOP3w8NDYz+eOctk9FdNm79lH4cE/w273jCqARw8b1J0zn4VbOt45q6s1By5fezD2o4s6uHf3m29densG/LLa2bU1Ts6CG957tY6040K4c8+0WjbjWjh39NhLl0++CNeH/29olSdqw49Df7M2yB4H59/9c2pQgz1cfaN7HJmwodZo+M7pUPNzf4+Fk96rtWpl7BNwxsxdeQ1jLsCtH4Wt6591Hm7tbBvZvf4xuG08HHGk40145MA75w5v+gce6bDItrJBItwyp8/bLScMgQeytj3ZYfFH8HBWhOGlmEXw69uS+ImnVsOJX/+3e5b9JrS/sqlmyqZVcEer5+pNtKB9yNNjbj92tgucod2cfPPGs3DH1o2Hd9R/Fe7sUTPp9IIt0NL6D3Pwo6lw8Kfy4ReWzYQnrNvSutR6Bn5zocHsuKeGwwMD01ufmvUk157B597stq+RHc5cMLtu2ivt4Pb+O3reOTcGLl61bHMbfSRccPzHljfeng/3Ho8YZw8Pg+erSp6epl0Njw16ddsjI76DRybV028/ccS1/p+6+OeE6M5w45afzZZFc8kZQETOBzGOv4XfEwDnl5viOKnrueziOKXrOfXiOLXrVRDIzPWqE0mc6xd82ExGldFwPxHGCQ2TI6ETGirheoMAqkdBwZGraTMVU4ZCoHK9rQdXKqUTVI5MQaNWUdiUMsq1mu0DqoxGTWFzvZGmSB/IaGwaqoxKyniju0qspitV0s2mc5QKnk5UKShvYroPFLSZWkWZqSV0H4iphOt1OI720E2QUQ5Uaj5sUtcLFNAwSulK5XQZBTNd6GGUymnXEhoo32grpXRLpVKe7pWL5XS/0S3VUHBcL2xiZ7zb9FepqN5Ry2R0E+jBoruKmcoqBV2GGXol3dcSuqvUYrpD6ISUHiwN7VpJmYnlVO+o1PSMp+ebREOvOTGNjV4LGrqvlTI+plBJpHQ99KSgsanVcroeepnR00WuYbwxZRQ8ZRQMCbFrm2qpguk3eiqr1ZQDlYqph55vcroeJR8jqekO0TBrWyahK6W7SsYsDHpSKJj5RpuJGVKlzFQKesnIpB47xG0YXa8PckwxehToOcqUkdPzTU73jpjua6Y9KgW9ZOiOV9OLVuVqqRtQDcPK9IVFRRMKM+OZ6aLkXipS5KJHLwwZQ+ssizGoqX5TspOCHgWJknbNXDU9jw8eYCpHxmBj1gLd1yqlhge1Sk73GzPF5Ewn0mVUdB9Imes2zRT0YtLQHSJjlqaEuZYwq5FZWczSpBtH9wE1jO4OGGwMh9DTRcFsPRhOVPPNEBVz0ZNQlcqZHqUvOWo1/pIgBgReTagNvJrQlSdh8gKvJtQGXk0YeDVh4NWEIPBqwsCrCQOvJqTz7vdXE+Iv9yOdNwWHOGecMT3D+ZOJqjgeZzbmZHA/fQhJyrGkO/MjcTzOrE8zOp+QFmE35RIFd1syc89xiOOeY/cfcFgcbztkLSMMZn26caDJhm+xcL81OdRq7Fd44zKnESV179RJxHM3NHWrs7e3QoM7/wck91MQNV8BAA==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'phyloseq_oo.Rdata'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
        });"><button class="btn btn-link downloadthis  " id="dnldts33142"><i class="bi bi-code-slash"></i> LotuS3 phyloseq data </button></a></p>
</div>
</div>
<p></p>
</section><section id="citable-resources" class="level2"><h2 class="anchored" data-anchor-id="citable-resources">Citable resources</h2>
<section id="packages-used-in-this-workflow" class="level4"><h4 class="anchored" data-anchor-id="packages-used-in-this-workflow">Packages used in this workflow:</h4>
<ol type="1">
<li>R microeco package <span class="citation" data-cites="liu2021microeco">(<a href="#ref-liu2021microeco" role="doc-biblioref">Liu et al. 2021</a>)</span>.</li>
<li>phyloseq R package <span class="citation" data-cites="mcmurdie2013phyloseq">(<a href="#ref-mcmurdie2013phyloseq" role="doc-biblioref">McMurdie and Holmes 2013</a>)</span>.</li>
<li>miaverse <span class="citation" data-cites="felix2024mia">(<a href="#ref-felix2024mia" role="doc-biblioref">Borman et al. 2024</a>)</span>
</li>
</ol>
<p></p>
</section></section><section id="references" class="level2"><h2 class="anchored" data-anchor-id="references">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-felix2024mia" class="csl-entry" role="listitem">
Borman, Tuomas, Felix G. M. Ernst, Sudarshan A. Shetty, and Leo Lahti. 2024. <em>Mia: Microbiome Analysis</em>. <a href="https://doi.org/10.18129/B9.bioc.mia">https://doi.org/10.18129/B9.bioc.mia</a>.
</div>
<div id="ref-liu2021microeco" class="csl-entry" role="listitem">
Liu, Chi, Yaoming Cui, Xiangzhen Li, and Minjie Yao. 2021. <span>“Microeco: An r Package for Data Mining in Microbial Community Ecology.”</span> <em>FEMS Microbiology Ecology</em> 97 (2): fiaa255. <a href="https://doi.org/10.1038/s41596-025-01239-4">https://doi.org/10.1038/s41596-025-01239-4</a>.
</div>
<div id="ref-mcmurdie2013phyloseq" class="csl-entry" role="listitem">
McMurdie, Paul J, and Susan Holmes. 2013. <span>“Phyloseq: An r Package for Reproducible Interactive Analysis and Graphics of Microbiome Census Data.”</span> <em>PLoS One</em> 8 (4): e61217. <a href="https://doi.org/10.1371/journal.pone.0061217">https://doi.org/10.1371/journal.pone.0061217</a>.
</div>
</div>
</section></section><section id="workflow" class="level1"><h1>Workflow</h1>
<section id="required-packages" class="level2"><h2 class="anchored" data-anchor-id="required-packages">Required Packages</h2>
<div class="cell">
<details class="code-fold"><summary>Click here for required R packages.</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">919191</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ChiLiubio/microeco">microeco</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia">mia</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://microbiome.github.io/microbiome">microbiome</a></span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="review-results" class="level2"><h2 class="anchored" data-anchor-id="review-results">Review Results</h2>
<p>The LotuS3 pipeline produces numerous output files but for our purposes there are four specific files we are interested in:</p>
<ol type="1">
<li><code>phyloseq.Rdata</code></li>
<li><code>OTU.fna</code></li>
<li><code>OTU.txt</code></li>
<li><code>hiera_BLAST.txt</code></li>
</ol>
<p>We will mainly work from the <code>phyloseq.Rdata</code> (renamed <code>phyloseq_oo.Rdata</code>) since this contains the OTU Table, Sample Data, Taxonomy Table, and Phylogenetic Tree. Note: a phylogenetic tree of microbial ASVs from short reads seems less than useful. Therefore, we will exclude this from the initial analysis.</p>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"working_files/LOTUS3/oo/phyloseq_oo.Rdata"</span><span class="op">)</span></span>
<span><span class="va">physeq</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 12091 taxa and 41 samples ]
sample_data() Sample Data:       [ 41 samples by 14 sample variables ]
tax_table()   Taxonomy Table:    [ 12091 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 12091 tips and 12089 internal nodes ]</code></pre>
<table class="caption-top table">
<colgroup>
<col style="width: 50%">
<col style="width: 50%">
</colgroup>
<thead><tr class="header">
<th>Metric</th>
<th>Results</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Min. number of reads</td>
<td>0</td>
</tr>
<tr class="even">
<td>Max. number of reads</td>
<td>89813</td>
</tr>
<tr class="odd">
<td>Total number of reads</td>
<td>336493</td>
</tr>
<tr class="even">
<td>Average number of reads</td>
<td>8207</td>
</tr>
<tr class="odd">
<td>Median number of reads</td>
<td>2333</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>192</td>
</tr>
<tr class="odd">
<td>Number of singleton ASVs</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Any ASVs sum to 1 or less?</td>
<td>FALSE</td>
</tr>
<tr class="odd">
<td>Percent of ASVs that are singletons</td>
<td>NA</td>
</tr>
<tr class="even">
<td>Sparsity</td>
<td>0.946</td>
</tr>
</tbody>
</table>
<p>In this next part of the workflow the goal is to create a <em>microtable object</em> using the R package <a href="https://chiliubio.github.io/microeco_tutorial/">microeco</a> <span class="citation" data-cites="liu2021microeco">(<a href="#ref-liu2021microeco" role="doc-biblioref">Liu et al. 2021</a>)</span>. The microtable will be used to store the ASV by sample data as well the taxonomic, fasta, and sample data in a single object. More on that in a moment.</p>
<p>We will also:</p>
<ul>
<li>Remove any ASVs without kingdom level classification.<br>
</li>
<li>Revome any contaminants (chloroplast, mitochondria, etc.).<br>
</li>
<li>Remove Negative Control (NC) samples.<br>
</li>
<li>Remove any low-count samples.</li>
</ul></section><section id="sample-data" class="level2"><h2 class="anchored" data-anchor-id="sample-data">Sample Data</h2>
<p>Before we begin, let’s create a summary table containing some basic sample metadata and the read count data. We want to inspect how total reads changed through the workflow. Table headers are as follows:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 21%">
<col style="width: 78%">
</colgroup>
<thead><tr class="header">
<th>Header</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>SampleID</code></td>
<td>New sample ID based on plot, depth, treatment, &amp; pair</td>
</tr>
<tr class="even">
<td><code>Plot</code></td>
<td>Experimental plot ID</td>
</tr>
<tr class="odd">
<td><code>Treatment</code></td>
<td>Control vs warming</td>
</tr>
<tr class="even">
<td><code>Temp</code></td>
<td>Warming treatment temperature</td>
</tr>
<tr class="odd">
<td><code>Depth</code></td>
<td>Sampling depth in soil</td>
</tr>
<tr class="even">
<td><code>Pairing</code></td>
<td>Sample pairings</td>
</tr>
<tr class="odd">
<td><code>raw_rc</code></td>
<td>No.&nbsp;of raw reads</td>
</tr>
<tr class="even">
<td><code>cutadapt_rc</code></td>
<td>Read count after cutdapt</td>
</tr>
<tr class="odd">
<td><code>lotus3_rc</code></td>
<td>Read count after LotuS3 pipeline</td>
</tr>
<tr class="even">
<td><code>lotus3_asv</code></td>
<td>No.&nbsp;of ASVs</td>
</tr>
</tbody>
</table>
<p><br></p>
<p>Now I do some wrangling of the sample data to format it for downstream analysis. WAhat I want to do is first add read counts from different point in the workflow.</p>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_otu</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_otu_sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">tmp_otu</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_otu_sum</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"lotus3_asv"</span></span>
<span><span class="va">tmp_read_sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">colSums</a></span><span class="op">(</span><span class="va">tmp_otu</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_read_sum</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"lotus3_rc"</span></span>
<span><span class="va">tmp_track_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span></span>
<span>    <span class="st">"files/CUTADAPT/oo_cutadapt_track.txt"</span>,</span>
<span>    header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_track_reads</span><span class="op">$</span><span class="va">SampleID</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span></span>
<span>  <span class="va">tmp_track_reads</span><span class="op">$</span><span class="va">SampleID</span>, <span class="st">"-"</span>, <span class="st">"_"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_track_reads</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"raw_rc"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_track_reads</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"cutadapt_rc"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then I want to merge this data with the sample metadata.</p>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_samp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_samp</span>, <span class="va">tmp_track_reads</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_read_sum</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_otu_sum</span>, by <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span> </span>
<span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">5</span>, <span class="fl">6</span>, <span class="fl">8</span>, <span class="fl">16</span>, <span class="fl">17</span>, <span class="fl">18</span>, </span>
<span>                <span class="fl">19</span>, <span class="fl">9</span>, <span class="fl">14</span>, <span class="fl">15</span>, <span class="fl">3</span>, <span class="fl">10</span>, <span class="fl">11</span>, <span class="fl">12</span>, <span class="fl">13</span>, <span class="fl">2</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_delim</a></span><span class="op">(</span><span class="va">sample_tab</span>, <span class="st">"files/CURATE/oo_sample_data_full.txt"</span>,</span>
<span>    delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sample_tab_trim</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span><span class="op">[</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">11</span><span class="op">:</span><span class="fl">19</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sample_tab_trim</span><span class="op">$</span><span class="va">per_reads_kept</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">sample_tab_trim</span><span class="op">$</span><span class="va">lotus3_rc</span><span class="op">/</span><span class="va">sample_tab_trim</span><span class="op">$</span><span class="va">raw_rc</span>, digits <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">sample_tab_trim</span> <span class="op">&lt;-</span> <span class="va">sample_tab_trim</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="va">per_reads_kept</span>, .before <span class="op">=</span> <span class="va">lotus3_asv</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p></p>
<div class="cell">
<div id="tbl-make-metadata-table-oo" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-make-metadata-table-oo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Sample metadata including read changes at start and end of workflow.
</figcaption><div aria-describedby="tbl-make-metadata-table-oo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-75d662642cd4c45aefa9" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-75d662642cd4c45aefa9">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"SampleID":["P00_D00_000_NNN","P01_D00_010_W4A","P01_D10_020_W4A","P01_D20_050_W4A","P01_D50_100_W4A","P02_D00_010_C0A","P02_D10_020_C0A","P02_D20_050_C0A","P02_D50_100_C0A","P03_D00_010_W4B","P03_D10_020_W4B","P03_D20_050_W4B","P03_D50_100_W4B","P04_D00_010_C0B","P04_D10_020_C0B","P04_D20_050_C0B","P04_D50_100_C0B","P05_D00_010_W4C","P05_D10_020_W4C","P05_D20_050_W4C","P05_D50_100_W4C","P06_D00_010_C0C","P06_D10_020_C0C","P06_D20_050_C0C","P06_D50_100_C0C","P07_D00_010_W4D","P07_D10_020_W4D","P07_D20_050_W4D","P07_D50_100_W4D","P08_D00_010_C0D","P08_D10_020_C0D","P08_D20_050_C0D","P08_D50_100_C0D","P09_D00_010_W4E","P09_D10_020_W4E","P09_D20_050_W4E","P09_D50_100_W4E","P10_D00_010_C0E","P10_D10_020_C0E","P10_D20_050_C0E","P10_D50_100_C0E"],"Plot":["P00","P01","P01","P01","P01","P02","P02","P02","P02","P03","P03","P03","P03","P04","P04","P04","P04","P05","P05","P05","P05","P06","P06","P06","P06","P07","P07","P07","P07","P08","P08","P08","P08","P09","P09","P09","P09","P10","P10","P10","P10"],"Depth":["00_000","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100"],"Treatment":["Negative","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control"],"Temp":["N","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0"],"Pairing":["N","A","A","A","A","A","A","A","A","B","B","B","B","B","B","B","B","C","C","C","C","C","C","C","C","D","D","D","D","D","D","D","D","E","E","E","E","E","E","E","E"],"raw_rc":[212,73424,48831,231,207,49248,24274,208,1599,178799,12756,4336,138,26457,4207,6471,2909,9961,2876,232,877,35277,31590,738,436,24302,54660,330,374,4080,7259,636,656,25665,16042,1017,171,56767,17317,985,617],"cutadapt_rc":[137,72970,48301,177,113,48540,23953,121,1528,175920,12582,4232,73,26156,4134,6366,2837,9769,2679,110,726,34658,30908,643,128,24061,53639,228,266,3819,7116,506,596,25413,15922,909,63,56414,17171,879,237],"lotus3_rc":[2,6979,7658,29,27,21540,11654,2,411,89813,9132,3581,5,13604,232,4264,2397,8625,5,0,3,25426,21447,3,2,11949,44015,9,65,3298,5998,209,231,7541,2333,1,1,21938,12063,0,1],"per_reads_kept":[0.009,0.095,0.157,0.126,0.13,0.437,0.48,0.01,0.257,0.502,0.716,0.826,0.036,0.514,0.055,0.659,0.824,0.866,0.002,0,0.003,0.721,0.679,0.004,0.005,0.492,0.805,0.027,0.174,0.808,0.826,0.329,0.352,0.294,0.145,0.001,0.006,0.386,0.697,0,0.002],"lotus3_asv":[2,36,16,2,2,28,10,2,6,45,9,7,4,36,5,6,4,7,3,0,2,20,11,3,2,9,17,4,4,11,6,4,4,28,11,1,1,44,13,0,1]},"columns":[{"id":"SampleID","name":"SampleID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P00_D00_000_NNN","P01_D00_010_W4A","P01_D10_020_W4A","P01_D20_050_W4A","P01_D50_100_W4A","P02_D00_010_C0A","P02_D10_020_C0A","P02_D20_050_C0A","P02_D50_100_C0A","P03_D00_010_W4B","P03_D10_020_W4B","P03_D20_050_W4B","P03_D50_100_W4B","P04_D00_010_C0B","P04_D10_020_C0B","P04_D20_050_C0B","P04_D50_100_C0B","P05_D00_010_W4C","P05_D10_020_W4C","P05_D20_050_W4C","P05_D50_100_W4C","P06_D00_010_C0C","P06_D10_020_C0C","P06_D20_050_C0C","P06_D50_100_C0C","P07_D00_010_W4D","P07_D10_020_W4D","P07_D20_050_W4D","P07_D50_100_W4D","P08_D00_010_C0D","P08_D10_020_C0D","P08_D20_050_C0D","P08_D50_100_C0D","P09_D00_010_W4E","P09_D10_020_W4E","P09_D20_050_W4E","P09_D50_100_W4E","P10_D00_010_C0E","P10_D10_020_C0E","P10_D20_050_C0E","P10_D50_100_C0E"],"header":"SampleID","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"id":"Plot","name":"Plot","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P00","P01","P01","P01","P01","P02","P02","P02","P02","P03","P03","P03","P03","P04","P04","P04","P04","P05","P05","P05","P05","P06","P06","P06","P06","P07","P07","P07","P07","P08","P08","P08","P08","P09","P09","P09","P09","P10","P10","P10","P10"],"header":"Plot","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"Depth","name":"Depth","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["00_000","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100","00_010","10_020","20_050","50_100"],"header":"Depth","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"Treatment","name":"Treatment","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["Negative","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control","Warm","Warm","Warm","Warm","Control","Control","Control","Control"],"header":"Treatment","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"Temp","name":"Temp","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["N","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0","4","4","4","4","0","0","0","0"],"header":"Temp","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"Pairing","name":"Pairing","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["N","A","A","A","A","A","A","A","A","B","B","B","B","B","B","B","B","C","C","C","C","C","C","C","C","D","D","D","D","D","D","D","D","E","E","E","E","E","E","E","E"],"header":"Pairing","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"raw_rc","name":"raw rc","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["212","73424","48831","231","207","49248","24274","208","1599","178799","12756","4336","138","26457","4207","6471","2909","9961","2876","232","877","35277","31590","738","436","24302","54660","330","374","4080","7259","636","656","25665","16042","1017","171","56767","17317","985","617"],"header":"raw rc","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"cutadapt_rc","name":"cutadapt rc","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["137","72970","48301","177","113","48540","23953","121","1528","175920","12582","4232","73","26156","4134","6366","2837","9769","2679","110","726","34658","30908","643","128","24061","53639","228","266","3819","7116","506","596","25413","15922","909","63","56414","17171","879","237"],"header":"cutadapt rc","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"lotus3_rc","name":"lotus3_rc","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["2","6979","7658","29","27","21540","11654","2","411","89813","9132","3581","5","13604","232","4264","2397","8625","5","0","3","25426","21447","3","2","11949","44015","9","65","3298","5998","209","231","7541","2333","1","1","21938","12063","0","1"],"header":"lotus3 rc","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"per_reads_kept","name":"per reads retain","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["0.009","0.095","0.157","0.126","0.13","0.437","0.48","0.01","0.257","0.502","0.716","0.826","0.036","0.514","0.055","0.659","0.824","0.866","0.002","0","0.003","0.721","0.679","0.004","0.005","0.492","0.805","0.027","0.174","0.808","0.826","0.329","0.352","0.294","0.145","0.001","0.006","0.386","0.697","0","0.002"],"header":"per reads retain","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"lotus3_asv","name":"lotus3_asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["2","36","16","2","2","28","10","2","6","45","9","7","4","36","5","6","4","7","3","0","2","20","11","3","2","9","17","4","4","11","6","4","4","28","11","1","1","44","13","0","1"],"header":"lotus3 asv","align":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,41],"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"b74ec7e92dfe3e00e5b1a9e00fc1462f"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</figure>
</div>
</div>
<p></p>
<p><a href="#shrimp_md" onclick="fetch('data:text/plain;base64,U2FtcGxlSUQJUGxvdAlEZXB0aAlUcmVhdG1lbnQJVGVtcAlQYWlyaW5nCXJhd19yYwljdXRhZGFwdF9yYwlsb3R1czNfcmMJcGVyX3JlYWRzX2tlcHQJbG90dXMzX2FzdgpQMDBfRDAwXzAwMF9OTk4JUDAwCTAwXzAwMAlOZWdhdGl2ZQlOCU4JMjEyCTEzNwkyCTAuMDA5CTIKUDAxX0QwMF8wMTBfVzRBCVAwMQkwMF8wMTAJV2FybQk0CUEJNzM0MjQJNzI5NzAJNjk3OQkwLjA5NQkzNgpQMDFfRDEwXzAyMF9XNEEJUDAxCTEwXzAyMAlXYXJtCTQJQQk0ODgzMQk0ODMwMQk3NjU4CTAuMTU3CTE2ClAwMV9EMjBfMDUwX1c0QQlQMDEJMjBfMDUwCVdhcm0JNAlBCTIzMQkxNzcJMjkJMC4xMjYJMgpQMDFfRDUwXzEwMF9XNEEJUDAxCTUwXzEwMAlXYXJtCTQJQQkyMDcJMTEzCTI3CTAuMTMJMgpQMDJfRDAwXzAxMF9DMEEJUDAyCTAwXzAxMAlDb250cm9sCTAJQQk0OTI0OAk0ODU0MAkyMTU0MAkwLjQzNwkyOApQMDJfRDEwXzAyMF9DMEEJUDAyCTEwXzAyMAlDb250cm9sCTAJQQkyNDI3NAkyMzk1MwkxMTY1NAkwLjQ4CTEwClAwMl9EMjBfMDUwX0MwQQlQMDIJMjBfMDUwCUNvbnRyb2wJMAlBCTIwOAkxMjEJMgkwLjAxCTIKUDAyX0Q1MF8xMDBfQzBBCVAwMgk1MF8xMDAJQ29udHJvbAkwCUEJMTU5OQkxNTI4CTQxMQkwLjI1Nwk2ClAwM19EMDBfMDEwX1c0QglQMDMJMDBfMDEwCVdhcm0JNAlCCTE3ODc5OQkxNzU5MjAJODk4MTMJMC41MDIJNDUKUDAzX0QxMF8wMjBfVzRCCVAwMwkxMF8wMjAJV2FybQk0CUIJMTI3NTYJMTI1ODIJOTEzMgkwLjcxNgk5ClAwM19EMjBfMDUwX1c0QglQMDMJMjBfMDUwCVdhcm0JNAlCCTQzMzYJNDIzMgkzNTgxCTAuODI2CTcKUDAzX0Q1MF8xMDBfVzRCCVAwMwk1MF8xMDAJV2FybQk0CUIJMTM4CTczCTUJMC4wMzYJNApQMDRfRDAwXzAxMF9DMEIJUDA0CTAwXzAxMAlDb250cm9sCTAJQgkyNjQ1NwkyNjE1NgkxMzYwNAkwLjUxNAkzNgpQMDRfRDEwXzAyMF9DMEIJUDA0CTEwXzAyMAlDb250cm9sCTAJQgk0MjA3CTQxMzQJMjMyCTAuMDU1CTUKUDA0X0QyMF8wNTBfQzBCCVAwNAkyMF8wNTAJQ29udHJvbAkwCUIJNjQ3MQk2MzY2CTQyNjQJMC42NTkJNgpQMDRfRDUwXzEwMF9DMEIJUDA0CTUwXzEwMAlDb250cm9sCTAJQgkyOTA5CTI4MzcJMjM5NwkwLjgyNAk0ClAwNV9EMDBfMDEwX1c0QwlQMDUJMDBfMDEwCVdhcm0JNAlDCTk5NjEJOTc2OQk4NjI1CTAuODY2CTcKUDA1X0QxMF8wMjBfVzRDCVAwNQkxMF8wMjAJV2FybQk0CUMJMjg3NgkyNjc5CTUJMC4wMDIJMwpQMDVfRDIwXzA1MF9XNEMJUDA1CTIwXzA1MAlXYXJtCTQJQwkyMzIJMTEwCTAJMAkwClAwNV9ENTBfMTAwX1c0QwlQMDUJNTBfMTAwCVdhcm0JNAlDCTg3Nwk3MjYJMwkwLjAwMwkyClAwNl9EMDBfMDEwX0MwQwlQMDYJMDBfMDEwCUNvbnRyb2wJMAlDCTM1Mjc3CTM0NjU4CTI1NDI2CTAuNzIxCTIwClAwNl9EMTBfMDIwX0MwQwlQMDYJMTBfMDIwCUNvbnRyb2wJMAlDCTMxNTkwCTMwOTA4CTIxNDQ3CTAuNjc5CTExClAwNl9EMjBfMDUwX0MwQwlQMDYJMjBfMDUwCUNvbnRyb2wJMAlDCTczOAk2NDMJMwkwLjAwNAkzClAwNl9ENTBfMTAwX0MwQwlQMDYJNTBfMTAwCUNvbnRyb2wJMAlDCTQzNgkxMjgJMgkwLjAwNQkyClAwN19EMDBfMDEwX1c0RAlQMDcJMDBfMDEwCVdhcm0JNAlECTI0MzAyCTI0MDYxCTExOTQ5CTAuNDkyCTkKUDA3X0QxMF8wMjBfVzRECVAwNwkxMF8wMjAJV2FybQk0CUQJNTQ2NjAJNTM2MzkJNDQwMTUJMC44MDUJMTcKUDA3X0QyMF8wNTBfVzRECVAwNwkyMF8wNTAJV2FybQk0CUQJMzMwCTIyOAk5CTAuMDI3CTQKUDA3X0Q1MF8xMDBfVzRECVAwNwk1MF8xMDAJV2FybQk0CUQJMzc0CTI2Ngk2NQkwLjE3NAk0ClAwOF9EMDBfMDEwX0MwRAlQMDgJMDBfMDEwCUNvbnRyb2wJMAlECTQwODAJMzgxOQkzMjk4CTAuODA4CTExClAwOF9EMTBfMDIwX0MwRAlQMDgJMTBfMDIwCUNvbnRyb2wJMAlECTcyNTkJNzExNgk1OTk4CTAuODI2CTYKUDA4X0QyMF8wNTBfQzBECVAwOAkyMF8wNTAJQ29udHJvbAkwCUQJNjM2CTUwNgkyMDkJMC4zMjkJNApQMDhfRDUwXzEwMF9DMEQJUDA4CTUwXzEwMAlDb250cm9sCTAJRAk2NTYJNTk2CTIzMQkwLjM1Mgk0ClAwOV9EMDBfMDEwX1c0RQlQMDkJMDBfMDEwCVdhcm0JNAlFCTI1NjY1CTI1NDEzCTc1NDEJMC4yOTQJMjgKUDA5X0QxMF8wMjBfVzRFCVAwOQkxMF8wMjAJV2FybQk0CUUJMTYwNDIJMTU5MjIJMjMzMwkwLjE0NQkxMQpQMDlfRDIwXzA1MF9XNEUJUDA5CTIwXzA1MAlXYXJtCTQJRQkxMDE3CTkwOQkxCTAuMDAxCTEKUDA5X0Q1MF8xMDBfVzRFCVAwOQk1MF8xMDAJV2FybQk0CUUJMTcxCTYzCTEJMC4wMDYJMQpQMTBfRDAwXzAxMF9DMEUJUDEwCTAwXzAxMAlDb250cm9sCTAJRQk1Njc2Nwk1NjQxNAkyMTkzOAkwLjM4Ngk0NApQMTBfRDEwXzAyMF9DMEUJUDEwCTEwXzAyMAlDb250cm9sCTAJRQkxNzMxNwkxNzE3MQkxMjA2MwkwLjY5NwkxMwpQMTBfRDIwXzA1MF9DMEUJUDEwCTIwXzA1MAlDb250cm9sCTAJRQk5ODUJODc5CTAJMAkwClAxMF9ENTBfMTAwX0MwRQlQMTAJNTBfMTAwCUNvbnRyb2wJMAlFCTYxNwkyMzcJMQkwLjAwMgkxCg==').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'oo_sample_data_with_rc.txt'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
        });"><button class="btn btn-primary downloadthis  data-button" id="shrimp_md"><i class="bi bi-table"></i> Download sample metadata </button></a></p>
</section><section id="prep-data-for-microeco" class="level2"><h2 class="anchored" data-anchor-id="prep-data-for-microeco">Prep Data for microeco</h2>
<p>Like any tool, the microeco package needs the data in a specific form. I formatted our data to match the mock data in this section, <a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html#prepare-the-example-data">microeco</a> tutorial.</p>
<section id="a.-taxonomy-table" class="level3"><h3 class="anchored" data-anchor-id="a.-taxonomy-table">A. Taxonomy Table</h3>
<p>Here is what the taxonomy table looks like in the dataset.</p>
<div class="cell">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>       Domain Phylum     Class          Order
ASV163 Fungi  Ascomycota Eurotiomycetes Chaetothyriales
ASV118 Fungi  Ascomycota Eurotiomycetes Chaetothyriales
ASV86  ?      ?          ?              ?
ASV179 ?      ?          ?              ?
ASV132 Fungi  Ascomycota Eurotiomycetes Chaetothyriales
ASV156 Fungi  Ascomycota Eurotiomycetes Chaetothyriales</code></pre>
<p>Next we need to add rank definitions (e.g., <code>k__</code>, <code>p__</code>, etc.) to each classification.</p>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">physeq</span><span class="op">)</span><span class="op">)</span>   </span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"Kingdom"</span> <span class="op">=</span> <span class="st">"Domain"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html">mutate_all</a></span><span class="op">(</span><span class="op">~</span><span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"\\?"</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Kingdom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"k"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Kingdom</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Phylum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"p"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Phylum</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Class</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"c"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Class</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Order</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"o"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Order</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Family</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"f"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Family</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"g"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Genus</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"s"</span>, <span class="va">tmp_tax</span><span class="op">$</span><span class="va">Species</span>, sep <span class="op">=</span> <span class="st">"__"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And now the final, modified taxonomy table.</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_tax</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>       Kingdom  Phylum        Class             Order
ASV163 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales
ASV118 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales
ASV86  k__      p__           c__               o__
ASV179 k__      p__           c__               o__
ASV132 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales
ASV156 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales</code></pre>
</section><section id="b.-sequence-table" class="level3"><h3 class="anchored" data-anchor-id="b.-sequence-table">B. Sequence Table</h3>
<p>Here is what the sequence table looks like in the dataset.</p>
<div class="cell">
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_otu</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>       P00_D00_000_NNN P01_D00_010_W4A P01_D10_020_W4A
ASV163               0               0               0 
ASV118               0               0               0 
ASV86                0               0               0 
ASV179               0               0               0 
ASV132               0               0               0 
ASV156               0               0               0 </code></pre>
<div class="cell">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_otu</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>[1] TRUE</code></pre>
</section><section id="c.-sample-table" class="level3"><h3 class="anchored" data-anchor-id="c.-sample-table">C. Sample Table</h3>
<p>Here is what the sample table looks like.</p>
<div class="cell">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sample_tab</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       SampleID Plot  Depth Treatment Temp Pairing
P00_D00_000_NNN P00_D00_000_NNN  P00 00_000  Negative    N       N
P01_D00_010_W4A P01_D00_010_W4A  P01 00_010      Warm    4       A
P01_D10_020_W4A P01_D10_020_W4A  P01 10_020      Warm    4       A
P01_D20_050_W4A P01_D20_050_W4A  P01 20_050      Warm    4       A
P01_D50_100_W4A P01_D50_100_W4A  P01 50_100      Warm    4       A
P02_D00_010_C0A P02_D00_010_C0A  P02 00_010   Control    0       A</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="va">sample_tab</span><span class="op">$</span><span class="va">SampleID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sample_tab</span><span class="op">)</span></span>
<span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="va">SampleID</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="create-a-microtable-object" class="level2"><h2 class="anchored" data-anchor-id="create-a-microtable-object">Create a Microtable Object</h2>
<p>With these three files in hand we are now ready to create a <a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html">microtable object</a>.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>A <a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html">microtable object</a> contains an ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications). It can also contain a phylogenetic tree of ASVs as well as representative sequences of each ASV.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span></span>
<span><span class="va">tax_tab</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span></span>
<span><span class="va">otu_tab</span> <span class="op">&lt;-</span> <span class="va">tmp_otu</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_me</span> <span class="op">&lt;-</span> <span class="va"><a href="https://rdrr.io/pkg/microeco/man/microtable.html">microtable</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>sample_table <span class="op">=</span> <span class="va">sample_tab</span>, </span>
<span>                         otu_table <span class="op">=</span> <span class="va">otu_tab</span>, </span>
<span>                         tax_table <span class="op">=</span> <span class="va">tax_tab</span><span class="op">)</span></span>
<span><span class="va">tmp_me</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>2 samples with 0 abundance are removed from the otu_table ...

microtable-class object:
sample_table have 41 rows and 11 columns
otu_table have 12091 rows and 41 columns
tax_table have 12091 rows and 7 columns</code></pre>
<section id="d.-add-representative-sequence" class="level3"><h3 class="anchored" data-anchor-id="d.-add-representative-sequence">D. Add Representative Sequence</h3>
<p>We can also add representative sequences for each OTU/ASV. For this step, we can simply grab the sequences from the row names of the DADA2 taxonomy object loaded above.</p>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readDNAStringSet</a></span><span class="op">(</span><span class="st">"working_files/LOTUS3/oo/OTU.fna"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">$</span><span class="va">tax_table</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">$</span><span class="va">otu_table</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>[1] TRUE</code></pre>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_fasta_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">rep_fasta</span><span class="op">)</span></span>
<span><span class="va">tmp_rep_fasta</span> <span class="op">&lt;-</span> <span class="va">rep_fasta</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">$</span><span class="va">tax_table</span><span class="op">)</span>, <span class="va">tmp_fasta_names</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="va">tmp_rep_fasta</span></span>
<span><span class="va">tmp_me</span><span class="op">$</span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="va">rep_fasta</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">$</span><span class="va">tax_table</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">rep_fasta</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_me</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>[1] TRUE</code></pre>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_asv_raw</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">)</span></span>
<span><span class="va">me_asv_raw</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 39 rows and 19 columns
otu_table have 192 rows and 39 columns
tax_table have 192 rows and 7 columns
rep_fasta have 192 sequences</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>The microeco object <code>me_asv_raw</code> contains all data from the LotuS3 pipeline, before any dataset curation.</p>
</blockquote>
</section></section><section id="curate-the-data-set" class="level2"><h2 class="anchored" data-anchor-id="curate-the-data-set">Curate the Data Set</h2>
<p>Pretty much the last thing to do is remove unwanted taxa, negative controls, and low-count samples.</p>
<section id="remove-non-fungal-asvs" class="level3"><h3 class="anchored" data-anchor-id="remove-non-fungal-asvs">Remove non-fungal ASVs</h3>
<p>First we can have a quick look at the kingdom level classifications to see what we are dealing with.</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_k</span> <span class="op">&lt;-</span> <span class="va">me_asv_raw</span><span class="op">$</span><span class="va">tax_table</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">distinct</span><span class="op">(</span><span class="va">Kingdom</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tmp_p</span> <span class="op">&lt;-</span> <span class="va">me_asv_raw</span><span class="op">$</span><span class="va">tax_table</span> <span class="op">%&gt;%</span></span>
<span>      <span class="fu">distinct</span><span class="op">(</span><span class="va">Phylum</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>-------KINGDOM------------------</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>k__Fungi
k__
k__Eukaryota_kgd_Incertae_sedis
k__Ichthyosporia
k__Stramenopila
k__Alveolata</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-------PHYLUM------------------</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>p__Ascomycota
p__
p__Basidiomycota
p__Eukaryota_phy_Incertae_sedis
p__Fungi_phy_Incertae_sedis
p__Ichthyosporia_phy_Incertae_sedis
p__Oomycota
p__Mucoromycota
p__Stramenopila_phy_Incertae_sedis
p__Ochrophyta
p__Dinophyta</code></pre>
</div>
</div>
<ol type="1">
<li>The first thing to notice is that at the kingdom level we have Fungi, unclassified (or NA), Stramenopila, some non-fungal classifications. We can remove anything that is not classified as Fungi at the kingdom level or Oomycota at the phylum level.</li>
</ol>
<p>The easiest thing to do is keep everything we want–in other words, all fungal taxa.</p>
<div class="cell">
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_na</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">)</span></span>
<span><span class="va">tmp_no_na</span><span class="op">$</span><span class="va">tax_table</span> <span class="op">%&lt;&gt;%</span> </span>
<span>  <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"k__Fungi"</span> <span class="op">|</span> <span class="va">Phylum</span> <span class="op">==</span> <span class="st">"p__Oomycota"</span><span class="op">)</span></span>
<span><span class="va">tmp_no_na</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_asv_no_na</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_na</span><span class="op">)</span></span>
<span><span class="va">me_asv_no_na</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 39 rows and 19 columns
otu_table have 91 rows and 39 columns
tax_table have 91 rows and 7 columns
rep_fasta have 91 sequences</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>The microeco object <code>me_asv_no_na</code> conatins only ASVs classified as Archaea or Bacteria.</p>
</blockquote>
</section><section id="remove-negative-controls-nc" class="level3"><h3 class="anchored" data-anchor-id="remove-negative-controls-nc">Remove Negative Controls (NC)</h3>
<p>Now we need to remove the NC samples <em>and</em> ASVs found in those sample. We first identified all ASVs that were present in at least one NC sample represented by at least 1 read. We did this by subsetting the NC samples from the new microtable object.</p>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_na</span><span class="op">)</span></span>
<span><span class="va">tmp_nc</span><span class="op">$</span><span class="va">sample_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_nc</span><span class="op">$</span><span class="va">sample_table</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"Negative"</span><span class="op">)</span></span>
<span><span class="va">tmp_nc</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>89 taxa with 0 abundance are removed from the otu_table ...</code></pre>
<div class="cell">
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_asv_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_nc</span><span class="op">)</span></span>
<span><span class="va">me_asv_nc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 1 rows and 19 columns
otu_table have 2 rows and 1 columns
tax_table have 2 rows and 7 columns
rep_fasta have 2 sequences</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>The microeco object <code>me_asv_nc</code> conatins only the negative control sample(s) and associated ASVs.</p>
</blockquote>
<p>Looks like there are 2 ASVs in the NC samples from a total of 2 reads.</p>
<div class="cell">
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_asvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_nc</span><span class="op">$</span><span class="va">tax_table</span><span class="op">)</span></span>
<span><span class="va">nc_asvs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2</code></pre>
</div>
</div>
<p>There are 2 ASVs found in the NC sample.</p>
<p>Essentially, for each ASV, the code below calculates:</p>
<ul>
<li>The total number of NC samples containing at least 1 read.<br>
</li>
<li>The total number of reads in NC samples.<br>
</li>
<li>The total number of non-NC samples containing at least 1 read.<br>
</li>
<li>The total number of reads in non-NC samples.<br>
</li>
<li>The percent of reads in the NC samples and the percent of NC samples containing reads.</li>
</ul>
<div class="cell">
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_rem_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_na</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span> </span>
<span>                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">nc_asvs</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>total_reads_NC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"NNN"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>         .after <span class="op">=</span> <span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="st">"NNN"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>total_reads_samps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">.</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>, </span>
<span>                .after <span class="op">=</span> <span class="st">"total_reads_NC"</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span><span class="op">[</span>, <span class="fl">4</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>perc_in_neg <span class="op">=</span> <span class="fl">100</span><span class="op">*</span><span class="op">(</span></span>
<span>    <span class="va">total_reads_NC</span> <span class="op">/</span> <span class="op">(</span><span class="va">total_reads_NC</span> <span class="op">+</span> <span class="va">total_reads_samps</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                .after <span class="op">=</span> <span class="st">"total_reads_samps"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_rem_nc_df</span><span class="op">$</span><span class="va">perc_in_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">$</span><span class="va">perc_in_neg</span>, digits <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_1</span> <span class="op">&lt;-</span> <span class="va">tmp_1</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span><span class="va">tmp_1</span> <span class="op">&lt;-</span> <span class="va">tmp_1</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"total_samples"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>  </span>
<span></span>
<span><span class="va">tmp_2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"NNN"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_2</span><span class="op">$</span><span class="va">num_samp_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_2</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">tmp_2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_2</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"num_samp_nc"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_2</span> <span class="op">&lt;-</span> <span class="va">tmp_2</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_3</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="st">"NNN"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_3</span><span class="op">$</span><span class="va">num_samp_no_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_3</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">tmp_3</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_3</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"num_samp_no_nc"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_3</span> <span class="op">&lt;-</span> <span class="va">tmp_3</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span>, <span class="va">tmp_1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    perc_in_neg_samp <span class="op">=</span> <span class="fl">100</span><span class="op">*</span><span class="op">(</span> <span class="va">num_samp_nc</span> <span class="op">/</span> <span class="op">(</span><span class="va">num_samp_nc</span> <span class="op">+</span> <span class="va">num_samp_no_nc</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    .after <span class="op">=</span> <span class="st">"num_samp_no_nc"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_check</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div id="tbl-make-asv-in-nc-table-oo" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-make-asv-in-nc-table-oo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;2: Summary of ASVs detected in Negative Control (NC) samples.
</figcaption><div aria-describedby="tbl-make-asv-in-nc-table-oo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-189bf7e795965c842df7" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-189bf7e795965c842df7">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"ASV_ID":["ASV62","ASV1"],"total_reads_NC":[1,1],"total_reads_samps":[496,117716],"perc_in_neg":[0.201207,0.000849],"total_samples":[5,30],"num_samp_nc":[1,1],"num_samp_no_nc":[4,29],"perc_in_neg_samp":[20,3.33333333333333]},"columns":[{"id":"ASV_ID","name":"ASV ID","type":"character","sortable":true,"resizable":true,"filterable":false,"header":"ASV ID","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":100,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"id":"total_reads_NC","name":"reads in NC","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"reads in NC","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"total_reads_samps","name":"reads in non NC","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"reads in non NC","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"perc_in_neg","name":"% in NC","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"% in NC","align":"center","footerStyle":{"fontWeight":"bold"},"format":{"cell":{"digits":2},"aggregated":{"digits":2}}},{"id":"total_samples","name":"total samples","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"total samples","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"num_samp_nc","name":"Total NC samples","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"Total NC samples","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"num_samp_no_nc","name":"Total non-NC samples","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"Total non-NC samples","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"perc_in_neg_samp","name":"% of samples","type":"numeric","sortable":true,"resizable":true,"filterable":false,"header":"% of samples","align":"center","footerStyle":{"fontWeight":"bold"},"format":{"cell":{"digits":2},"aggregated":{"digits":2}}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,2],"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"9e276cb946e2fea26019517f210de32a"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</figure>
</div>
</div>
<p><a href="#asv_in_nc_samples" onclick="fetch('data:text/plain;base64,QVNWX0lECXRvdGFsX3JlYWRzX05DCXRvdGFsX3JlYWRzX3NhbXBzCXBlcmNfaW5fbmVnCXRvdGFsX3NhbXBsZXMJbnVtX3NhbXBfbmMJbnVtX3NhbXBfbm9fbmMJcGVyY19pbl9uZWdfc2FtcApBU1Y2MgkxCTQ5NgkwLjIwMTIwNwk1CTEJNAkyMApBU1YxCTEJMTE3NzE2CTguNDllLTQJMzAJMQkyOQkzLjMzMzMzMzMzMzMzMzMzMzUK').then(res => res.blob()).then(blob => {
      const downloadURL = window.URL.createObjectURL(blob);
      const a = document.createElement('a');
      document.body.appendChild(a);
      a.href = downloadURL;
      a.download = 'oo_asv_in_nc_samples.txt'; a.click();
      window.URL.revokeObjectURL(downloadURL);
      document.body.removeChild(a);
        });"><button class="btn btn-primary downloadthis  data-button" id="asv_in_nc_samples"><i class="bi bi-table"></i> Download ASVs in NC samples </button></a></p>
<p>Looking at these data we can see the NC sample only has 1 read from each of the two ASVs. We can simply ignore these ASVs and just remove the NC sample. We will still go through the process…</p>
<div class="cell">
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_remove</span> <span class="op">&lt;-</span> <span class="va">nc_check</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">perc_in_neg</span> <span class="op">&gt;</span> <span class="fl">10</span> <span class="op">|</span> <span class="va">num_samp_no_nc</span> <span class="op">&lt;</span> <span class="fl">4</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="caption-top table">
<colgroup>
<col style="width: 11%">
<col style="width: 24%">
<col style="width: 20%">
<col style="width: 21%">
<col style="width: 21%">
</colgroup>
<thead><tr class="header">
<th></th>
<th>Total ASVs</th>
<th>NC reads</th>
<th>non NC reads</th>
<th>% NC reads</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Removed</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>NaN</td>
</tr>
<tr class="even">
<td>Retained</td>
<td>2</td>
<td>2</td>
<td>118212</td>
<td>0.002</td>
</tr>
</tbody>
</table>
<p>We identified a total of 2 ASVs that were present in at least 1 NC sample by at least 1 read. We removed any ASV where more than 10% of total reads were found in NC samples OR any ASV found in more than 10% of NC samples. Based on these criteria we removed 0 ASVs from the data set, which represented 0 total reads in NC samples and 0 total reads in non-NC samples. Of the total reads removed NaN% came from NC samples. Of all ASVs identified in NC samples,2 were retained because the fell below the threshhold criteria. These ASVs accounted for 2 reads in NC samples and 118212 reads in non-NC samples. NC samples accounted for 0.002% of these reads.</p>
<p>OK, now we can remove the NC samples and any ASVs that met our criteria described above.</p>
<div class="cell">
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_na</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_rem_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">nc_remove</span><span class="op">$</span><span class="va">ASV_ID</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">otu_table</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">tmp_rem_asv</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">sample_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">sample_table</span>, </span>
<span>                                 <span class="va">Treatment</span> <span class="op">!=</span> <span class="st">"Negative"</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_asv_no_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">)</span></span>
<span><span class="va">me_asv_no_nc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 38 rows and 19 columns
otu_table have 91 rows and 38 columns
tax_table have 91 rows and 7 columns
rep_fasta have 91 sequences</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>The microeco object <code>me_asv_no_nc</code> does not conatins the negative control sample(s).</p>
</blockquote>
</section><section id="remove-low-count-samples" class="level3"><h3 class="anchored" data-anchor-id="remove-low-count-samples">Remove Low-Count Samples</h3>
<p>Next, we can remove samples with really low read counts. I know from looking at these data that the are many low coverage samples. We can take a look at different thresholds to see how many samples will be eliminated.</p>
<div class="cell">
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">samp_sum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">me_asv_no_nc</span><span class="op">$</span><span class="fu">sample_sums</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>--- &lt;1000 reads --- 18  samples </code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>--- &lt;500 reads --- 17  samples </code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>--- &lt;100 reads --- 14  samples </code></pre>
</div>
</div>
<p>We can start by eliminating samples with fewer than 100 reads.</p>
<div class="cell">
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_low</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_low</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">&lt;-</span> <span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">%&gt;%</span></span>
<span>          <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">is.numeric</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>3 taxa with 0 abundance are removed from the otu_table ...</code></pre>
<p>Check if any row sum (i.e., ASVs) is equal to 0 after removing the negative control sample(s).</p>
<div class="cell">
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_row_sums</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_no_low</span><span class="op">$</span><span class="va">otu_table</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="va">tmp_row_sums</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<pre><code>[1] FALSE</code></pre>
<div class="cell">
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_asv_no_low</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_low</span><span class="op">)</span></span>
<span><span class="va">me_asv_no_low</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 24 rows and 19 columns
otu_table have 88 rows and 24 columns
tax_table have 88 rows and 7 columns
rep_fasta have 88 sequences</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>The microeco object <code>me_asv_no_low</code> does not conatins the low abundance samples.</p>
</blockquote>
<p>We will quickly add final read counts and ASVs, then create the final microtable object.</p>
<div class="cell">
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_final_rc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">me_asv_no_low</span><span class="op">$</span><span class="fu">sample_sums</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"final_rc"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_final_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">me_asv_no_low</span><span class="op">$</span><span class="va">otu_table</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_final_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_final_asv</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"final_asv"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">tmp_final</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_final_rc</span>, <span class="va">tmp_final_asv</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">full_join</a></span><span class="op">(</span><span class="va">sample_tab</span>, <span class="va">tmp_final</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>        <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">final_rc</span>, <span class="va">final_asv</span><span class="op">)</span>, .after <span class="op">=</span> <span class="va">lotus3_asv</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="va">sample_tab</span><span class="op">$</span><span class="va">SampleID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">sample_tab</span><span class="op">)</span></span>
<span><span class="va">sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="va">SampleID</span><span class="op">)</span></span>
<span></span>
<span><span class="va">full_sample_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_samp_tab</span> <span class="op">&lt;-</span> <span class="va">sample_tab</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/complete.cases.html">complete.cases</a></span><span class="op">(</span><span class="va">sample_tab</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">tmp_tax_tab</span> <span class="op">&lt;-</span> <span class="va">me_asv_no_low</span><span class="op">$</span><span class="va">tax_table</span></span>
<span><span class="va">tmp_otu_tab</span> <span class="op">&lt;-</span> <span class="va">me_asv_no_low</span><span class="op">$</span><span class="va">otu_table</span></span>
<span><span class="va">tmp_rep_fasta</span> <span class="op">&lt;-</span> <span class="va">me_asv_no_low</span><span class="op">$</span><span class="va">rep_fasta</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_asv</span> <span class="op">&lt;-</span> <span class="va"><a href="https://rdrr.io/pkg/microeco/man/microtable.html">microtable</a></span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>sample_table <span class="op">=</span> <span class="va">tmp_samp_tab</span>, </span>
<span>                         otu_table <span class="op">=</span> <span class="va">tmp_otu_tab</span>, </span>
<span>                         tax_table <span class="op">=</span> <span class="va">tmp_tax_tab</span>, </span>
<span>                         rep_fasta <span class="op">=</span> <span class="va">tmp_rep_fasta</span><span class="op">)</span></span>
<span><span class="va">me_asv</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 24 rows and 21 columns
otu_table have 88 rows and 24 columns
tax_table have 88 rows and 7 columns
rep_fasta have 88 sequences</code></pre>
</div>
</div>
</section></section></section><section id="summary" class="level1"><h1>Summary</h1>
<p>Now time to summarize the data. For this we use the R package <a href="https://microbiome.github.io">miaverse</a> <span class="citation" data-cites="felix2024mia">(<a href="#ref-felix2024mia" role="doc-biblioref">Borman et al. 2024</a>)</span>.</p>
<p>Here we can see how the total number of ASVs, reads, and samples changed during the curation process.</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;">me_dataset</th>
<th style="text-align: right;">total_asvs</th>
<th style="text-align: right;">total_reads</th>
<th style="text-align: right;">total_samples</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">original</td>
<td style="text-align: right;">192</td>
<td style="text-align: right;">336493</td>
<td style="text-align: right;">39</td>
</tr>
<tr class="even">
<td style="text-align: left;">No NA kingdom</td>
<td style="text-align: right;">91</td>
<td style="text-align: right;">268599</td>
<td style="text-align: right;">39</td>
</tr>
<tr class="odd">
<td style="text-align: left;">no neg control</td>
<td style="text-align: right;">91</td>
<td style="text-align: right;">268597</td>
<td style="text-align: right;">38</td>
</tr>
<tr class="even">
<td style="text-align: left;">no low count samps</td>
<td style="text-align: right;">88</td>
<td style="text-align: right;">268446</td>
<td style="text-align: right;">24</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>And some additional metrics describing the changes.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 21%">
<col style="width: 41%">
<col style="width: 37%">
</colgroup>
<thead><tr class="header">
<th>Metric</th>
<th>Start</th>
<th>End</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Min. number reads</td>
<td>1</td>
<td>231</td>
</tr>
<tr class="even">
<td>Max. number reads</td>
<td>89813</td>
<td>51535</td>
</tr>
<tr class="odd">
<td>Total number reads</td>
<td>336493</td>
<td>268446</td>
</tr>
<tr class="even">
<td>Avg number reads</td>
<td>8628</td>
<td>11185</td>
</tr>
<tr class="odd">
<td>Median no. reads</td>
<td>2397</td>
<td>5919.5</td>
</tr>
<tr class="even">
<td>Total ASVs</td>
<td>192</td>
<td>88</td>
</tr>
<tr class="odd">
<td>No.&nbsp;singleton ASVs</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="even">
<td>Avg ASVs per sample.</td>
<td>11</td>
<td>9</td>
</tr>
</tbody>
</table>
<p>We started off with 192 ASVs and 39 samples. Screening for <code>NA</code> kingdom assignment removed an additional 101 ASVs. After removing the negative controls there were 91 ASVs and 39. After removing low-count samples, there were 88 ASVs and 24 samples remaining.</p>
<div class="cell">
<div id="tbl-make-final-rc-asv-table-oo" class="cell quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure"><figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-make-final-rc-asv-table-oo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3: Beginning and ending read counts &amp; total ASVs by sample.
</figcaption><div aria-describedby="tbl-make-final-rc-asv-table-oo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-4e280cda47fb19178af4" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4e280cda47fb19178af4">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"SampleID":["P00_D00_000_NNN","P01_D00_010_W4A","P01_D10_020_W4A","P01_D20_050_W4A","P01_D50_100_W4A","P02_D00_010_C0A","P02_D10_020_C0A","P02_D20_050_C0A","P02_D50_100_C0A","P03_D00_010_W4B","P03_D10_020_W4B","P03_D20_050_W4B","P03_D50_100_W4B","P04_D00_010_C0B","P04_D10_020_C0B","P04_D20_050_C0B","P04_D50_100_C0B","P05_D00_010_W4C","P05_D10_020_W4C","P05_D50_100_W4C","P06_D00_010_C0C","P06_D10_020_C0C","P06_D20_050_C0C","P06_D50_100_C0C","P07_D00_010_W4D","P07_D10_020_W4D","P07_D20_050_W4D","P07_D50_100_W4D","P08_D00_010_C0D","P08_D10_020_C0D","P08_D20_050_C0D","P08_D50_100_C0D","P09_D00_010_W4E","P09_D10_020_W4E","P09_D20_050_W4E","P09_D50_100_W4E","P10_D00_010_C0E","P10_D10_020_C0E","P10_D50_100_C0E"],"start_rc":[2,6979,7658,29,27,21540,11654,2,411,89813,9132,3581,5,13604,232,4264,2397,8625,5,3,25426,21447,3,2,11949,44015,9,65,3298,5998,209,231,7541,2333,1,1,21938,12063,1],"start_asv":[2,36,16,2,2,28,10,2,6,45,9,7,4,36,5,6,4,7,3,2,20,11,3,2,9,17,4,4,11,6,4,4,28,11,1,1,44,13,1],"final_rc":["NA",5842,2734,"NA","NA",18602,11558,"NA",395,51535,8361,3581,"NA",12398,232,4262,2395,6175,"NA","NA",25133,20286,"NA","NA",5832,42417,"NA","NA",3108,5997,"NA",231,4999,853,"NA","NA",19458,12062,"NA"],"final_asv":["NA",11,8,"NA","NA",18,9,"NA",5,11,2,7,"NA",20,5,4,2,6,"NA","NA",16,8,"NA","NA",4,13,"NA","NA",7,5,"NA",4,14,5,"NA","NA",28,12,"NA"]},"columns":[{"id":"SampleID","name":"SampleID","type":"character","sortable":true,"resizable":true,"filterable":false,"cell":["P00_D00_000_NNN","P01_D00_010_W4A","P01_D10_020_W4A","P01_D20_050_W4A","P01_D50_100_W4A","P02_D00_010_C0A","P02_D10_020_C0A","P02_D20_050_C0A","P02_D50_100_C0A","P03_D00_010_W4B","P03_D10_020_W4B","P03_D20_050_W4B","P03_D50_100_W4B","P04_D00_010_C0B","P04_D10_020_C0B","P04_D20_050_C0B","P04_D50_100_C0B","P05_D00_010_W4C","P05_D10_020_W4C","P05_D50_100_W4C","P06_D00_010_C0C","P06_D10_020_C0C","P06_D20_050_C0C","P06_D50_100_C0C","P07_D00_010_W4D","P07_D10_020_W4D","P07_D20_050_W4D","P07_D50_100_W4D","P08_D00_010_C0D","P08_D10_020_C0D","P08_D20_050_C0D","P08_D50_100_C0D","P09_D00_010_W4E","P09_D10_020_W4E","P09_D20_050_W4E","P09_D50_100_W4E","P10_D00_010_C0E","P10_D10_020_C0E","P10_D50_100_C0E"],"header":"SampleID","align":"left","footerStyle":{"fontWeight":"bold"},"minWidth":150,"sticky":"left","style":{"borderRight":"1px solid #eee"},"headerStyle":{"borderRight":"1px solid #eee"}},{"id":"start_rc","name":"start rc","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["2","6979","7658","29","27","21540","11654","2","411","89813","9132","3581","5","13604","232","4264","2397","8625","5","3","25426","21447","3","2","11949","44015","9","65","3298","5998","209","231","7541","2333","1","1","21938","12063","1"],"header":"start rc","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"start_asv","name":"start asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["2","36","16","2","2","28","10","2","6","45","9","7","4","36","5","6","4","7","3","2","20","11","3","2","9","17","4","4","11","6","4","4","28","11","1","1","44","13","1"],"header":"start asv","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"final_rc","name":"final rc","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["NA","5842","2734","NA","NA","18602","11558","NA","395","51535","8361","3581","NA","12398","232","4262","2395","6175","NA","NA","25133","20286","NA","NA","5832","42417","NA","NA","3108","5997","NA","231","4999","853","NA","NA","19458","12062","NA"],"header":"final rc","align":"center","footerStyle":{"fontWeight":"bold"}},{"id":"final_asv","name":"final asv","type":"numeric","sortable":true,"resizable":true,"filterable":false,"cell":["NA","11","8","NA","NA","18","9","NA","5","11","2","7","NA","20","5","4","2","6","NA","NA","16","8","NA","NA","4","13","NA","NA","7","5","NA","4","14","5","NA","NA","28","12","NA"],"header":"final asv","align":"center","footerStyle":{"fontWeight":"bold"}}],"defaultPageSize":5,"showPageSizeOptions":true,"pageSizeOptions":[5,10,39],"highlight":true,"bordered":true,"striped":true,"nowrap":true,"showSortable":true,"theme":{"style":{"fontSize":"0.8em"}},"dataKey":"206318389a10b66a3da69045ce699174"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</figure>
</div>
</div>
<p></p>
<p>We lost 15 samples after curating the dataset.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "P00_D00_000_NNN" "P01_D20_050_W4A" "P01_D50_100_W4A" "P02_D20_050_C0A"
 [5] "P03_D50_100_W4B" "P05_D10_020_W4C" "P05_D50_100_W4C" "P06_D20_050_C0C"
 [9] "P06_D50_100_C0C" "P07_D20_050_W4D" "P07_D50_100_W4D" "P08_D20_050_C0D"
[13] "P09_D20_050_W4E" "P09_D50_100_W4E" "P10_D50_100_C0E"</code></pre>
</div>
</div>
</section><section class="level1"><h1>Links</h1>
<ul>
<li><a href="https://chiliubio.github.io/microeco_tutorial/">microeco</a></li>
<li><a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html#prepare-the-example-data">microeco</a></li>
<li><a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html">microtable object</a></li>
<li><a href="https://microbiome.github.io">miaverse</a></li>
<li><a href="https://github.com/sweltr/depth/blob/main/curate_oo.qmd">curate_oo.qmd</a></li>
<li><a href="https://doi.org/10.25573/data.16828063">10.25573/data.16828063</a></li>
<li><a href="https://metacrobe.github.io/">metacrobe</a></li>
<li><a href="https://quarto.org/">Quarto</a></li>
</ul></section><div id="quarto-appendix" class="default"><section id="source-code" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Source Code</h2><div class="quarto-appendix-contents">
<p>The source code for this page can be found on GitHub <svg aria-hidden="true" role="img" viewbox="0 0 496 512" style="height:1em;width:0.97em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"></path></svg> by visiting the <a href="https://github.com/sweltr/depth/blob/main/curate_oo.qmd">curate_oo.qmd</a> page.</p>
</div></section><section id="data-availability" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Data Availability</h2><div class="quarto-appendix-contents">
<p>Data generated in this workflow and the Rdata need to run the workflow can be accessed on figshare at <a href="https://doi.org/10.25573/data.16828063">10.25573/data.16828063</a>.</p>
</div></section><section id="last-updated-on" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<p>2025-08-28 15:39:27.44519</p>


<!-- -->

</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/sweltr\.github\.io\/depth\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./curate_amf.html" class="pagination-link" aria-label="AMF Data Curation">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">AMF Data Curation</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb66" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Oomycete Data Curation"</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Workflow for curating the ASV dataset--generating microeco &amp; phyloseq objects, removing spurious ASVs and low-count/negative control samples."</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a><span class="co">    other-links: </span></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="co">      - text: Data</span></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="co">        href: https://doi.org/10.25573/data.16828063</span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/code/_setup.qmd &gt;}}</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/code/_curate.qmd &gt;}}</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a><span class="fu"># Overview</span></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>On this page we use the phyloseq output from LotuS3 to format the otu, taxonomy, and sample data tables so we can create a microeco object. Once that is complete we curate the dataset by removing possible contaminants, negative control samples, NA kingdoms, and low-count samples.  </span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data </span></span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>All you need to run this workflow is the phyloseq object generated by the LotuS3 pipeline.</span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-23"><a href="#cb66-23" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="default" icon=false}</span>
<span id="cb66-24"><a href="#cb66-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-25"><a href="#cb66-25" aria-hidden="true" tabindex="-1"></a><span class="fu">### {{&lt; fa download &gt;}} &amp;nbsp; phyloseq_oo.Rdata</span></span>
<span id="cb66-26"><a href="#cb66-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-27"><a href="#cb66-27" aria-hidden="true" tabindex="-1"></a>{{&lt; downloadthis files/CURATE/phyloseq_oo.Rdata dname="phyloseq_oo" label="LotuS3 phyloseq data" icon="code-slash" type="link" &gt;}}</span>
<span id="cb66-28"><a href="#cb66-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-29"><a href="#cb66-29" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb66-30"><a href="#cb66-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-31"><a href="#cb66-31" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;/</span><span class="kw">br</span><span class="dt">&gt;</span></span>
<span id="cb66-32"><a href="#cb66-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-33"><a href="#cb66-33" aria-hidden="true" tabindex="-1"></a><span class="fu">## Citable resources </span></span>
<span id="cb66-34"><a href="#cb66-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-35"><a href="#cb66-35" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Packages used in this workflow:</span></span>
<span id="cb66-36"><a href="#cb66-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-37"><a href="#cb66-37" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>R microeco package <span class="co">[</span><span class="ot">@liu2021microeco</span><span class="co">]</span>. </span>
<span id="cb66-38"><a href="#cb66-38" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>phyloseq R package <span class="co">[</span><span class="ot">@mcmurdie2013phyloseq</span><span class="co">]</span>.</span>
<span id="cb66-39"><a href="#cb66-39" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>miaverse <span class="co">[</span><span class="ot">@felix2024mia</span><span class="co">]</span></span>
<span id="cb66-40"><a href="#cb66-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-41"><a href="#cb66-41" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;/</span><span class="kw">br</span><span class="dt">&gt;</span></span>
<span id="cb66-42"><a href="#cb66-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-43"><a href="#cb66-43" aria-hidden="true" tabindex="-1"></a><span class="fu">## References</span></span>
<span id="cb66-44"><a href="#cb66-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-45"><a href="#cb66-45" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb66-46"><a href="#cb66-46" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb66-47"><a href="#cb66-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-48"><a href="#cb66-48" aria-hidden="true" tabindex="-1"></a><span class="fu"># Workflow</span></span>
<span id="cb66-49"><a href="#cb66-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-50"><a href="#cb66-50" aria-hidden="true" tabindex="-1"></a><span class="fu">## Required Packages</span></span>
<span id="cb66-51"><a href="#cb66-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-54"><a href="#cb66-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-55"><a href="#cb66-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb66-56"><a href="#cb66-56" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb66-57"><a href="#cb66-57" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb66-58"><a href="#cb66-58" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Click here for required R packages."</span></span>
<span id="cb66-59"><a href="#cb66-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-60"><a href="#cb66-60" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">919191</span>)</span>
<span id="cb66-61"><a href="#cb66-61" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microeco)</span>
<span id="cb66-62"><a href="#cb66-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb66-63"><a href="#cb66-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb66-64"><a href="#cb66-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb66-65"><a href="#cb66-65" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-66"><a href="#cb66-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-69"><a href="#cb66-69" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-70"><a href="#cb66-70" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-71"><a href="#cb66-71" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-72"><a href="#cb66-72" aria-hidden="true" tabindex="-1"></a><span class="co">#rm(list = ls())</span></span>
<span id="cb66-73"><a href="#cb66-73" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"page_build/oo_curate.rdata"</span>)</span>
<span id="cb66-74"><a href="#cb66-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-75"><a href="#cb66-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-76"><a href="#cb66-76" aria-hidden="true" tabindex="-1"></a><span class="fu">## Review Results</span></span>
<span id="cb66-77"><a href="#cb66-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-78"><a href="#cb66-78" aria-hidden="true" tabindex="-1"></a>The LotuS3 pipeline produces numerous output files  but for our purposes there are four specific files we are interested in:</span>
<span id="cb66-79"><a href="#cb66-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-80"><a href="#cb66-80" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span><span class="in">`phyloseq.Rdata`</span></span>
<span id="cb66-81"><a href="#cb66-81" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span><span class="in">`OTU.fna`</span></span>
<span id="cb66-82"><a href="#cb66-82" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span><span class="in">`OTU.txt`</span></span>
<span id="cb66-83"><a href="#cb66-83" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span><span class="in">`hiera_BLAST.txt`</span></span>
<span id="cb66-84"><a href="#cb66-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-85"><a href="#cb66-85" aria-hidden="true" tabindex="-1"></a>We will mainly work from the <span class="in">`phyloseq.Rdata`</span> (renamed <span class="in">`phyloseq_oo.Rdata`</span>) since this contains the OTU Table, Sample Data, Taxonomy Table, and Phylogenetic Tree. Note: a phylogenetic tree of microbial ASVs from short reads seems less than useful. Therefore, we will exclude this from the initial analysis. </span>
<span id="cb66-86"><a href="#cb66-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-89"><a href="#cb66-89" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-90"><a href="#cb66-90" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-91"><a href="#cb66-91" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-92"><a href="#cb66-92" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"working_files/LOTUS3/oo/phyloseq_oo.Rdata"</span>)</span>
<span id="cb66-93"><a href="#cb66-93" aria-hidden="true" tabindex="-1"></a>physeq</span>
<span id="cb66-94"><a href="#cb66-94" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-95"><a href="#cb66-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-96"><a href="#cb66-96" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-97"><a href="#cb66-97" aria-hidden="true" tabindex="-1"></a><span class="in">phyloseq-class experiment-level object</span></span>
<span id="cb66-98"><a href="#cb66-98" aria-hidden="true" tabindex="-1"></a><span class="in">otu_table()   OTU Table:         [ 12091 taxa and 41 samples ]</span></span>
<span id="cb66-99"><a href="#cb66-99" aria-hidden="true" tabindex="-1"></a><span class="in">sample_data() Sample Data:       [ 41 samples by 14 sample variables ]</span></span>
<span id="cb66-100"><a href="#cb66-100" aria-hidden="true" tabindex="-1"></a><span class="in">tax_table()   Taxonomy Table:    [ 12091 taxa by 7 taxonomic ranks ]</span></span>
<span id="cb66-101"><a href="#cb66-101" aria-hidden="true" tabindex="-1"></a><span class="in">phy_tree()    Phylogenetic Tree: [ 12091 tips and 12089 internal nodes ]</span></span>
<span id="cb66-102"><a href="#cb66-102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-103"><a href="#cb66-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-106"><a href="#cb66-106" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb66-107"><a href="#cb66-107" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-108"><a href="#cb66-108" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-109"><a href="#cb66-109" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> working_files/LOTUS3/oo/phyloseq_oo.Rdata files/CURATE</span>
<span id="cb66-110"><a href="#cb66-110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-111"><a href="#cb66-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-114"><a href="#cb66-114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-115"><a href="#cb66-115" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-116"><a href="#cb66-116" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-117"><a href="#cb66-117" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: this is for fixing two column names in the sample_data rather than rerunning the pipeline. </span></span>
<span id="cb66-118"><a href="#cb66-118" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(speedyseq)</span>
<span id="cb66-119"><a href="#cb66-119" aria-hidden="true" tabindex="-1"></a>physeq <span class="ot">&lt;-</span> physeq <span class="sc">%&gt;%</span> </span>
<span id="cb66-120"><a href="#cb66-120" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rename_sample_data</span>(<span class="at">I7_Index =</span> index) <span class="sc">%&gt;%</span></span>
<span id="cb66-121"><a href="#cb66-121" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rename_sample_data</span>(<span class="at">I5_Index =</span> index2) </span>
<span id="cb66-122"><a href="#cb66-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-123"><a href="#cb66-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-126"><a href="#cb66-126" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-127"><a href="#cb66-127" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-128"><a href="#cb66-128" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-129"><a href="#cb66-129" aria-hidden="true" tabindex="-1"></a>min_read_ps <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">readcount</span>(physeq))</span>
<span id="cb66-130"><a href="#cb66-130" aria-hidden="true" tabindex="-1"></a>max_read_ps <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">readcount</span>(physeq))</span>
<span id="cb66-131"><a href="#cb66-131" aria-hidden="true" tabindex="-1"></a>total_reads_ps <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">readcount</span>(physeq))</span>
<span id="cb66-132"><a href="#cb66-132" aria-hidden="true" tabindex="-1"></a>mean_reads_ps <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">mean</span>(<span class="fu">readcount</span>(physeq)), <span class="at">digits =</span> <span class="dv">0</span>)</span>
<span id="cb66-133"><a href="#cb66-133" aria-hidden="true" tabindex="-1"></a>median_reads_ps <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">readcount</span>(physeq))</span>
<span id="cb66-134"><a href="#cb66-134" aria-hidden="true" tabindex="-1"></a>total_asvs_ps <span class="ot">&lt;-</span> <span class="fu">ntaxa</span>(physeq)</span>
<span id="cb66-135"><a href="#cb66-135" aria-hidden="true" tabindex="-1"></a>singleton_ps <span class="ot">&lt;-</span> <span class="fu">tryCatch</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(physeq, <span class="at">detection =</span> <span class="dv">1</span>, </span>
<span id="cb66-136"><a href="#cb66-136" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">prevalence =</span> <span class="dv">0</span>)), </span>
<span id="cb66-137"><a href="#cb66-137" aria-hidden="true" tabindex="-1"></a>                         <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb66-138"><a href="#cb66-138" aria-hidden="true" tabindex="-1"></a>singleton_ps_perc <span class="ot">&lt;-</span> </span>
<span id="cb66-139"><a href="#cb66-139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>(<span class="fu">round</span>((<span class="dv">100</span><span class="sc">*</span>(<span class="fu">ntaxa</span>(<span class="fu">rare</span>(physeq, <span class="at">detection =</span> <span class="dv">1</span>, <span class="at">prevalence =</span> <span class="dv">0</span>)) <span class="sc">/</span></span>
<span id="cb66-140"><a href="#cb66-140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ntaxa</span>(physeq))), <span class="at">digits =</span> <span class="dv">3</span>), <span class="at">error=</span><span class="cf">function</span>(err) <span class="cn">NA</span>)</span>
<span id="cb66-141"><a href="#cb66-141" aria-hidden="true" tabindex="-1"></a>sparsity_ps <span class="ot">&lt;-</span> </span>
<span id="cb66-142"><a href="#cb66-142" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(<span class="fu">length</span>(<span class="fu">which</span>(<span class="fu">abundances</span>(physeq) <span class="sc">==</span> <span class="dv">0</span>))<span class="sc">/</span></span>
<span id="cb66-143"><a href="#cb66-143" aria-hidden="true" tabindex="-1"></a>          <span class="fu">length</span>(<span class="fu">abundances</span>(physeq)), <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb66-144"><a href="#cb66-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-145"><a href="#cb66-145" aria-hidden="true" tabindex="-1"></a>ds_metrics <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rbind</span>(min_read_ps, max_read_ps, total_reads_ps, </span>
<span id="cb66-146"><a href="#cb66-146" aria-hidden="true" tabindex="-1"></a>                               mean_reads_ps, median_reads_ps, total_asvs_ps, </span>
<span id="cb66-147"><a href="#cb66-147" aria-hidden="true" tabindex="-1"></a>                               singleton_ps, singleton_ps_perc, sparsity_ps))</span>
<span id="cb66-148"><a href="#cb66-148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-149"><a href="#cb66-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-150"><a href="#cb66-150" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Metric                              <span class="pp">|</span> Results                             <span class="pp">|</span></span>
<span id="cb66-151"><a href="#cb66-151" aria-hidden="true" tabindex="-1"></a><span class="pp">|-------------------------------------|-------------------------------------|</span></span>
<span id="cb66-152"><a href="#cb66-152" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Min. number of reads                <span class="pp">|</span> <span class="in">`r ds_metrics[1, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-153"><a href="#cb66-153" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Max. number of reads                <span class="pp">|</span> <span class="in">`r ds_metrics[2, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-154"><a href="#cb66-154" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Total number of reads               <span class="pp">|</span> <span class="in">`r ds_metrics[3, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-155"><a href="#cb66-155" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Average number of reads             <span class="pp">|</span> <span class="in">`r ds_metrics[4, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-156"><a href="#cb66-156" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Median number of reads              <span class="pp">|</span> <span class="in">`r ds_metrics[5, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-157"><a href="#cb66-157" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Total ASVs                          <span class="pp">|</span> <span class="in">`r ds_metrics[6, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-158"><a href="#cb66-158" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Number of singleton ASVs            <span class="pp">|</span> <span class="in">`r ds_metrics[7, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-159"><a href="#cb66-159" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Any ASVs sum to 1 or less?          <span class="pp">|</span> <span class="in">`r isTRUE(ds_metrics[7, ] &gt;= 1)`</span>    <span class="pp">|</span></span>
<span id="cb66-160"><a href="#cb66-160" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Percent of ASVs that are singletons <span class="pp">|</span> <span class="in">`r ds_metrics[8, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-161"><a href="#cb66-161" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Sparsity                            <span class="pp">|</span> <span class="in">`r ds_metrics[9, ]`</span>                 <span class="pp">|</span></span>
<span id="cb66-162"><a href="#cb66-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-163"><a href="#cb66-163" aria-hidden="true" tabindex="-1"></a>In this next part of the workflow the goal is to create a *microtable object* using the R package <span class="co">[</span><span class="ot">microeco</span><span class="co">](https://chiliubio.github.io/microeco_tutorial/)</span> <span class="co">[</span><span class="ot">@liu2021microeco</span><span class="co">]</span>. The microtable will be used to store the ASV by sample data as well the taxonomic, fasta, and sample data in a single object. More on that in a moment.</span>
<span id="cb66-164"><a href="#cb66-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-165"><a href="#cb66-165" aria-hidden="true" tabindex="-1"></a>We will also:</span>
<span id="cb66-166"><a href="#cb66-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-167"><a href="#cb66-167" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove any ASVs without kingdom level classification.\</span>
<span id="cb66-168"><a href="#cb66-168" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Revome any contaminants (chloroplast, mitochondria, etc.).\</span>
<span id="cb66-169"><a href="#cb66-169" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove Negative Control (NC) samples.\</span>
<span id="cb66-170"><a href="#cb66-170" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove any low-count samples.</span>
<span id="cb66-171"><a href="#cb66-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-172"><a href="#cb66-172" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sample Data</span></span>
<span id="cb66-173"><a href="#cb66-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-174"><a href="#cb66-174" aria-hidden="true" tabindex="-1"></a>Before we begin, let's create a summary table containing some basic sample metadata and the read count data. We want to inspect how total reads changed through the workflow. Table headers are as follows:</span>
<span id="cb66-175"><a href="#cb66-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-176"><a href="#cb66-176" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Header         <span class="pp">|</span> Description                                             <span class="pp">|</span></span>
<span id="cb66-177"><a href="#cb66-177" aria-hidden="true" tabindex="-1"></a><span class="pp">|----------------|---------------------------------------------------------|</span></span>
<span id="cb66-178"><a href="#cb66-178" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`SampleID`</span>     <span class="pp">|</span> New sample ID based on plot, depth, treatment, &amp; pair   <span class="pp">|</span></span>
<span id="cb66-179"><a href="#cb66-179" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Plot`</span>         <span class="pp">|</span> Experimental plot ID                                    <span class="pp">|</span></span>
<span id="cb66-180"><a href="#cb66-180" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Treatment`</span>    <span class="pp">|</span> Control vs warming                                      <span class="pp">|</span></span>
<span id="cb66-181"><a href="#cb66-181" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Temp`</span>         <span class="pp">|</span> Warming treatment temperature                           <span class="pp">|</span></span>
<span id="cb66-182"><a href="#cb66-182" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Depth`</span>        <span class="pp">|</span> Sampling depth in soil                                  <span class="pp">|</span></span>
<span id="cb66-183"><a href="#cb66-183" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Pairing`</span>      <span class="pp">|</span> Sample pairings                                         <span class="pp">|</span></span>
<span id="cb66-184"><a href="#cb66-184" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`raw_rc`</span>       <span class="pp">|</span> No. of raw reads                                        <span class="pp">|</span></span>
<span id="cb66-185"><a href="#cb66-185" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`cutadapt_rc`</span>  <span class="pp">|</span> Read count after cutdapt                                <span class="pp">|</span></span>
<span id="cb66-186"><a href="#cb66-186" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`lotus3_rc`</span>    <span class="pp">|</span> Read count after LotuS3 pipeline                        <span class="pp">|</span></span>
<span id="cb66-187"><a href="#cb66-187" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`lotus3_asv`</span>   <span class="pp">|</span> No. of ASVs                                             <span class="pp">|</span></span>
<span id="cb66-188"><a href="#cb66-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-189"><a href="#cb66-189" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;</span><span class="kw">br</span><span class="dt">/&gt;</span></span>
<span id="cb66-190"><a href="#cb66-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-191"><a href="#cb66-191" aria-hidden="true" tabindex="-1"></a>Now I do some wrangling of the sample data to format it for downstream analysis. WAhat I want to do is first add read counts from different point in the workflow. </span>
<span id="cb66-192"><a href="#cb66-192" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-195"><a href="#cb66-195" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-196"><a href="#cb66-196" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-197"><a href="#cb66-197" aria-hidden="true" tabindex="-1"></a>tmp_otu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">otu_table</span>(physeq))</span>
<span id="cb66-198"><a href="#cb66-198" aria-hidden="true" tabindex="-1"></a>tmp_otu_sum <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">colSums</span>(tmp_otu <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb66-199"><a href="#cb66-199" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb66-200"><a href="#cb66-200" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tmp_otu_sum)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"lotus3_asv"</span></span>
<span id="cb66-201"><a href="#cb66-201" aria-hidden="true" tabindex="-1"></a>tmp_read_sum <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">colSums</span>(tmp_otu)) <span class="sc">%&gt;%</span></span>
<span id="cb66-202"><a href="#cb66-202" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb66-203"><a href="#cb66-203" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tmp_read_sum)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"lotus3_rc"</span></span>
<span id="cb66-204"><a href="#cb66-204" aria-hidden="true" tabindex="-1"></a>tmp_track_reads <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb66-205"><a href="#cb66-205" aria-hidden="true" tabindex="-1"></a>    <span class="st">"files/CUTADAPT/oo_cutadapt_track.txt"</span>,</span>
<span id="cb66-206"><a href="#cb66-206" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb66-207"><a href="#cb66-207" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb66-208"><a href="#cb66-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-209"><a href="#cb66-209" aria-hidden="true" tabindex="-1"></a>tmp_track_reads<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(</span>
<span id="cb66-210"><a href="#cb66-210" aria-hidden="true" tabindex="-1"></a>  tmp_track_reads<span class="sc">$</span>SampleID, <span class="st">"-"</span>, <span class="st">"_"</span>)</span>
<span id="cb66-211"><a href="#cb66-211" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tmp_track_reads)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">"raw_rc"</span></span>
<span id="cb66-212"><a href="#cb66-212" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tmp_track_reads)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">"cutadapt_rc"</span></span>
<span id="cb66-213"><a href="#cb66-213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-214"><a href="#cb66-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-215"><a href="#cb66-215" aria-hidden="true" tabindex="-1"></a>Then I want to merge this data with the sample metadata. </span>
<span id="cb66-216"><a href="#cb66-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-219"><a href="#cb66-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-220"><a href="#cb66-220" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-221"><a href="#cb66-221" aria-hidden="true" tabindex="-1"></a>tmp_samp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(physeq)) <span class="sc">%&gt;%</span> </span>
<span id="cb66-222"><a href="#cb66-222" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb66-223"><a href="#cb66-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-224"><a href="#cb66-224" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_samp, tmp_track_reads, <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-225"><a href="#cb66-225" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_read_sum, <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-226"><a href="#cb66-226" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_otu_sum, <span class="at">by =</span> <span class="st">"SampleID"</span>) </span>
<span id="cb66-227"><a href="#cb66-227" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> sample_tab <span class="sc">%&gt;%</span></span>
<span id="cb66-228"><a href="#cb66-228" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">7</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">8</span>, <span class="dv">16</span>, <span class="dv">17</span>, <span class="dv">18</span>, </span>
<span id="cb66-229"><a href="#cb66-229" aria-hidden="true" tabindex="-1"></a>                <span class="dv">19</span>, <span class="dv">9</span>, <span class="dv">14</span>, <span class="dv">15</span>, <span class="dv">3</span>, <span class="dv">10</span>, <span class="dv">11</span>, <span class="dv">12</span>, <span class="dv">13</span>, <span class="dv">2</span>)</span>
<span id="cb66-230"><a href="#cb66-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-231"><a href="#cb66-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-234"><a href="#cb66-234" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-235"><a href="#cb66-235" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_delim</span>(sample_tab, <span class="st">"files/CURATE/oo_sample_data_full.txt"</span>,</span>
<span id="cb66-236"><a href="#cb66-236" aria-hidden="true" tabindex="-1"></a>    <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb66-237"><a href="#cb66-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-238"><a href="#cb66-238" aria-hidden="true" tabindex="-1"></a>sample_tab_trim <span class="ot">&lt;-</span> sample_tab[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">11</span><span class="sc">:</span><span class="dv">19</span>)]</span>
<span id="cb66-239"><a href="#cb66-239" aria-hidden="true" tabindex="-1"></a>sample_tab_trim<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> </span>
<span id="cb66-240"><a href="#cb66-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">round</span>(sample_tab_trim<span class="sc">$</span>lotus3_rc<span class="sc">/</span>sample_tab_trim<span class="sc">$</span>raw_rc, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb66-241"><a href="#cb66-241" aria-hidden="true" tabindex="-1"></a>sample_tab_trim <span class="ot">&lt;-</span> sample_tab_trim <span class="sc">%&gt;%</span> </span>
<span id="cb66-242"><a href="#cb66-242" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(per_reads_kept, <span class="at">.before =</span> lotus3_asv)</span>
<span id="cb66-243"><a href="#cb66-243" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-244"><a href="#cb66-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-245"><a href="#cb66-245" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;/</span><span class="kw">br</span><span class="dt">&gt;</span></span>
<span id="cb66-246"><a href="#cb66-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-249"><a href="#cb66-249" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-250"><a href="#cb66-250" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-251"><a href="#cb66-251" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-252"><a href="#cb66-252" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-make-metadata-table-oo</span></span>
<span id="cb66-253"><a href="#cb66-253" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Sample metadata including read changes at start and end of workflow."</span></span>
<span id="cb66-254"><a href="#cb66-254" aria-hidden="true" tabindex="-1"></a><span class="fu">make_metadata_table</span>(sample_tab_trim)</span>
<span id="cb66-255"><a href="#cb66-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-256"><a href="#cb66-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-259"><a href="#cb66-259" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-260"><a href="#cb66-260" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-261"><a href="#cb66-261" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-262"><a href="#cb66-262" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_delim</span>(sample_tab_trim, <span class="st">"files/CURATE/oo_sample_data_with_rc.txt"</span>,</span>
<span id="cb66-263"><a href="#cb66-263" aria-hidden="true" tabindex="-1"></a>    <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb66-264"><a href="#cb66-264" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-265"><a href="#cb66-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-266"><a href="#cb66-266" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;/</span><span class="kw">br</span><span class="dt">&gt;</span></span>
<span id="cb66-267"><a href="#cb66-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-268"><a href="#cb66-268" aria-hidden="true" tabindex="-1"></a>{{&lt; downloadthis files/CURATE/oo_sample_data_with_rc.txt dname=oo_sample_data_with_rc label="Download sample metadata" icon=table type=primary class=data-button id=shrimp_md &gt;}}</span>
<span id="cb66-269"><a href="#cb66-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-270"><a href="#cb66-270" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prep Data for microeco</span></span>
<span id="cb66-271"><a href="#cb66-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-272"><a href="#cb66-272" aria-hidden="true" tabindex="-1"></a>Like any tool, the microeco package needs the data in a specific form. I formatted our data to match the mock data in this section,  <span class="co">[</span><span class="ot">microeco</span><span class="co">](https://chiliubio.github.io/microeco_tutorial/basic-class.html#prepare-the-example-data)</span> tutorial.</span>
<span id="cb66-273"><a href="#cb66-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-274"><a href="#cb66-274" aria-hidden="true" tabindex="-1"></a><span class="fu">### A. Taxonomy Table</span></span>
<span id="cb66-275"><a href="#cb66-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-276"><a href="#cb66-276" aria-hidden="true" tabindex="-1"></a>Here is what the taxonomy table looks like in the dataset.</span>
<span id="cb66-277"><a href="#cb66-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-280"><a href="#cb66-280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-281"><a href="#cb66-281" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-282"><a href="#cb66-282" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-283"><a href="#cb66-283" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(physeq))</span>
<span id="cb66-284"><a href="#cb66-284" aria-hidden="true" tabindex="-1"></a>tmp_tax[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb66-285"><a href="#cb66-285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-286"><a href="#cb66-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-287"><a href="#cb66-287" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb66-288"><a href="#cb66-288" aria-hidden="true" tabindex="-1"></a><span class="in">       Domain Phylum     Class          Order</span></span>
<span id="cb66-289"><a href="#cb66-289" aria-hidden="true" tabindex="-1"></a><span class="in">ASV163 Fungi  Ascomycota Eurotiomycetes Chaetothyriales</span></span>
<span id="cb66-290"><a href="#cb66-290" aria-hidden="true" tabindex="-1"></a><span class="in">ASV118 Fungi  Ascomycota Eurotiomycetes Chaetothyriales</span></span>
<span id="cb66-291"><a href="#cb66-291" aria-hidden="true" tabindex="-1"></a><span class="in">ASV86  ?      ?          ?              ?</span></span>
<span id="cb66-292"><a href="#cb66-292" aria-hidden="true" tabindex="-1"></a><span class="in">ASV179 ?      ?          ?              ?</span></span>
<span id="cb66-293"><a href="#cb66-293" aria-hidden="true" tabindex="-1"></a><span class="in">ASV132 Fungi  Ascomycota Eurotiomycetes Chaetothyriales</span></span>
<span id="cb66-294"><a href="#cb66-294" aria-hidden="true" tabindex="-1"></a><span class="in">ASV156 Fungi  Ascomycota Eurotiomycetes Chaetothyriales</span></span>
<span id="cb66-295"><a href="#cb66-295" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-296"><a href="#cb66-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-297"><a href="#cb66-297" aria-hidden="true" tabindex="-1"></a>Next we need to add rank definitions (e.g., <span class="in">`k__`</span>, <span class="in">`p__`</span>, etc.) to each classification. </span>
<span id="cb66-298"><a href="#cb66-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-301"><a href="#cb66-301" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-302"><a href="#cb66-302" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-303"><a href="#cb66-303" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-304"><a href="#cb66-304" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">tax_table</span>(physeq))   </span>
<span id="cb66-305"><a href="#cb66-305" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"Kingdom"</span> <span class="ot">=</span> <span class="st">"Domain"</span>)</span>
<span id="cb66-306"><a href="#cb66-306" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span></span>
<span id="cb66-307"><a href="#cb66-307" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate_all</span>(<span class="sc">~</span>stringr<span class="sc">::</span><span class="fu">str_replace_all</span>(., <span class="st">"</span><span class="sc">\\</span><span class="st">?"</span>, <span class="st">""</span>))</span>
<span id="cb66-308"><a href="#cb66-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-309"><a href="#cb66-309" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Kingdom <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"k"</span>, tmp_tax<span class="sc">$</span>Kingdom, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-310"><a href="#cb66-310" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"p"</span>, tmp_tax<span class="sc">$</span>Phylum, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-311"><a href="#cb66-311" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Class <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"c"</span>, tmp_tax<span class="sc">$</span>Class, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-312"><a href="#cb66-312" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Order <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"o"</span>, tmp_tax<span class="sc">$</span>Order, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-313"><a href="#cb66-313" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Family <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"f"</span>, tmp_tax<span class="sc">$</span>Family, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-314"><a href="#cb66-314" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Genus <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"g"</span>, tmp_tax<span class="sc">$</span>Genus, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-315"><a href="#cb66-315" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Species <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">"s"</span>, tmp_tax<span class="sc">$</span>Species, <span class="at">sep =</span> <span class="st">"__"</span>)</span>
<span id="cb66-316"><a href="#cb66-316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-317"><a href="#cb66-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-318"><a href="#cb66-318" aria-hidden="true" tabindex="-1"></a>And now the final, modified taxonomy table.</span>
<span id="cb66-319"><a href="#cb66-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-322"><a href="#cb66-322" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-323"><a href="#cb66-323" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-324"><a href="#cb66-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-325"><a href="#cb66-325" aria-hidden="true" tabindex="-1"></a>tmp_tax[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span>
<span id="cb66-326"><a href="#cb66-326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-327"><a href="#cb66-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-328"><a href="#cb66-328" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-329"><a href="#cb66-329" aria-hidden="true" tabindex="-1"></a><span class="in">       Kingdom  Phylum        Class             Order</span></span>
<span id="cb66-330"><a href="#cb66-330" aria-hidden="true" tabindex="-1"></a><span class="in">ASV163 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales</span></span>
<span id="cb66-331"><a href="#cb66-331" aria-hidden="true" tabindex="-1"></a><span class="in">ASV118 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales</span></span>
<span id="cb66-332"><a href="#cb66-332" aria-hidden="true" tabindex="-1"></a><span class="in">ASV86  k__      p__           c__               o__</span></span>
<span id="cb66-333"><a href="#cb66-333" aria-hidden="true" tabindex="-1"></a><span class="in">ASV179 k__      p__           c__               o__</span></span>
<span id="cb66-334"><a href="#cb66-334" aria-hidden="true" tabindex="-1"></a><span class="in">ASV132 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales</span></span>
<span id="cb66-335"><a href="#cb66-335" aria-hidden="true" tabindex="-1"></a><span class="in">ASV156 k__Fungi p__Ascomycota c__Eurotiomycetes o__Chaetothyriales</span></span>
<span id="cb66-336"><a href="#cb66-336" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-337"><a href="#cb66-337" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-338"><a href="#cb66-338" aria-hidden="true" tabindex="-1"></a><span class="fu">### B. Sequence Table</span></span>
<span id="cb66-339"><a href="#cb66-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-340"><a href="#cb66-340" aria-hidden="true" tabindex="-1"></a>Here is what the sequence table looks like in the dataset.</span>
<span id="cb66-341"><a href="#cb66-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-344"><a href="#cb66-344" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-345"><a href="#cb66-345" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-346"><a href="#cb66-346" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-347"><a href="#cb66-347" aria-hidden="true" tabindex="-1"></a>tmp_otu[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb66-348"><a href="#cb66-348" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-349"><a href="#cb66-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-350"><a href="#cb66-350" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-351"><a href="#cb66-351" aria-hidden="true" tabindex="-1"></a><span class="in">       P00_D00_000_NNN P01_D00_010_W4A P01_D10_020_W4A</span></span>
<span id="cb66-352"><a href="#cb66-352" aria-hidden="true" tabindex="-1"></a><span class="in">ASV163               0               0               0 </span></span>
<span id="cb66-353"><a href="#cb66-353" aria-hidden="true" tabindex="-1"></a><span class="in">ASV118               0               0               0 </span></span>
<span id="cb66-354"><a href="#cb66-354" aria-hidden="true" tabindex="-1"></a><span class="in">ASV86                0               0               0 </span></span>
<span id="cb66-355"><a href="#cb66-355" aria-hidden="true" tabindex="-1"></a><span class="in">ASV179               0               0               0 </span></span>
<span id="cb66-356"><a href="#cb66-356" aria-hidden="true" tabindex="-1"></a><span class="in">ASV132               0               0               0 </span></span>
<span id="cb66-357"><a href="#cb66-357" aria-hidden="true" tabindex="-1"></a><span class="in">ASV156               0               0               0 </span></span>
<span id="cb66-358"><a href="#cb66-358" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-359"><a href="#cb66-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-362"><a href="#cb66-362" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-363"><a href="#cb66-363" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-364"><a href="#cb66-364" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-365"><a href="#cb66-365" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">row.names</span>(tmp_otu), <span class="fu">row.names</span>(tmp_tax))</span>
<span id="cb66-366"><a href="#cb66-366" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-367"><a href="#cb66-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-368"><a href="#cb66-368" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-369"><a href="#cb66-369" aria-hidden="true" tabindex="-1"></a><span class="in">[1] TRUE</span></span>
<span id="cb66-370"><a href="#cb66-370" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-371"><a href="#cb66-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-372"><a href="#cb66-372" aria-hidden="true" tabindex="-1"></a><span class="fu">### C. Sample Table</span></span>
<span id="cb66-373"><a href="#cb66-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-374"><a href="#cb66-374" aria-hidden="true" tabindex="-1"></a>Here is what the sample table looks like.</span>
<span id="cb66-375"><a href="#cb66-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-378"><a href="#cb66-378" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-379"><a href="#cb66-379" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-380"><a href="#cb66-380" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-381"><a href="#cb66-381" aria-hidden="true" tabindex="-1"></a>sample_tab[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb66-382"><a href="#cb66-382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-383"><a href="#cb66-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-386"><a href="#cb66-386" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-387"><a href="#cb66-387" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-388"><a href="#cb66-388" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-389"><a href="#cb66-389" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> sample_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb66-390"><a href="#cb66-390" aria-hidden="true" tabindex="-1"></a>sample_tab<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sample_tab)</span>
<span id="cb66-391"><a href="#cb66-391" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> sample_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(SampleID)</span>
<span id="cb66-392"><a href="#cb66-392" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-393"><a href="#cb66-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-394"><a href="#cb66-394" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create a Microtable Object</span></span>
<span id="cb66-395"><a href="#cb66-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-396"><a href="#cb66-396" aria-hidden="true" tabindex="-1"></a>With these three files in hand we are now ready to create a <span class="co">[</span><span class="ot">microtable object</span><span class="co">](https://chiliubio.github.io/microeco_tutorial/basic-class.html)</span>.</span>
<span id="cb66-397"><a href="#cb66-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-398"><a href="#cb66-398" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb66-399"><a href="#cb66-399" aria-hidden="true" tabindex="-1"></a>A <span class="co">[</span><span class="ot">microtable object</span><span class="co">](https://chiliubio.github.io/microeco_tutorial/basic-class.html)</span> contains an ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications). It can also contain a phylogenetic tree of ASVs as well as representative sequences of each ASV. </span>
<span id="cb66-400"><a href="#cb66-400" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb66-401"><a href="#cb66-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-404"><a href="#cb66-404" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-405"><a href="#cb66-405" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-406"><a href="#cb66-406" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-407"><a href="#cb66-407" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> sample_tab</span>
<span id="cb66-408"><a href="#cb66-408" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> tmp_tax</span>
<span id="cb66-409"><a href="#cb66-409" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> tmp_otu</span>
<span id="cb66-410"><a href="#cb66-410" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-411"><a href="#cb66-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-414"><a href="#cb66-414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-415"><a href="#cb66-415" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-416"><a href="#cb66-416" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-417"><a href="#cb66-417" aria-hidden="true" tabindex="-1"></a>tmp_me <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> sample_tab, </span>
<span id="cb66-418"><a href="#cb66-418" aria-hidden="true" tabindex="-1"></a>                         <span class="at">otu_table =</span> otu_tab, </span>
<span id="cb66-419"><a href="#cb66-419" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tax_table =</span> tax_tab)</span>
<span id="cb66-420"><a href="#cb66-420" aria-hidden="true" tabindex="-1"></a>tmp_me</span>
<span id="cb66-421"><a href="#cb66-421" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-422"><a href="#cb66-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-423"><a href="#cb66-423" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb66-424"><a href="#cb66-424" aria-hidden="true" tabindex="-1"></a><span class="in">2 samples with 0 abundance are removed from the otu_table ...</span></span>
<span id="cb66-425"><a href="#cb66-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-426"><a href="#cb66-426" aria-hidden="true" tabindex="-1"></a><span class="in">microtable-class object:</span></span>
<span id="cb66-427"><a href="#cb66-427" aria-hidden="true" tabindex="-1"></a><span class="in">sample_table have 41 rows and 11 columns</span></span>
<span id="cb66-428"><a href="#cb66-428" aria-hidden="true" tabindex="-1"></a><span class="in">otu_table have 12091 rows and 41 columns</span></span>
<span id="cb66-429"><a href="#cb66-429" aria-hidden="true" tabindex="-1"></a><span class="in">tax_table have 12091 rows and 7 columns</span></span>
<span id="cb66-430"><a href="#cb66-430" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-431"><a href="#cb66-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-432"><a href="#cb66-432" aria-hidden="true" tabindex="-1"></a><span class="fu">### D. Add Representative Sequence</span></span>
<span id="cb66-433"><a href="#cb66-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-434"><a href="#cb66-434" aria-hidden="true" tabindex="-1"></a>We can also add representative sequences for each OTU/ASV. For this step, we can simply grab the sequences from the row names of the DADA2 taxonomy object loaded above.</span>
<span id="cb66-435"><a href="#cb66-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-438"><a href="#cb66-438" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-439"><a href="#cb66-439" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-440"><a href="#cb66-440" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-441"><a href="#cb66-441" aria-hidden="true" tabindex="-1"></a>rep_fasta <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(<span class="st">"working_files/LOTUS3/oo/OTU.fna"</span>)</span>
<span id="cb66-442"><a href="#cb66-442" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">row.names</span>(tmp_me<span class="sc">$</span>tax_table), <span class="fu">row.names</span>(tmp_me<span class="sc">$</span>otu_table))</span>
<span id="cb66-443"><a href="#cb66-443" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-444"><a href="#cb66-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-445"><a href="#cb66-445" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-446"><a href="#cb66-446" aria-hidden="true" tabindex="-1"></a><span class="in">[1] TRUE</span></span>
<span id="cb66-447"><a href="#cb66-447" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-448"><a href="#cb66-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-451"><a href="#cb66-451" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-452"><a href="#cb66-452" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-453"><a href="#cb66-453" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-454"><a href="#cb66-454" aria-hidden="true" tabindex="-1"></a>tmp_fasta_names <span class="ot">&lt;-</span> <span class="fu">names</span>(rep_fasta)</span>
<span id="cb66-455"><a href="#cb66-455" aria-hidden="true" tabindex="-1"></a>tmp_rep_fasta <span class="ot">&lt;-</span> rep_fasta[<span class="fu">match</span>(<span class="fu">row.names</span>(tmp_me<span class="sc">$</span>tax_table), tmp_fasta_names)]</span>
<span id="cb66-456"><a href="#cb66-456" aria-hidden="true" tabindex="-1"></a>rep_fasta <span class="ot">&lt;-</span> tmp_rep_fasta</span>
<span id="cb66-457"><a href="#cb66-457" aria-hidden="true" tabindex="-1"></a>tmp_me<span class="sc">$</span>rep_fasta <span class="ot">&lt;-</span> rep_fasta</span>
<span id="cb66-458"><a href="#cb66-458" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">row.names</span>(tmp_me<span class="sc">$</span>tax_table), <span class="fu">names</span>(rep_fasta))</span>
<span id="cb66-459"><a href="#cb66-459" aria-hidden="true" tabindex="-1"></a>tmp_me<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-460"><a href="#cb66-460" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-461"><a href="#cb66-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-462"><a href="#cb66-462" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-463"><a href="#cb66-463" aria-hidden="true" tabindex="-1"></a><span class="in">[1] TRUE</span></span>
<span id="cb66-464"><a href="#cb66-464" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-465"><a href="#cb66-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-468"><a href="#cb66-468" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-469"><a href="#cb66-469" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-470"><a href="#cb66-470" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-471"><a href="#cb66-471" aria-hidden="true" tabindex="-1"></a>me_asv_raw <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_me)</span>
<span id="cb66-472"><a href="#cb66-472" aria-hidden="true" tabindex="-1"></a>me_asv_raw</span>
<span id="cb66-473"><a href="#cb66-473" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-474"><a href="#cb66-474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-477"><a href="#cb66-477" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-478"><a href="#cb66-478" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-479"><a href="#cb66-479" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-480"><a href="#cb66-480" aria-hidden="true" tabindex="-1"></a>me_asv_raw</span>
<span id="cb66-481"><a href="#cb66-481" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-482"><a href="#cb66-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-483"><a href="#cb66-483" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The microeco object </span><span class="in">`me_asv_raw`</span><span class="at"> contains all data from the LotuS3 pipeline, before any dataset curation. </span></span>
<span id="cb66-484"><a href="#cb66-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-485"><a href="#cb66-485" aria-hidden="true" tabindex="-1"></a><span class="fu">## Curate the Data Set</span></span>
<span id="cb66-486"><a href="#cb66-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-487"><a href="#cb66-487" aria-hidden="true" tabindex="-1"></a>Pretty much the last thing to do is remove unwanted taxa, negative controls, and low-count samples.</span>
<span id="cb66-488"><a href="#cb66-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-489"><a href="#cb66-489" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove non-fungal ASVs</span></span>
<span id="cb66-490"><a href="#cb66-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-491"><a href="#cb66-491" aria-hidden="true" tabindex="-1"></a>First we can have a quick look at the kingdom  level classifications to see what we are dealing with. </span>
<span id="cb66-492"><a href="#cb66-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-495"><a href="#cb66-495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-496"><a href="#cb66-496" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-497"><a href="#cb66-497" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-498"><a href="#cb66-498" aria-hidden="true" tabindex="-1"></a>tmp_k <span class="ot">&lt;-</span> me_asv_raw<span class="sc">$</span>tax_table <span class="sc">%&gt;%</span></span>
<span id="cb66-499"><a href="#cb66-499" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(Kingdom, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-500"><a href="#cb66-500" aria-hidden="true" tabindex="-1"></a>tmp_p <span class="ot">&lt;-</span> me_asv_raw<span class="sc">$</span>tax_table <span class="sc">%&gt;%</span></span>
<span id="cb66-501"><a href="#cb66-501" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(Phylum, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-502"><a href="#cb66-502" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-503"><a href="#cb66-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-506"><a href="#cb66-506" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-507"><a href="#cb66-507" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-508"><a href="#cb66-508" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-509"><a href="#cb66-509" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: true</span></span>
<span id="cb66-510"><a href="#cb66-510" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: markup</span></span>
<span id="cb66-511"><a href="#cb66-511" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"-------KINGDOM------------------"</span>)</span>
<span id="cb66-512"><a href="#cb66-512" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(tmp_k<span class="sc">$</span>Kingdom, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-513"><a href="#cb66-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-514"><a href="#cb66-514" aria-hidden="true" tabindex="-1"></a><span class="fu">message</span>(<span class="st">"-------PHYLUM------------------"</span>)</span>
<span id="cb66-515"><a href="#cb66-515" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(tmp_p<span class="sc">$</span>Phylum, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb66-516"><a href="#cb66-516" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-517"><a href="#cb66-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-518"><a href="#cb66-518" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>The first thing to notice is that at the kingdom level we have Fungi, unclassified (or NA), Stramenopila, some non-fungal classifications. We can remove anything that is not classified as Fungi at the kingdom level or Oomycota at the phylum level.  </span>
<span id="cb66-519"><a href="#cb66-519" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-520"><a href="#cb66-520" aria-hidden="true" tabindex="-1"></a>The easiest thing to do is keep everything we want--in other words, all fungal taxa.</span>
<span id="cb66-521"><a href="#cb66-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-524"><a href="#cb66-524" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-525"><a href="#cb66-525" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-526"><a href="#cb66-526" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-527"><a href="#cb66-527" aria-hidden="true" tabindex="-1"></a>tmp_no_na <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_me)</span>
<span id="cb66-528"><a href="#cb66-528" aria-hidden="true" tabindex="-1"></a>tmp_no_na<span class="sc">$</span>tax_table <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb66-529"><a href="#cb66-529" aria-hidden="true" tabindex="-1"></a>  base<span class="sc">::</span><span class="fu">subset</span>(Kingdom <span class="sc">==</span> <span class="st">"k__Fungi"</span> <span class="sc">|</span> Phylum <span class="sc">==</span> <span class="st">"p__Oomycota"</span>)</span>
<span id="cb66-530"><a href="#cb66-530" aria-hidden="true" tabindex="-1"></a>tmp_no_na<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-531"><a href="#cb66-531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-532"><a href="#cb66-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-535"><a href="#cb66-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-536"><a href="#cb66-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-537"><a href="#cb66-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-538"><a href="#cb66-538" aria-hidden="true" tabindex="-1"></a>me_asv_no_na <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_na)</span>
<span id="cb66-539"><a href="#cb66-539" aria-hidden="true" tabindex="-1"></a>me_asv_no_na</span>
<span id="cb66-540"><a href="#cb66-540" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-541"><a href="#cb66-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-544"><a href="#cb66-544" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-545"><a href="#cb66-545" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-546"><a href="#cb66-546" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-547"><a href="#cb66-547" aria-hidden="true" tabindex="-1"></a>me_asv_no_na</span>
<span id="cb66-548"><a href="#cb66-548" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-549"><a href="#cb66-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-550"><a href="#cb66-550" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The microeco object </span><span class="in">`me_asv_no_na`</span><span class="at"> conatins only ASVs classified as Archaea or Bacteria. </span></span>
<span id="cb66-551"><a href="#cb66-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-552"><a href="#cb66-552" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Negative Controls (NC)</span></span>
<span id="cb66-553"><a href="#cb66-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-554"><a href="#cb66-554" aria-hidden="true" tabindex="-1"></a>Now we need to remove the NC samples *and* ASVs found in those sample. We first identified all ASVs that were present in at least one NC sample represented by at least 1 read. We did this by subsetting the NC samples from the new microtable object.</span>
<span id="cb66-555"><a href="#cb66-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-558"><a href="#cb66-558" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-559"><a href="#cb66-559" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-560"><a href="#cb66-560" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-561"><a href="#cb66-561" aria-hidden="true" tabindex="-1"></a>tmp_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_na)</span>
<span id="cb66-562"><a href="#cb66-562" aria-hidden="true" tabindex="-1"></a>tmp_nc<span class="sc">$</span>sample_table <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_nc<span class="sc">$</span>sample_table, Treatment <span class="sc">==</span> <span class="st">"Negative"</span>)</span>
<span id="cb66-563"><a href="#cb66-563" aria-hidden="true" tabindex="-1"></a>tmp_nc<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-564"><a href="#cb66-564" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-565"><a href="#cb66-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-566"><a href="#cb66-566" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-567"><a href="#cb66-567" aria-hidden="true" tabindex="-1"></a><span class="in">89 taxa with 0 abundance are removed from the otu_table ...</span></span>
<span id="cb66-568"><a href="#cb66-568" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-569"><a href="#cb66-569" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-572"><a href="#cb66-572" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-573"><a href="#cb66-573" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-574"><a href="#cb66-574" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-575"><a href="#cb66-575" aria-hidden="true" tabindex="-1"></a>me_asv_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_nc)</span>
<span id="cb66-576"><a href="#cb66-576" aria-hidden="true" tabindex="-1"></a>me_asv_nc</span>
<span id="cb66-577"><a href="#cb66-577" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-578"><a href="#cb66-578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-581"><a href="#cb66-581" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-582"><a href="#cb66-582" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-583"><a href="#cb66-583" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-584"><a href="#cb66-584" aria-hidden="true" tabindex="-1"></a>me_asv_nc</span>
<span id="cb66-585"><a href="#cb66-585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-586"><a href="#cb66-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-587"><a href="#cb66-587" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The microeco object </span><span class="in">`me_asv_nc`</span><span class="at"> conatins only the negative control sample(s) and associated ASVs. </span></span>
<span id="cb66-588"><a href="#cb66-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-589"><a href="#cb66-589" aria-hidden="true" tabindex="-1"></a>Looks like there are <span class="in">`r nrow(me_asv_nc$tax_table)`</span> ASVs in the NC samples from a total of <span class="in">`r sum(me_asv_nc$taxa_sums())`</span> reads.</span>
<span id="cb66-590"><a href="#cb66-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-593"><a href="#cb66-593" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-594"><a href="#cb66-594" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-595"><a href="#cb66-595" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-596"><a href="#cb66-596" aria-hidden="true" tabindex="-1"></a>nc_asvs <span class="ot">&lt;-</span> <span class="fu">row.names</span>(tmp_nc<span class="sc">$</span>tax_table)</span>
<span id="cb66-597"><a href="#cb66-597" aria-hidden="true" tabindex="-1"></a>nc_asvs</span>
<span id="cb66-598"><a href="#cb66-598" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-599"><a href="#cb66-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-602"><a href="#cb66-602" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-603"><a href="#cb66-603" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-604"><a href="#cb66-604" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-605"><a href="#cb66-605" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(nc_asvs)</span>
<span id="cb66-606"><a href="#cb66-606" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-607"><a href="#cb66-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-608"><a href="#cb66-608" aria-hidden="true" tabindex="-1"></a>There are <span class="in">`r length(nc_asvs)`</span> ASVs found in the NC sample. </span>
<span id="cb66-609"><a href="#cb66-609" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-610"><a href="#cb66-610" aria-hidden="true" tabindex="-1"></a>Essentially, for each ASV, the code below calculates:</span>
<span id="cb66-611"><a href="#cb66-611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-612"><a href="#cb66-612" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of NC samples containing at least 1 read.\</span>
<span id="cb66-613"><a href="#cb66-613" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of reads in NC samples.\</span>
<span id="cb66-614"><a href="#cb66-614" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of non-NC samples containing at least 1 read.\</span>
<span id="cb66-615"><a href="#cb66-615" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of reads in non-NC samples.\</span>
<span id="cb66-616"><a href="#cb66-616" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The percent of reads in the NC samples and the percent of NC samples containing reads.</span>
<span id="cb66-617"><a href="#cb66-617" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-620"><a href="#cb66-620" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-621"><a href="#cb66-621" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-622"><a href="#cb66-622" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-623"><a href="#cb66-623" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_na)</span>
<span id="cb66-624"><a href="#cb66-624" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc<span class="sc">$</span>otu_table</span>
<span id="cb66-625"><a href="#cb66-625" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span> </span>
<span id="cb66-626"><a href="#cb66-626" aria-hidden="true" tabindex="-1"></a>                 dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">row.names</span>(tmp_rem_nc_df) <span class="sc">%in%</span> nc_asvs)</span>
<span id="cb66-627"><a href="#cb66-627" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb66-628"><a href="#cb66-628" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-629"><a href="#cb66-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-632"><a href="#cb66-632" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-633"><a href="#cb66-633" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-634"><a href="#cb66-634" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-635"><a href="#cb66-635" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df  <span class="sc">%&gt;%</span> </span>
<span id="cb66-636"><a href="#cb66-636" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">total_reads_NC =</span> <span class="fu">rowSums</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(., <span class="fu">contains</span>(<span class="st">"NNN"</span>))), </span>
<span id="cb66-637"><a href="#cb66-637" aria-hidden="true" tabindex="-1"></a>         <span class="at">.after =</span> <span class="st">"ASV_ID"</span>)</span>
<span id="cb66-638"><a href="#cb66-638" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_rem_nc_df, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"NNN"</span>))</span>
<span id="cb66-639"><a href="#cb66-639" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span></span>
<span id="cb66-640"><a href="#cb66-640" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">total_reads_samps =</span> <span class="fu">rowSums</span>(.[<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_rem_nc_df)]), </span>
<span id="cb66-641"><a href="#cb66-641" aria-hidden="true" tabindex="-1"></a>                <span class="at">.after =</span> <span class="st">"total_reads_NC"</span>)</span>
<span id="cb66-642"><a href="#cb66-642" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_rem_nc_df)] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb66-643"><a href="#cb66-643" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span></span>
<span id="cb66-644"><a href="#cb66-644" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">perc_in_neg =</span> <span class="dv">100</span><span class="sc">*</span>(</span>
<span id="cb66-645"><a href="#cb66-645" aria-hidden="true" tabindex="-1"></a>    total_reads_NC <span class="sc">/</span> (total_reads_NC <span class="sc">+</span> total_reads_samps)),</span>
<span id="cb66-646"><a href="#cb66-646" aria-hidden="true" tabindex="-1"></a>                <span class="at">.after =</span> <span class="st">"total_reads_samps"</span>)</span>
<span id="cb66-647"><a href="#cb66-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-648"><a href="#cb66-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-651"><a href="#cb66-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-652"><a href="#cb66-652" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-653"><a href="#cb66-653" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-654"><a href="#cb66-654" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df<span class="sc">$</span>perc_in_neg <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_rem_nc_df<span class="sc">$</span>perc_in_neg, <span class="at">digits =</span> <span class="dv">6</span>)</span>
<span id="cb66-655"><a href="#cb66-655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-656"><a href="#cb66-656" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(tmp_rem_nc<span class="sc">$</span>otu_table <span class="sc">!=</span> <span class="dv">0</span>))</span>
<span id="cb66-657"><a href="#cb66-657" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> tmp_1 <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb66-658"><a href="#cb66-658" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> tmp_1 <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"total_samples"</span> <span class="ot">=</span> <span class="dv">2</span>)  </span>
<span id="cb66-659"><a href="#cb66-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-660"><a href="#cb66-660" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_rem_nc<span class="sc">$</span>otu_table, <span class="fu">contains</span>(<span class="st">"NNN"</span>))</span>
<span id="cb66-661"><a href="#cb66-661" aria-hidden="true" tabindex="-1"></a>tmp_2<span class="sc">$</span>num_samp_nc <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tmp_2 <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb66-662"><a href="#cb66-662" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_2, <span class="fu">contains</span>(<span class="st">"num_samp_nc"</span>))</span>
<span id="cb66-663"><a href="#cb66-663" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> tmp_2 <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb66-664"><a href="#cb66-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-665"><a href="#cb66-665" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_rem_nc<span class="sc">$</span>otu_table, <span class="sc">-</span><span class="fu">contains</span>(<span class="st">"NNN"</span>))</span>
<span id="cb66-666"><a href="#cb66-666" aria-hidden="true" tabindex="-1"></a>tmp_3<span class="sc">$</span>num_samp_no_nc <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tmp_3 <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb66-667"><a href="#cb66-667" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_3, <span class="fu">contains</span>(<span class="st">"num_samp_no_nc"</span>))</span>
<span id="cb66-668"><a href="#cb66-668" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> tmp_3 <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb66-669"><a href="#cb66-669" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-670"><a href="#cb66-670" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_rem_nc_df, tmp_1) <span class="sc">%&gt;%</span></span>
<span id="cb66-671"><a href="#cb66-671" aria-hidden="true" tabindex="-1"></a>                 dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_2) <span class="sc">%&gt;%</span></span>
<span id="cb66-672"><a href="#cb66-672" aria-hidden="true" tabindex="-1"></a>                 dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_3)</span>
<span id="cb66-673"><a href="#cb66-673" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-674"><a href="#cb66-674" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span></span>
<span id="cb66-675"><a href="#cb66-675" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb66-676"><a href="#cb66-676" aria-hidden="true" tabindex="-1"></a>    <span class="at">perc_in_neg_samp =</span> <span class="dv">100</span><span class="sc">*</span>( num_samp_nc <span class="sc">/</span> (num_samp_nc <span class="sc">+</span> num_samp_no_nc)),</span>
<span id="cb66-677"><a href="#cb66-677" aria-hidden="true" tabindex="-1"></a>    <span class="at">.after =</span> <span class="st">"num_samp_no_nc"</span>)</span>
<span id="cb66-678"><a href="#cb66-678" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-679"><a href="#cb66-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-682"><a href="#cb66-682" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-683"><a href="#cb66-683" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-684"><a href="#cb66-684" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-685"><a href="#cb66-685" aria-hidden="true" tabindex="-1"></a>nc_check <span class="ot">&lt;-</span> tmp_rem_nc_df</span>
<span id="cb66-686"><a href="#cb66-686" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-687"><a href="#cb66-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-690"><a href="#cb66-690" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-691"><a href="#cb66-691" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-692"><a href="#cb66-692" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-693"><a href="#cb66-693" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-make-asv-in-nc-table-oo</span></span>
<span id="cb66-694"><a href="#cb66-694" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Summary of ASVs detected in Negative Control (NC) samples."</span></span>
<span id="cb66-695"><a href="#cb66-695" aria-hidden="true" tabindex="-1"></a><span class="fu">make_asv_in_nc_table</span>(nc_check)</span>
<span id="cb66-696"><a href="#cb66-696" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-697"><a href="#cb66-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-700"><a href="#cb66-700" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-701"><a href="#cb66-701" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-702"><a href="#cb66-702" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-703"><a href="#cb66-703" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(nc_check, <span class="st">"files/CURATE/oo_asv_in_nc_samples.txt"</span>,</span>
<span id="cb66-704"><a href="#cb66-704" aria-hidden="true" tabindex="-1"></a>    <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb66-705"><a href="#cb66-705" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-706"><a href="#cb66-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-707"><a href="#cb66-707" aria-hidden="true" tabindex="-1"></a>{{&lt; downloadthis files/CURATE/oo_asv_in_nc_samples.txt dname=oo_asv_in_nc_samples label="Download ASVs in NC samples" icon=table type=primary class=data-button id=asv_in_nc_samples &gt;}}</span>
<span id="cb66-708"><a href="#cb66-708" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-709"><a href="#cb66-709" aria-hidden="true" tabindex="-1"></a>Looking at these data we can see the NC sample only has 1 read from each of the two ASVs. We can simply ignore these ASVs and just remove the NC sample. We will still go through the process...</span>
<span id="cb66-710"><a href="#cb66-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-713"><a href="#cb66-713" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-714"><a href="#cb66-714" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-715"><a href="#cb66-715" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-716"><a href="#cb66-716" aria-hidden="true" tabindex="-1"></a>nc_remove <span class="ot">&lt;-</span> nc_check <span class="sc">%&gt;%</span> </span>
<span id="cb66-717"><a href="#cb66-717" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(perc_in_neg <span class="sc">&gt;</span> <span class="dv">10</span> <span class="sc">|</span> num_samp_no_nc <span class="sc">&lt;</span> <span class="dv">4</span>)</span>
<span id="cb66-718"><a href="#cb66-718" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-719"><a href="#cb66-719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-722"><a href="#cb66-722" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-723"><a href="#cb66-723" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-724"><a href="#cb66-724" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-725"><a href="#cb66-725" aria-hidden="true" tabindex="-1"></a>nc_remain <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">anti_join</span>(nc_check, nc_remove)</span>
<span id="cb66-726"><a href="#cb66-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-727"><a href="#cb66-727" aria-hidden="true" tabindex="-1"></a>rem_nc_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remove<span class="sc">$</span>total_reads_NC)</span>
<span id="cb66-728"><a href="#cb66-728" aria-hidden="true" tabindex="-1"></a>rem_sam_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remove<span class="sc">$</span>total_reads_samps)</span>
<span id="cb66-729"><a href="#cb66-729" aria-hidden="true" tabindex="-1"></a>per_reads_rem <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>( rem_nc_reads <span class="sc">/</span> (rem_nc_reads <span class="sc">+</span> rem_sam_reads)), </span>
<span id="cb66-730"><a href="#cb66-730" aria-hidden="true" tabindex="-1"></a>                       <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb66-731"><a href="#cb66-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-732"><a href="#cb66-732" aria-hidden="true" tabindex="-1"></a>ret_nc_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remain<span class="sc">$</span>total_reads_NC)</span>
<span id="cb66-733"><a href="#cb66-733" aria-hidden="true" tabindex="-1"></a>ret_sam_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remain<span class="sc">$</span>total_reads_samps)</span>
<span id="cb66-734"><a href="#cb66-734" aria-hidden="true" tabindex="-1"></a>per_reads_ret <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>( ret_nc_reads <span class="sc">/</span> (ret_nc_reads <span class="sc">+</span> ret_sam_reads)), </span>
<span id="cb66-735"><a href="#cb66-735" aria-hidden="true" tabindex="-1"></a>                       <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb66-736"><a href="#cb66-736" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-737"><a href="#cb66-737" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-738"><a href="#cb66-738" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span>          <span class="pp">|</span> Total ASVs          <span class="pp">|</span> NC reads         <span class="pp">|</span> non NC reads      <span class="pp">|</span> \% NC reads       <span class="pp">|</span></span>
<span id="cb66-739"><a href="#cb66-739" aria-hidden="true" tabindex="-1"></a><span class="pp">|----------|---------------------|------------------|-------------------|-------------------|</span></span>
<span id="cb66-740"><a href="#cb66-740" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Removed  <span class="pp">|</span> <span class="in">`r nrow(nc_remove)`</span> <span class="pp">|</span> <span class="in">`r rem_nc_reads`</span> <span class="pp">|</span> <span class="in">`r rem_sam_reads`</span> <span class="pp">|</span> <span class="in">`r per_reads_rem`</span> <span class="pp">|</span></span>
<span id="cb66-741"><a href="#cb66-741" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Retained <span class="pp">|</span> <span class="in">`r nrow(nc_remain)`</span> <span class="pp">|</span> <span class="in">`r ret_nc_reads`</span> <span class="pp">|</span> <span class="in">`r ret_sam_reads`</span> <span class="pp">|</span> <span class="in">`r per_reads_ret`</span> <span class="pp">|</span></span>
<span id="cb66-742"><a href="#cb66-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-743"><a href="#cb66-743" aria-hidden="true" tabindex="-1"></a>We identified a total of <span class="in">`r nrow(nc_check)`</span> ASVs that were present in at least 1 NC sample by at least 1 read. We removed any ASV where more than 10% of total reads were found in NC samples OR any ASV found in more than 10% of NC samples. Based on these criteria we removed <span class="in">`r nrow(nc_remove)`</span> ASVs from the data set, which represented <span class="in">`r rem_nc_reads`</span> total reads in NC samples and <span class="in">`r rem_sam_reads`</span> total reads in non-NC samples. Of the total reads removed <span class="in">`r per_reads_rem`</span>% came from NC samples. Of all ASVs identified in NC samples,<span class="in">`r nrow(nc_remain)`</span> were retained because the fell below the threshhold criteria. These ASVs accounted for <span class="in">`r ret_nc_reads`</span> reads in NC samples and <span class="in">`r ret_sam_reads`</span> reads in non-NC samples. NC samples accounted for <span class="in">`r per_reads_ret`</span>% of these reads.</span>
<span id="cb66-744"><a href="#cb66-744" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-745"><a href="#cb66-745" aria-hidden="true" tabindex="-1"></a>OK, now we can remove the NC samples and any ASVs that met our criteria described above.</span>
<span id="cb66-746"><a href="#cb66-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-749"><a href="#cb66-749" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-750"><a href="#cb66-750" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-751"><a href="#cb66-751" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-752"><a href="#cb66-752" aria-hidden="true" tabindex="-1"></a>tmp_no_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_na)</span>
<span id="cb66-753"><a href="#cb66-753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-754"><a href="#cb66-754" aria-hidden="true" tabindex="-1"></a>tmp_rem_asv <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(nc_remove<span class="sc">$</span>ASV_ID)</span>
<span id="cb66-755"><a href="#cb66-755" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span>otu_table <span class="ot">&lt;-</span> tmp_rem_nc<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span> </span>
<span id="cb66-756"><a href="#cb66-756" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">row.names</span>(tmp_no_nc<span class="sc">$</span>otu_table) <span class="sc">%in%</span> tmp_rem_asv)</span>
<span id="cb66-757"><a href="#cb66-757" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-758"><a href="#cb66-758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-759"><a href="#cb66-759" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span>sample_table <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_no_nc<span class="sc">$</span>sample_table, </span>
<span id="cb66-760"><a href="#cb66-760" aria-hidden="true" tabindex="-1"></a>                                 Treatment <span class="sc">!=</span> <span class="st">"Negative"</span>)</span>
<span id="cb66-761"><a href="#cb66-761" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-762"><a href="#cb66-762" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-763"><a href="#cb66-763" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-766"><a href="#cb66-766" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-767"><a href="#cb66-767" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-768"><a href="#cb66-768" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-769"><a href="#cb66-769" aria-hidden="true" tabindex="-1"></a>me_asv_no_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_nc)</span>
<span id="cb66-770"><a href="#cb66-770" aria-hidden="true" tabindex="-1"></a>me_asv_no_nc</span>
<span id="cb66-771"><a href="#cb66-771" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-772"><a href="#cb66-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-775"><a href="#cb66-775" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-776"><a href="#cb66-776" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-777"><a href="#cb66-777" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-778"><a href="#cb66-778" aria-hidden="true" tabindex="-1"></a>me_asv_no_nc</span>
<span id="cb66-779"><a href="#cb66-779" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-780"><a href="#cb66-780" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-781"><a href="#cb66-781" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The microeco object </span><span class="in">`me_asv_no_nc`</span><span class="at"> does not conatins the negative control sample(s). </span></span>
<span id="cb66-782"><a href="#cb66-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-783"><a href="#cb66-783" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Low-Count Samples</span></span>
<span id="cb66-784"><a href="#cb66-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-785"><a href="#cb66-785" aria-hidden="true" tabindex="-1"></a>Next, we can remove samples with really low read counts. I know from looking at these data that the are many low coverage samples. We can take a look at different thresholds to see how many samples will be eliminated. </span>
<span id="cb66-786"><a href="#cb66-786" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-789"><a href="#cb66-789" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-790"><a href="#cb66-790" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-791"><a href="#cb66-791" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-792"><a href="#cb66-792" aria-hidden="true" tabindex="-1"></a>samp_sum <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(me_asv_no_nc<span class="sc">$</span><span class="fu">sample_sums</span>())</span>
<span id="cb66-793"><a href="#cb66-793" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-794"><a href="#cb66-794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-797"><a href="#cb66-797" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-798"><a href="#cb66-798" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-799"><a href="#cb66-799" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-800"><a href="#cb66-800" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: markup</span></span>
<span id="cb66-801"><a href="#cb66-801" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: true</span></span>
<span id="cb66-802"><a href="#cb66-802" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">c</span>(<span class="st">"--- &lt;1000 reads ---"</span>, <span class="fu">length</span>(samp_sum[samp_sum[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">1000</span>, ]), <span class="st">" samples"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb66-803"><a href="#cb66-803" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">c</span>(<span class="st">"--- &lt;500 reads ---"</span>, <span class="fu">length</span>(samp_sum[samp_sum[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">500</span>, ]), <span class="st">" samples"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb66-804"><a href="#cb66-804" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">c</span>(<span class="st">"--- &lt;100 reads ---"</span>, <span class="fu">length</span>(samp_sum[samp_sum[<span class="dv">1</span>] <span class="sc">&lt;</span> <span class="dv">100</span>, ]), <span class="st">" samples"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb66-805"><a href="#cb66-805" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-806"><a href="#cb66-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-807"><a href="#cb66-807" aria-hidden="true" tabindex="-1"></a>We can start by eliminating samples with fewer than 100 reads. </span>
<span id="cb66-808"><a href="#cb66-808" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-811"><a href="#cb66-811" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-812"><a href="#cb66-812" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-813"><a href="#cb66-813" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-814"><a href="#cb66-814" aria-hidden="true" tabindex="-1"></a>tmp_no_low <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_nc)</span>
<span id="cb66-815"><a href="#cb66-815" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-816"><a href="#cb66-816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-819"><a href="#cb66-819" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-820"><a href="#cb66-820" aria-hidden="true" tabindex="-1"></a>tmp_no_low<span class="sc">$</span>otu_table <span class="ot">&lt;-</span> tmp_no_nc<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span></span>
<span id="cb66-821"><a href="#cb66-821" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">is.numeric</span>(.) <span class="sc">&amp;&amp;</span> <span class="fu">sum</span>(.) <span class="sc">&gt;</span> <span class="dv">100</span>))</span>
<span id="cb66-822"><a href="#cb66-822" aria-hidden="true" tabindex="-1"></a>tmp_no_low<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-823"><a href="#cb66-823" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-824"><a href="#cb66-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-825"><a href="#cb66-825" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-826"><a href="#cb66-826" aria-hidden="true" tabindex="-1"></a><span class="in">3 taxa with 0 abundance are removed from the otu_table ...</span></span>
<span id="cb66-827"><a href="#cb66-827" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-828"><a href="#cb66-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-829"><a href="#cb66-829" aria-hidden="true" tabindex="-1"></a>Check if any row sum (i.e., ASVs) is equal to 0 after removing the negative control sample(s).</span>
<span id="cb66-830"><a href="#cb66-830" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-833"><a href="#cb66-833" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-834"><a href="#cb66-834" aria-hidden="true" tabindex="-1"></a>tmp_row_sums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tmp_no_low<span class="sc">$</span>otu_table)</span>
<span id="cb66-835"><a href="#cb66-835" aria-hidden="true" tabindex="-1"></a><span class="fu">any</span>(tmp_row_sums <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb66-836"><a href="#cb66-836" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-837"><a href="#cb66-837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-838"><a href="#cb66-838" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-839"><a href="#cb66-839" aria-hidden="true" tabindex="-1"></a><span class="in">[1] FALSE</span></span>
<span id="cb66-840"><a href="#cb66-840" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-841"><a href="#cb66-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-844"><a href="#cb66-844" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-845"><a href="#cb66-845" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-846"><a href="#cb66-846" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-847"><a href="#cb66-847" aria-hidden="true" tabindex="-1"></a>me_asv_no_low <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_low)</span>
<span id="cb66-848"><a href="#cb66-848" aria-hidden="true" tabindex="-1"></a>me_asv_no_low</span>
<span id="cb66-849"><a href="#cb66-849" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-850"><a href="#cb66-850" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-853"><a href="#cb66-853" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-854"><a href="#cb66-854" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-855"><a href="#cb66-855" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-856"><a href="#cb66-856" aria-hidden="true" tabindex="-1"></a>me_asv_no_low</span>
<span id="cb66-857"><a href="#cb66-857" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-858"><a href="#cb66-858" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-859"><a href="#cb66-859" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; The microeco object </span><span class="in">`me_asv_no_low`</span><span class="at"> does not conatins the low abundance samples. </span></span>
<span id="cb66-860"><a href="#cb66-860" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-861"><a href="#cb66-861" aria-hidden="true" tabindex="-1"></a>We will quickly add final read counts and ASVs, then create the final microtable object.</span>
<span id="cb66-862"><a href="#cb66-862" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-865"><a href="#cb66-865" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-866"><a href="#cb66-866" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-867"><a href="#cb66-867" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-868"><a href="#cb66-868" aria-hidden="true" tabindex="-1"></a>tmp_final_rc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(me_asv_no_low<span class="sc">$</span><span class="fu">sample_sums</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb66-869"><a href="#cb66-869" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-870"><a href="#cb66-870" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"final_rc"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb66-871"><a href="#cb66-871" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-872"><a href="#cb66-872" aria-hidden="true" tabindex="-1"></a>tmp_final_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(me_asv_no_low<span class="sc">$</span>otu_table))</span>
<span id="cb66-873"><a href="#cb66-873" aria-hidden="true" tabindex="-1"></a>tmp_final_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(tmp_final_asv <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb66-874"><a href="#cb66-874" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-875"><a href="#cb66-875" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"final_asv"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb66-876"><a href="#cb66-876" aria-hidden="true" tabindex="-1"></a>tmp_final <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_final_rc, tmp_final_asv)</span>
<span id="cb66-877"><a href="#cb66-877" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-878"><a href="#cb66-878" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(sample_tab, tmp_final) <span class="sc">%&gt;%</span> </span>
<span id="cb66-879"><a href="#cb66-879" aria-hidden="true" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">relocate</span>(<span class="fu">c</span>(final_rc, final_asv), <span class="at">.after =</span> lotus3_asv)</span>
<span id="cb66-880"><a href="#cb66-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-881"><a href="#cb66-881" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> sample_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb66-882"><a href="#cb66-882" aria-hidden="true" tabindex="-1"></a>sample_tab<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sample_tab)</span>
<span id="cb66-883"><a href="#cb66-883" aria-hidden="true" tabindex="-1"></a>sample_tab <span class="ot">&lt;-</span> sample_tab <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">relocate</span>(SampleID)</span>
<span id="cb66-884"><a href="#cb66-884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-885"><a href="#cb66-885" aria-hidden="true" tabindex="-1"></a>full_sample_tab <span class="ot">&lt;-</span> sample_tab</span>
<span id="cb66-886"><a href="#cb66-886" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-887"><a href="#cb66-887" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-888"><a href="#cb66-888" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-891"><a href="#cb66-891" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-892"><a href="#cb66-892" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-893"><a href="#cb66-893" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-894"><a href="#cb66-894" aria-hidden="true" tabindex="-1"></a>tmp_samp_tab <span class="ot">&lt;-</span> sample_tab[<span class="fu">complete.cases</span>(sample_tab), ]</span>
<span id="cb66-895"><a href="#cb66-895" aria-hidden="true" tabindex="-1"></a>tmp_tax_tab <span class="ot">&lt;-</span> me_asv_no_low<span class="sc">$</span>tax_table</span>
<span id="cb66-896"><a href="#cb66-896" aria-hidden="true" tabindex="-1"></a>tmp_otu_tab <span class="ot">&lt;-</span> me_asv_no_low<span class="sc">$</span>otu_table</span>
<span id="cb66-897"><a href="#cb66-897" aria-hidden="true" tabindex="-1"></a>tmp_rep_fasta <span class="ot">&lt;-</span> me_asv_no_low<span class="sc">$</span>rep_fasta</span>
<span id="cb66-898"><a href="#cb66-898" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-899"><a href="#cb66-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-902"><a href="#cb66-902" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-903"><a href="#cb66-903" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb66-904"><a href="#cb66-904" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-905"><a href="#cb66-905" aria-hidden="true" tabindex="-1"></a>me_asv <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> tmp_samp_tab, </span>
<span id="cb66-906"><a href="#cb66-906" aria-hidden="true" tabindex="-1"></a>                         <span class="at">otu_table =</span> tmp_otu_tab, </span>
<span id="cb66-907"><a href="#cb66-907" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tax_table =</span> tmp_tax_tab, </span>
<span id="cb66-908"><a href="#cb66-908" aria-hidden="true" tabindex="-1"></a>                         <span class="at">rep_fasta =</span> tmp_rep_fasta)</span>
<span id="cb66-909"><a href="#cb66-909" aria-hidden="true" tabindex="-1"></a>me_asv<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb66-910"><a href="#cb66-910" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-911"><a href="#cb66-911" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-914"><a href="#cb66-914" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-915"><a href="#cb66-915" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-916"><a href="#cb66-916" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-917"><a href="#cb66-917" aria-hidden="true" tabindex="-1"></a>me_asv</span>
<span id="cb66-918"><a href="#cb66-918" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-919"><a href="#cb66-919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-920"><a href="#cb66-920" aria-hidden="true" tabindex="-1"></a><span class="fu"># Summary</span></span>
<span id="cb66-921"><a href="#cb66-921" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-922"><a href="#cb66-922" aria-hidden="true" tabindex="-1"></a>Now time to summarize the data. For this we use the R package <span class="co">[</span><span class="ot">miaverse</span><span class="co">](https://microbiome.github.io)</span> <span class="co">[</span><span class="ot">@felix2024mia</span><span class="co">]</span>. </span>
<span id="cb66-923"><a href="#cb66-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-926"><a href="#cb66-926" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-927"><a href="#cb66-927" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-928"><a href="#cb66-928" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-929"><a href="#cb66-929" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(me_asv_raw<span class="sc">$</span>sample_table), <span class="fu">colnames</span>(me_asv_raw<span class="sc">$</span>otu_table))</span>
<span id="cb66-930"><a href="#cb66-930" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(me_asv<span class="sc">$</span>sample_table), <span class="fu">colnames</span>(me_asv<span class="sc">$</span>otu_table))</span>
<span id="cb66-931"><a href="#cb66-931" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-932"><a href="#cb66-932" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-933"><a href="#cb66-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-936"><a href="#cb66-936" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-937"><a href="#cb66-937" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-938"><a href="#cb66-938" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-939"><a href="#cb66-939" aria-hidden="true" tabindex="-1"></a><span class="co"># https://github.com/microbiome/OMA/issues/202</span></span>
<span id="cb66-940"><a href="#cb66-940" aria-hidden="true" tabindex="-1"></a>tmp_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(me_asv_raw<span class="sc">$</span>otu_table)</span>
<span id="cb66-941"><a href="#cb66-941" aria-hidden="true" tabindex="-1"></a>tmp_assays <span class="ot">&lt;-</span>  <span class="fu">SimpleList</span>(<span class="at">counts =</span> tmp_counts)</span>
<span id="cb66-942"><a href="#cb66-942" aria-hidden="true" tabindex="-1"></a>mia_me_asv_raw <span class="ot">&lt;-</span> <span class="fu">TreeSummarizedExperiment</span>(<span class="at">assays =</span> tmp_assays,</span>
<span id="cb66-943"><a href="#cb66-943" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colData =</span> <span class="fu">DataFrame</span>(me_asv_raw<span class="sc">$</span>sample_table),</span>
<span id="cb66-944"><a href="#cb66-944" aria-hidden="true" tabindex="-1"></a>                                <span class="at">rowData =</span> <span class="fu">DataFrame</span>(me_asv_raw<span class="sc">$</span>tax_table))</span>
<span id="cb66-945"><a href="#cb66-945" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb66-946"><a href="#cb66-946" aria-hidden="true" tabindex="-1"></a>tmp_counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(me_asv<span class="sc">$</span>otu_table)</span>
<span id="cb66-947"><a href="#cb66-947" aria-hidden="true" tabindex="-1"></a>tmp_assays <span class="ot">&lt;-</span>  <span class="fu">SimpleList</span>(<span class="at">counts =</span> tmp_counts)</span>
<span id="cb66-948"><a href="#cb66-948" aria-hidden="true" tabindex="-1"></a>mia_me_asv <span class="ot">&lt;-</span> <span class="fu">TreeSummarizedExperiment</span>(<span class="at">assays =</span> tmp_assays,</span>
<span id="cb66-949"><a href="#cb66-949" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colData =</span> <span class="fu">DataFrame</span>(me_asv<span class="sc">$</span>sample_table),</span>
<span id="cb66-950"><a href="#cb66-950" aria-hidden="true" tabindex="-1"></a>                                <span class="at">rowData =</span> <span class="fu">DataFrame</span>(me_asv<span class="sc">$</span>tax_table))</span>
<span id="cb66-951"><a href="#cb66-951" aria-hidden="true" tabindex="-1"></a>mia_me_asv_raw_summ <span class="ot">&lt;-</span> <span class="fu">summary</span>(mia_me_asv_raw, <span class="at">assay.type =</span> <span class="st">"counts"</span>)</span>
<span id="cb66-952"><a href="#cb66-952" aria-hidden="true" tabindex="-1"></a>mia_me_asv_summ <span class="ot">&lt;-</span> <span class="fu">summary</span>(mia_me_asv, <span class="at">assay.type =</span> <span class="st">"counts"</span>)</span>
<span id="cb66-953"><a href="#cb66-953" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb66-954"><a href="#cb66-954" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-955"><a href="#cb66-955" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-958"><a href="#cb66-958" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-959"><a href="#cb66-959" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-960"><a href="#cb66-960" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-961"><a href="#cb66-961" aria-hidden="true" tabindex="-1"></a>objs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"me_asv_raw"</span>, <span class="st">"me_asv_no_na"</span>, <span class="st">"me_asv_no_cont"</span>, </span>
<span id="cb66-962"><a href="#cb66-962" aria-hidden="true" tabindex="-1"></a>          <span class="st">"me_asv_no_nc"</span>, <span class="st">"me_asv_no_low"</span>)</span>
<span id="cb66-963"><a href="#cb66-963" aria-hidden="true" tabindex="-1"></a>pipe_summary <span class="ot">&lt;-</span> <span class="fu">summarize_objs</span>(objs)</span>
<span id="cb66-964"><a href="#cb66-964" aria-hidden="true" tabindex="-1"></a><span class="co">#metric_name &lt;- c("original", "No NA kingdom", </span></span>
<span id="cb66-965"><a href="#cb66-965" aria-hidden="true" tabindex="-1"></a><span class="co">#                "no contaminants", "no neg control", </span></span>
<span id="cb66-966"><a href="#cb66-966" aria-hidden="true" tabindex="-1"></a><span class="co">#                "no low count samps")</span></span>
<span id="cb66-967"><a href="#cb66-967" aria-hidden="true" tabindex="-1"></a>pipe_summary<span class="sc">$</span>me_dataset <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb66-968"><a href="#cb66-968" aria-hidden="true" tabindex="-1"></a>  <span class="st">"original"</span>, <span class="st">"No NA kingdom"</span>,</span>
<span id="cb66-969"><a href="#cb66-969" aria-hidden="true" tabindex="-1"></a>  <span class="st">"no neg control"</span>, <span class="st">"no low count samps"</span>)</span>
<span id="cb66-970"><a href="#cb66-970" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pipe_summary)</span>
<span id="cb66-971"><a href="#cb66-971" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-972"><a href="#cb66-972" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-973"><a href="#cb66-973" aria-hidden="true" tabindex="-1"></a>Here we can see how the total number of ASVs, reads, and samples changed during the curation process. </span>
<span id="cb66-974"><a href="#cb66-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-977"><a href="#cb66-977" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-978"><a href="#cb66-978" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-979"><a href="#cb66-979" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-980"><a href="#cb66-980" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(pipe_summary)</span>
<span id="cb66-981"><a href="#cb66-981" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-982"><a href="#cb66-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-983"><a href="#cb66-983" aria-hidden="true" tabindex="-1"></a>And some additional metrics describing the changes. </span>
<span id="cb66-984"><a href="#cb66-984" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-985"><a href="#cb66-985" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Metric                <span class="pp">|</span> Start                                      <span class="pp">|</span> End                                    <span class="pp">|</span></span>
<span id="cb66-986"><a href="#cb66-986" aria-hidden="true" tabindex="-1"></a><span class="pp">|-----------------------|--------------------------------------------|----------------------------------------|</span></span>
<span id="cb66-987"><a href="#cb66-987" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Min. number reads     <span class="pp">|</span> <span class="in">`r mia_me_asv_raw_summ$samples[2]`</span>         <span class="pp">|</span> <span class="in">`r mia_me_asv_summ$samples[2]`</span>         <span class="pp">|</span></span>
<span id="cb66-988"><a href="#cb66-988" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Max. number reads     <span class="pp">|</span> <span class="in">`r mia_me_asv_raw_summ$samples[3]`</span>         <span class="pp">|</span> <span class="in">`r mia_me_asv_summ$samples[3]`</span>         <span class="pp">|</span></span>
<span id="cb66-989"><a href="#cb66-989" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Total number reads    <span class="pp">|</span> <span class="in">`r mia_me_asv_raw_summ$samples[1]`</span>         <span class="pp">|</span> <span class="in">`r mia_me_asv_summ$samples[1]`</span>         <span class="pp">|</span></span>
<span id="cb66-990"><a href="#cb66-990" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Avg number reads      <span class="pp">|</span> <span class="in">`r round(mia_me_asv_raw_summ$samples[5])`</span>  <span class="pp">|</span> <span class="in">`r round(mia_me_asv_summ$samples[5])`</span>  <span class="pp">|</span></span>
<span id="cb66-991"><a href="#cb66-991" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Median no. reads      <span class="pp">|</span> <span class="in">`r mia_me_asv_raw_summ$samples[4]`</span>         <span class="pp">|</span> <span class="in">`r mia_me_asv_summ$samples[4]`</span>         <span class="pp">|</span></span>
<span id="cb66-992"><a href="#cb66-992" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Total ASVs            <span class="pp">|</span> <span class="in">`r mia_me_asv_raw_summ$features[1]`</span>        <span class="pp">|</span> <span class="in">`r mia_me_asv_summ$features[1]`</span>        <span class="pp">|</span></span>
<span id="cb66-993"><a href="#cb66-993" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> No. singleton ASVs    <span class="pp">|</span> <span class="in">`r mia_me_asv_raw_summ$features[2]`</span>        <span class="pp">|</span> <span class="in">`r mia_me_asv_summ$features[2]`</span>        <span class="pp">|</span></span>
<span id="cb66-994"><a href="#cb66-994" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Avg ASVs per sample.  <span class="pp">|</span> <span class="in">`r round(mia_me_asv_raw_summ$features[3])`</span> <span class="pp">|</span> <span class="in">`r round(mia_me_asv_summ$features[3])`</span> <span class="pp">|</span></span>
<span id="cb66-995"><a href="#cb66-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-996"><a href="#cb66-996" aria-hidden="true" tabindex="-1"></a>We started off with <span class="in">`r nrow(me_asv_raw$tax_table)`</span> ASVs and <span class="in">`r nrow(me_asv_raw$sample_table)`</span> samples. Screening for <span class="in">`NA`</span> kingdom assignment removed an additional <span class="in">`r nrow(me_asv_raw$tax_table) - nrow(me_asv_no_na$tax_table)`</span> ASVs. After removing the negative controls there were <span class="in">`r nrow(me_asv_no_nc$tax_table)`</span> ASVs and <span class="in">`r nrow(me_asv_no_na$sample_table)`</span>. After removing low-count samples, there were <span class="in">`r nrow(me_asv$tax_table)`</span> ASVs and <span class="in">`r nrow(me_asv$sample_table)`</span> samples remaining.</span>
<span id="cb66-997"><a href="#cb66-997" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1000"><a href="#cb66-1000" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1001"><a href="#cb66-1001" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1002"><a href="#cb66-1002" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1003"><a href="#cb66-1003" aria-hidden="true" tabindex="-1"></a><span class="co">#---------Starting data-------------#</span></span>
<span id="cb66-1004"><a href="#cb66-1004" aria-hidden="true" tabindex="-1"></a>tmp_start <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(me_asv_raw)</span>
<span id="cb66-1005"><a href="#cb66-1005" aria-hidden="true" tabindex="-1"></a>tmp_start_rc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tmp_start<span class="sc">$</span><span class="fu">sample_sums</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb66-1006"><a href="#cb66-1006" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-1007"><a href="#cb66-1007" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"start_rc"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb66-1008"><a href="#cb66-1008" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1009"><a href="#cb66-1009" aria-hidden="true" tabindex="-1"></a>tmp_start_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(tmp_start<span class="sc">$</span>otu_table))</span>
<span id="cb66-1010"><a href="#cb66-1010" aria-hidden="true" tabindex="-1"></a>tmp_start_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(tmp_start_asv <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb66-1011"><a href="#cb66-1011" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-1012"><a href="#cb66-1012" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"start_asv"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb66-1013"><a href="#cb66-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1014"><a href="#cb66-1014" aria-hidden="true" tabindex="-1"></a><span class="co">#---------Final data-------------#</span></span>
<span id="cb66-1015"><a href="#cb66-1015" aria-hidden="true" tabindex="-1"></a>tmp_final <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(me_asv)</span>
<span id="cb66-1016"><a href="#cb66-1016" aria-hidden="true" tabindex="-1"></a>tmp_lotus3_rc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tmp_final<span class="sc">$</span><span class="fu">sample_sums</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb66-1017"><a href="#cb66-1017" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-1018"><a href="#cb66-1018" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"final_rc"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb66-1019"><a href="#cb66-1019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1020"><a href="#cb66-1020" aria-hidden="true" tabindex="-1"></a>tmp_final_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(tmp_final<span class="sc">$</span>otu_table))</span>
<span id="cb66-1021"><a href="#cb66-1021" aria-hidden="true" tabindex="-1"></a>tmp_final_asv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(tmp_final_asv <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb66-1022"><a href="#cb66-1022" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb66-1023"><a href="#cb66-1023" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"final_asv"</span> <span class="ot">=</span> <span class="dv">2</span>)</span>
<span id="cb66-1024"><a href="#cb66-1024" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1025"><a href="#cb66-1025" aria-hidden="true" tabindex="-1"></a><span class="co">#---------Summary data-------------#</span></span>
<span id="cb66-1026"><a href="#cb66-1026" aria-hidden="true" tabindex="-1"></a>tmp_summary <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">full_join</span>(tmp_start_rc, tmp_start_asv) <span class="sc">%&gt;%</span></span>
<span id="cb66-1027"><a href="#cb66-1027" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_lotus3_rc) <span class="sc">%&gt;%</span></span>
<span id="cb66-1028"><a href="#cb66-1028" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_final_asv)</span>
<span id="cb66-1029"><a href="#cb66-1029" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1030"><a href="#cb66-1030" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_delim</span>(tmp_summary, </span>
<span id="cb66-1031"><a href="#cb66-1031" aria-hidden="true" tabindex="-1"></a>                   <span class="st">"files/CURATE/oo_final_summary.txt"</span>, </span>
<span id="cb66-1032"><a href="#cb66-1032" aria-hidden="true" tabindex="-1"></a>                   <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb66-1033"><a href="#cb66-1033" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb66-1034"><a href="#cb66-1034" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1035"><a href="#cb66-1035" aria-hidden="true" tabindex="-1"></a>curate_summary <span class="ot">&lt;-</span> tmp_summary</span>
<span id="cb66-1036"><a href="#cb66-1036" aria-hidden="true" tabindex="-1"></a>removed_samples <span class="ot">&lt;-</span> curate_summary<span class="sc">$</span>SampleID[<span class="fu">is.na</span>(curate_summary<span class="sc">$</span>final_rc)]</span>
<span id="cb66-1037"><a href="#cb66-1037" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1038"><a href="#cb66-1038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1041"><a href="#cb66-1041" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1042"><a href="#cb66-1042" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb66-1043"><a href="#cb66-1043" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1044"><a href="#cb66-1044" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-1045"><a href="#cb66-1045" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-make-final-rc-asv-table-oo</span></span>
<span id="cb66-1046"><a href="#cb66-1046" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Beginning and ending read counts &amp; total ASVs by sample."</span></span>
<span id="cb66-1047"><a href="#cb66-1047" aria-hidden="true" tabindex="-1"></a><span class="fu">make_final_asv_rc_table</span>(curate_summary)</span>
<span id="cb66-1048"><a href="#cb66-1048" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1049"><a href="#cb66-1049" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1050"><a href="#cb66-1050" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;/</span><span class="kw">br</span><span class="dt">&gt;</span></span>
<span id="cb66-1051"><a href="#cb66-1051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1052"><a href="#cb66-1052" aria-hidden="true" tabindex="-1"></a>We lost <span class="in">`r length(removed_samples)`</span> samples after curating the dataset. </span>
<span id="cb66-1053"><a href="#cb66-1053" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1056"><a href="#cb66-1056" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1057"><a href="#cb66-1057" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb66-1058"><a href="#cb66-1058" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1059"><a href="#cb66-1059" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb66-1060"><a href="#cb66-1060" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(removed_samples)</span>
<span id="cb66-1061"><a href="#cb66-1061" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1062"><a href="#cb66-1062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1063"><a href="#cb66-1063" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1066"><a href="#cb66-1066" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1067"><a href="#cb66-1067" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1068"><a href="#cb66-1068" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1069"><a href="#cb66-1069" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: the following code blocks are for exporting data</span></span>
<span id="cb66-1070"><a href="#cb66-1070" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1071"><a href="#cb66-1071" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1074"><a href="#cb66-1074" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb66-1075"><a href="#cb66-1075" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1076"><a href="#cb66-1076" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1077"><a href="#cb66-1077" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> <span class="at">-p</span> files/SHARE/oo_curated_data/</span>
<span id="cb66-1078"><a href="#cb66-1078" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1079"><a href="#cb66-1079" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1082"><a href="#cb66-1082" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1083"><a href="#cb66-1083" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1084"><a href="#cb66-1084" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1085"><a href="#cb66-1085" aria-hidden="true" tabindex="-1"></a>tmp_path <span class="ot">&lt;-</span> <span class="st">"files/SHARE/oo_curated_data/"</span></span>
<span id="cb66-1086"><a href="#cb66-1086" aria-hidden="true" tabindex="-1"></a>tmp_otu_n <span class="ot">&lt;-</span> <span class="st">"oo_otu_table.txt"</span></span>
<span id="cb66-1087"><a href="#cb66-1087" aria-hidden="true" tabindex="-1"></a>tmp_tax_n <span class="ot">&lt;-</span> <span class="st">"oo_tax_table.txt"</span></span>
<span id="cb66-1088"><a href="#cb66-1088" aria-hidden="true" tabindex="-1"></a>tmp_sam_n <span class="ot">&lt;-</span> <span class="st">"oo_sample_table.txt"</span></span>
<span id="cb66-1089"><a href="#cb66-1089" aria-hidden="true" tabindex="-1"></a>tmp_rep_n <span class="ot">&lt;-</span> <span class="st">"oo_rep_fasta.fasta"</span></span>
<span id="cb66-1090"><a href="#cb66-1090" aria-hidden="true" tabindex="-1"></a>tmp_me_n <span class="ot">&lt;-</span> <span class="st">"oo_me_asv.rds"</span></span>
<span id="cb66-1091"><a href="#cb66-1091" aria-hidden="true" tabindex="-1"></a>tmp_ps_n <span class="ot">&lt;-</span> <span class="st">"oo_ps_asv.rds"</span></span>
<span id="cb66-1092"><a href="#cb66-1092" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1093"><a href="#cb66-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1096"><a href="#cb66-1096" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1097"><a href="#cb66-1097" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1098"><a href="#cb66-1098" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1099"><a href="#cb66-1099" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(seqinr)</span>
<span id="cb66-1100"><a href="#cb66-1100" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------OTU Table--------------#</span></span>
<span id="cb66-1101"><a href="#cb66-1101" aria-hidden="true" tabindex="-1"></a>tmp_otu <span class="ot">&lt;-</span> me_asv<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb66-1102"><a href="#cb66-1102" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(tmp_otu, <span class="fu">paste</span>(tmp_path, tmp_otu_n), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb66-1103"><a href="#cb66-1103" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------TAX Table--------------#</span></span>
<span id="cb66-1104"><a href="#cb66-1104" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> me_asv<span class="sc">$</span>tax_table <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb66-1105"><a href="#cb66-1105" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(tmp_tax, <span class="fu">paste</span>(tmp_path, tmp_tax_n), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb66-1106"><a href="#cb66-1106" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------SAMP Table--------------#</span></span>
<span id="cb66-1107"><a href="#cb66-1107" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(me_asv<span class="sc">$</span>sample_table, <span class="fu">paste</span>(tmp_path, tmp_sam_n), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb66-1108"><a href="#cb66-1108" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------REP FASTA--------------#</span></span>
<span id="cb66-1109"><a href="#cb66-1109" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(</span>
<span id="cb66-1110"><a href="#cb66-1110" aria-hidden="true" tabindex="-1"></a>  <span class="at">sequences =</span> <span class="fu">as.list</span>(<span class="fu">as.character</span>(me_asv<span class="sc">$</span>rep_fasta)),</span>
<span id="cb66-1111"><a href="#cb66-1111" aria-hidden="true" tabindex="-1"></a>  <span class="at">names =</span> <span class="fu">names</span>(me_asv<span class="sc">$</span>rep_fasta),</span>
<span id="cb66-1112"><a href="#cb66-1112" aria-hidden="true" tabindex="-1"></a>  <span class="at">file.out =</span> <span class="fu">paste</span>(tmp_path, tmp_rep_n)</span>
<span id="cb66-1113"><a href="#cb66-1113" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb66-1114"><a href="#cb66-1114" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(me_asv, <span class="fu">paste</span>(tmp_path, tmp_me_n))</span>
<span id="cb66-1115"><a href="#cb66-1115" aria-hidden="true" tabindex="-1"></a>ps_asv <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_asv)</span>
<span id="cb66-1116"><a href="#cb66-1116" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_asv, <span class="fu">paste</span>(tmp_path, tmp_ps_n))</span>
<span id="cb66-1117"><a href="#cb66-1117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1118"><a href="#cb66-1118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1121"><a href="#cb66-1121" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb66-1122"><a href="#cb66-1122" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1123"><a href="#cb66-1123" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1124"><a href="#cb66-1124" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> files/SHARE</span>
<span id="cb66-1125"><a href="#cb66-1125" aria-hidden="true" tabindex="-1"></a><span class="fu">zip</span> <span class="at">-r</span> oo_curated_data.zip oo_curated_data/</span>
<span id="cb66-1126"><a href="#cb66-1126" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="at">-r</span> oo_curated_data/</span>
<span id="cb66-1127"><a href="#cb66-1127" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1128"><a href="#cb66-1128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1131"><a href="#cb66-1131" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb66-1132"><a href="#cb66-1132" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb66-1133"><a href="#cb66-1133" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb66-1134"><a href="#cb66-1134" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb66-1135"><a href="#cb66-1135" aria-hidden="true" tabindex="-1"></a>gdata<span class="sc">::</span><span class="fu">keep</span>(ds_metrics, sample_tab, full_sample_tab, tax_tab, otu_tab, samp_sum,</span>
<span id="cb66-1136"><a href="#cb66-1136" aria-hidden="true" tabindex="-1"></a>            me_asv_raw, me_asv_no_na, me_asv_nc, sample_tab_trim, </span>
<span id="cb66-1137"><a href="#cb66-1137" aria-hidden="true" tabindex="-1"></a>            me_asv_no_nc, me_asv_no_low, me_asv, nc_asvs, nc_check, </span>
<span id="cb66-1138"><a href="#cb66-1138" aria-hidden="true" tabindex="-1"></a>            nc_remove, nc_remain, rem_nc_reads, rem_sam_reads, per_reads_rem, </span>
<span id="cb66-1139"><a href="#cb66-1139" aria-hidden="true" tabindex="-1"></a>            per_reads_ret, ret_nc_reads, ret_sam_reads, mia_me_asv_raw, ps_asv,</span>
<span id="cb66-1140"><a href="#cb66-1140" aria-hidden="true" tabindex="-1"></a>            mia_me_asv, mia_me_asv_raw_summ, mia_me_asv_summ, pipe_summary,</span>
<span id="cb66-1141"><a href="#cb66-1141" aria-hidden="true" tabindex="-1"></a>            curate_summary, removed_samples, </span>
<span id="cb66-1142"><a href="#cb66-1142" aria-hidden="true" tabindex="-1"></a>            <span class="at">sure =</span> <span class="cn">TRUE</span>)</span>
<span id="cb66-1143"><a href="#cb66-1143" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb66-1144"><a href="#cb66-1144" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="st">"page_build/oo_curate.rdata"</span>)</span>
<span id="cb66-1145"><a href="#cb66-1145" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(me_asv, <span class="st">"files/CURATE/oo_me_asv.rds"</span>)</span>
<span id="cb66-1146"><a href="#cb66-1146" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_asv, <span class="st">"files/CURATE/oo_ps_asv.rds"</span>)</span>
<span id="cb66-1147"><a href="#cb66-1147" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb66-1148"><a href="#cb66-1148" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb66-1149"><a href="#cb66-1149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1150"><a href="#cb66-1150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-1151"><a href="#cb66-1151" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/code/_footer.qmd &gt;}}</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>© Copyright 2025 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; The SWELTR Project. Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sweltr/depth/edit/main/curate_oo.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sweltr/depth/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>


</body></html>