---
title: "Primer Removal"
subtitle: "Scripts for primer removal using cutadapt for the SSU, ITS, AMF, & Oomycete datasets"
format:
  html:
    other-links: 
      - text: Raw SSU fastq files
        href: https://figshare.com/s/9b449e05451c13bfc061
      - text: Raw ITS fastq files
        href: https://figshare.com/s/c3eec1f49a13813da3e9
      - text: Raw AMF fastq files
        href: https://figshare.com/s/0b4d6bb16df5bfc434aa
      - text: Raw OO fastq files
        href: https://figshare.com/s/53b39b09cca3a96919eb
---

{{< include include/code/_setup.qmd >}}

{{< include include/code/_primers.qmd >}}

# Workflow Overview

This workflow uses cutadapt to remove primer sequences from the four amplicon datasets. Included are links to R scripts, associated processing files, and access to raw fastq files.

Here is the basic rundown for renaming samples and removing primer pairs. All of the data and scripts you need can be downloaded below. 

1. Download raw fastq files from figshare using the links in the table below. 
2. Rename fastq files using the `rename.sh` script and the corresponding lookup table.
3. Run the corresponding cutadapt R script to remove primers. 

At this point your samples are ready for ASV calling. For a more detailed explaination of the workflow and access to summary data please see the **Workflow** section below.

::: {.callout-note}
For simplicity and consistency we use the abbreviation `SSU` or `ssu` to refer to the 16S rRNA dataset. We recognize that the small subunit (SSU) applies to multiple domains but here we specifically refer to Bacteria and Archaea. Further we use the abbreviations `ITS` or `its` for general fungi, `AMF` or `amf` for Arbuscular Mycorrhizal Fungi, and `OO` or `oo` for Oomycetes. 
:::

## Raw Data & Scripts

Here is everything you need to run this workflow--raw data files, fastq renaming scripts and lookup tables, and cutadapt R scripts. 

</br>

{{< include include/data/_raw_data_and_scripts.qmd >}}

</br>

::: {.callout-note appearance="default" icon=false}

### {{< fa download >}} &nbsp; rename.sh

{{< downloadthis files/RENAME/rename.sh dname="rename" label="Bash script to rename samples" icon="code-slash" type="link" >}}

:::

</br>

You can also download the scripts and lookup tables for each dataset from [figshare](https://figshare.com/s/532e2ab690d3c9b1ab43). 

## Citable resources 

+-----------+-----------------------------+--------------------------------+
| Dataset   | Forward primer              | Reverse Primer                 |
+===========+=============================+================================+
| SSU       | GTGCCAGCMGCCGCGGTAA <br>    | GGACTACHVGGGTWTCTAAT  <br>     |
|           | 515f [@caporaso2011global]  | 806r [@caporaso2011global]     |
+-----------+-----------------------------+--------------------------------+
| ITS       | CTTGGTCATTTAGAGGAAGTAA <br> | GCTGCGTTCTTCATCGATGC  <br>     |
|           | ITS1f [@gardes1993its]      | ITS2 [@white1990amplification] |
+-----------+-----------------------------+--------------------------------+
| AMF       | AAGCTCGTAGTTGAATTTCG <br>   | CCCAACTATCCCTATTAATCAT <br>    |
|           | AMV4-5NF [@sato2005new]     | AMDGR [@sato2005new]           |
+-----------+-----------------------------+--------------------------------+
| Oomycete  | GGAAGGATCATTACCACA <br>     | GCTGCGTTCTTCATCGATGC <br>      |
|           | ITS1oo [@riit2016oomycete]  | ITS2 [@white1990amplification] |
+-----------+-----------------------------+--------------------------------+

: Primer pairs used for each dataset along with primer names and associated references. {tbl-colwidths="[20,40,40]"}

#### Packages used in this workflow:

1. [cutadapt](https://github.com/marcelm/cutadapt) [@martin2011cutadapt]

</br>

## References

::: {#refs}
:::

# Workflow

## Required Packages

```{r}
#| code-fold: true
#| code-summary: "Click to see required packages"
#| code-overflow: wrap
set.seed(919191)
library(dada2)
library(ShortRead)
library(Biostrings)
library(DECIPHER)
```

## Rename fastq Files

The first thing we did in this workflow is rename all fastq files so that the names are more informative. Here we generate sample names that contain the sample plot information, the soil depth, warming treatment, and sample pairing. To accomplish this we use a two column tab-delimited file (e.g., `ssu_rename.txt`) where the first column contains the original name (e.g., `SWELTR-9_R1.fastq.gz`) and the second column contains the new name (e.g., `P03-D00-010-W4B_R1.fastq.gz`). You can download this table above. 

This handy bash script will rename all fastq files. 

::: {.callout-note appearance="default" icon=false}

### {{< fa download >}} &nbsp; rename.sh

{{< downloadthis files/RENAME/rename.sh dname="rename" label="Bash script to rename samples" icon="code-slash" type="link" >}}

:::

<details>

<summary>Click here to see the bash code for renaming fastq files.</summary>

{{< include include/partials/_rename_sh.qmd >}}

</details>

Simply run as such with the script by passing it a directory containing fastq files and a lookup table. 

```{bash}
bash rename.sh /path/to/fastq_files /path/to/rename_table
```

```
`SWELTR-1_R1.fastq.gz' -> `P01-D00-010-W4A_R1.fastq.gz'
`SWELTR-2_R1.fastq.gz' -> `P01-D10-020-W4A_R1.fastq.gz'
`SWELTR-3_R1.fastq.gz' -> `P01-D20-050-W4A_R1.fastq.gz'
`SWELTR-4_R1.fastq.gz' -> `P01-D50-100-W4A_R1.fastq.gz'
`SWELTR-5_R1.fastq.gz' -> `P02-D00-010-C0A_R1.fastq.gz'
`SWELTR-6_R1.fastq.gz' -> `P02-D10-020-C0A_R1.fastq.gz'
```

## Cutadapt Workflows

In each tab below you can find the complete cutadapt workflow for each dataset. The worflows are nearly identical except for the the cutadapt parameters `--minimum-length` and `--maximum-length`. The differences are described here: 

+-----------+------------------+--------------------+
| Dataset   | `minimum-length` | `maximum-length`   |
+===========+==================+====================+
| SSU       | 200              | 300                |
+-----------+------------------+--------------------+
| ITS       | 20               | 500                |
+-----------+------------------+--------------------+
| AMF       | 100              | 300                |
+-----------+------------------+--------------------+
| Oomycete  | 20               | 500                |
+-----------+------------------+--------------------+

: Differences in cutadapt parameters for `--minimum-length` and `--maximum-length`. {tbl-colwidths="[30,30,30]"}

</br>
</br>

::: {.panel-tabset .nav-pills .nav-fill}

# SSU

{{< include include/partials/_cutadapt_primer_remove_ssu.qmd >}}

# ITS

{{< include include/partials/_cutadapt_primer_remove_its.qmd >}}

# AMF

{{< include include/partials/_cutadapt_primer_remove_amf.qmd >}}

# Oomycete

{{< include include/partials/_cutadapt_primer_remove_oo.qmd >}}

:::

# Workflow Summary

```{r}
#| echo: false
#| eval: true

me_objs <- c("ssu", "its", "amf", "oo")
tab_r_path <- "working_files/CUTADAPT/"
tab_s_path <- "files/CUTADAPT/"

for (i in me_objs) {
  tmp_name <- i
  temp1 <- read.table(paste0(tab_r_path, tmp_name, "/", tmp_name, "_cutadapt_track.txt"), header = TRUE)
  temp1 <- dplyr::select(temp1, 1, 2, 6)
  colnames(temp1)[2] <- "raw_reads"
  colnames(temp1)[3] <- "cutadapt_reads"
  write.table(temp1, paste0(tab_s_path, tmp_name, "_cutadapt_track.txt"), sep = "\t", quote = FALSE, row.names = FALSE)

  temp1$per_reads_kept <- round(temp1$cutadapt_reads/temp1$raw_reads, digits = 3)
  tmp_tab_name <- paste0(tmp_name, "_seq_table")
  assign(tmp_tab_name, temp1)
  rm(list = ls(pattern = "tmp_"))
}  
```

::: {.panel-tabset .nav-pills .nav-fill}

# SSU

```{r}
#| echo: false
#| eval: true
#| label: tbl-make-cutadapt-rc-table-ssu
#| tbl-cap: "Raw read count vs. reads remaining after cutadapt."
make_cutadapt_rc_table(ssu_seq_table)
```

</br>

{{< downloadthis files/CUTADAPT/ssu_cutadapt_track.txt dname="ssu_cutadapt_track" label="Download table" icon="table" type="primary" >}}

```{r}
#| echo: false
#| eval: true
raw_sum <- sum(ssu_seq_table$raw_reads)
cutadapt_sum <- sum(ssu_seq_table$cutadapt_reads)
ratio <- 100*(cutadapt_sum / raw_sum)
```

We started with `r raw_sum` total reads. After primer removal we are left with `r round(ratio, digits = 2)`% of the reads we started with, or a total of `r cutadapt_sum` reads. 

# ITS

```{r}
#| echo: false
#| eval: true
#| label: tbl-make-cutadapt-rc-table-its
#| tbl-cap: "Raw read count vs. reads remaining after cutadapt."
make_cutadapt_rc_table(its_seq_table)
```

</br>

{{< downloadthis files/CUTADAPT/its_cutadapt_track.txt dname="its_cutadapt_track" label="Download table" icon="table" type="primary" >}}

```{r}
#| echo: false
#| eval: true
raw_sum <- sum(its_seq_table$raw_reads)
cutadapt_sum <- sum(its_seq_table$cutadapt_reads)
ratio <- 100*(cutadapt_sum / raw_sum)
```

We started with `r raw_sum` total reads. After primer removal we are left with `r round(ratio, digits = 2)`% of the reads we started with, or a total of `r cutadapt_sum` reads. 

# AMF

```{r}
#| echo: false
#| eval: true
#| label: tbl-make-cutadapt-rc-table-amf
#| tbl-cap: "Raw read count vs. reads remaining after cutadapt."
make_cutadapt_rc_table(amf_seq_table)
```

</br>

{{< downloadthis files/CUTADAPT/amf_cutadapt_track.txt dname="amf_cutadapt_track" label="Download table" icon="table" type="primary" >}}

```{r}
#| echo: false
#| eval: true
raw_sum <- sum(amf_seq_table$raw_reads)
cutadapt_sum <- sum(amf_seq_table$cutadapt_reads)
ratio <- 100*(cutadapt_sum / raw_sum)
```

We started with `r raw_sum` total reads. After primer removal we are left with `r round(ratio, digits = 2)`% of the reads we started with, or a total of `r cutadapt_sum` reads. 

# Oomycete

```{r}
#| echo: false
#| eval: true
#| label: tbl-make-cutadapt-rc-table-oo
#| tbl-cap: "Raw read count vs. reads remaining after cutadapt."
make_cutadapt_rc_table(oo_seq_table)
```

</br>

{{< downloadthis files/CUTADAPT/oo_cutadapt_track.txt dname="oo_cutadapt_track" label="Download table" icon="table" type="primary" >}}

```{r}
#| echo: false
#| eval: true
raw_sum <- sum(oo_seq_table$raw_reads)
cutadapt_sum <- sum(oo_seq_table$cutadapt_reads)
ratio <- 100*(cutadapt_sum / raw_sum)
```

We started with `r raw_sum` total reads. After primer removal we are left with `r round(ratio, digits = 2)`% of the reads we started with, or a total of `r cutadapt_sum` reads. 

:::

# Session Information

```{r}
sessionInfo()
devtools::session_info()
```

<details>

<summary>Click here for Session Information</summary>

{{< include include/data/_session_info.qmd >}}

</details>

{{< include include/code/_footer.qmd >}}

